{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "7d50c71c-d2a2-4677-a763-88c6bdd3123a",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "# TWAS\n",
    "a) Manhattan plot of glyco-TWAS results (unadjusted data); annotate points to indicate MR significance\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8bbe5663-26c0-4106-ad47-8d95bc26233c",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "## Combine Data\n",
    "TWAS and MR results generaated by pipelines were combined using the following commands\n",
    "```\n",
    "python /mnt/lustre/home/rl3328/rl3328/motor_qtl/combine_twas.py /mnt/lustre/lab/ctcn/hklein/motor_qtl_project/unadjusted_wp_2020/susie_twas/noQC_nonimputed_result/twas --type both --output ROSMAP_eQTL_wp_unadjusted\n",
    "```\n",
    "### Input\n",
    "Input files are all on the new AWS-based HPC\n",
    "* Combined TWAS file:`/mnt/lustre/lab/ctcn/hklein/motor_qtl_project/wp_2020/susie_twas/noQC_nonimputed_result/twas/ROSMAP_eQTL_wp_unadjusted.combined_twas.tsv.gz`\n",
    "* Combined MR file:`/mnt/lustre/lab/ctcn/hklein/motor_qtl_project/unadjusted_wp_2020/susie_twas/noQC_nonimputed_result/twas/ROSMAP_eQTL_wp_unadjusted.combined_mr.tsv.gz`\n",
    "* reference file for gene annotation: `/mnt/lustre/home/rl3328/rl3328/resource/Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.region_list.2025`\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01773a49",
   "metadata": {},
   "source": [
    "### Landscape\n",
    "\n",
    "* The TWAS results show substantial variation in the number of available genes across contexts. Monocyte, DLPFC, and AC have notably fewer genes available for analysis. In particular, the DLPFC bulk eQTL dataset includes only 57 genes, which likely explains why no genes reach significance under the most accurate predictive model for DLPFC bulk eQTL.\n",
    "* Consistently, the MR analysis also yields no significant genes for monocyte, DLPFC, or AC.\n",
    "\n",
    "* **Given these limitations, it may be more appropriate to combine bulk and single-nucleus (sn) data for DLPFC in downstream analyses.**\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09fa2c4a",
   "metadata": {},
   "source": [
    "#### TWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d55a570c-74e3-4642-86c7-64fa97d3cf82",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching core tidyverse packages\u001b[22m ──────────────────────── tidyverse 2.0.0 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.4     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.6\n",
      "\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.1     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.6.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2  \u001b[39m 3.5.2     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.3.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.4     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.2\n",
      "\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.2.0     \n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mbetween()\u001b[39m     masks \u001b[34mdata.table\u001b[39m::between()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m      masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfirst()\u001b[39m       masks \u001b[34mdata.table\u001b[39m::first()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mhour()\u001b[39m    masks \u001b[34mdata.table\u001b[39m::hour()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32misoweek()\u001b[39m masks \u001b[34mdata.table\u001b[39m::isoweek()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m         masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlast()\u001b[39m        masks \u001b[34mdata.table\u001b[39m::last()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mmday()\u001b[39m    masks \u001b[34mdata.table\u001b[39m::mday()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mminute()\u001b[39m  masks \u001b[34mdata.table\u001b[39m::minute()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mmonth()\u001b[39m   masks \u001b[34mdata.table\u001b[39m::month()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mquarter()\u001b[39m masks \u001b[34mdata.table\u001b[39m::quarter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32msecond()\u001b[39m  masks \u001b[34mdata.table\u001b[39m::second()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mpurrr\u001b[39m::\u001b[32mtranspose()\u001b[39m   masks \u001b[34mdata.table\u001b[39m::transpose()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mwday()\u001b[39m    masks \u001b[34mdata.table\u001b[39m::wday()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32mweek()\u001b[39m    masks \u001b[34mdata.table\u001b[39m::week()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32myday()\u001b[39m    masks \u001b[34mdata.table\u001b[39m::yday()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mlubridate\u001b[39m::\u001b[32myear()\u001b[39m    masks \u001b[34mdata.table\u001b[39m::year()\n",
      "\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n"
     ]
    }
   ],
   "source": [
    "library(data.table)\n",
    "library(tidyverse)\n",
    "twas <- fread('/mnt/lustre/lab/ctcn/hklein/motor_qtl_project/unadjusted_wp_2020/susie_twas/noQC_nonimputed_result/twas/ROSMAP_eQTL_wp_unadjusted.combined_twas.tsv.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5ce0b8c3",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>molecular_id</th><th scope=col>TSS</th><th scope=col>start</th><th scope=col>end</th><th scope=col>context</th><th scope=col>gwas_study</th><th scope=col>method</th><th scope=col>is_imputable</th><th scope=col>is_selected_method</th><th scope=col>rsq_cv</th><th scope=col>pval_cv</th><th scope=col>twas_z</th><th scope=col>twas_pval</th><th scope=col>type</th><th scope=col>block</th><th scope=col>method_selected_original</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>Inh_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>enet </td><td>TRUE</td><td>FALSE</td><td>-0.0002636756</td><td>3.460743e-01</td><td>-0.8321645</td><td>0.40531612</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>Inh_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>lasso</td><td>TRUE</td><td>FALSE</td><td>-0.0008220066</td><td>4.181955e-01</td><td>-0.9241237</td><td>0.35542192</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>Inh_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>mrash</td><td>TRUE</td><td>FALSE</td><td> 0.0074837805</td><td>4.222085e-02</td><td>-2.4786840</td><td>0.01318681</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>Inh_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>susie</td><td>TRUE</td><td> TRUE</td><td> 0.0157468862</td><td>5.809734e-03</td><td>-0.6523170</td><td>0.51419672</td><td>eQTL</td><td>chr10_100331627_104378781</td><td> TRUE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>PCC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>enet </td><td>TRUE</td><td>FALSE</td><td> 0.1176354683</td><td>7.716102e-14</td><td>-0.2702251</td><td>0.78698710</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>PCC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>lasso</td><td>TRUE</td><td>FALSE</td><td> 0.1157138583</td><td>1.253604e-13</td><td>-0.1913938</td><td>0.84821705</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " chr & molecular\\_id & TSS & start & end & context & gwas\\_study & method & is\\_imputable & is\\_selected\\_method & rsq\\_cv & pval\\_cv & twas\\_z & twas\\_pval & type & block & method\\_selected\\_original & region & study\\_context & source\\_file\\\\\n",
       " <int> & <chr> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <lgl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & Inh\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & enet  & TRUE & FALSE & -0.0002636756 & 3.460743e-01 & -0.8321645 & 0.40531612 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & Inh\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & lasso & TRUE & FALSE & -0.0008220066 & 4.181955e-01 & -0.9241237 & 0.35542192 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & Inh\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & mrash & TRUE & FALSE &  0.0074837805 & 4.222085e-02 & -2.4786840 & 0.01318681 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & Inh\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & susie & TRUE &  TRUE &  0.0157468862 & 5.809734e-03 & -0.6523170 & 0.51419672 & eQTL & chr10\\_100331627\\_104378781 &  TRUE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & PCC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & enet  & TRUE & FALSE &  0.1176354683 & 7.716102e-14 & -0.2702251 & 0.78698710 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & PCC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & lasso & TRUE & FALSE &  0.1157138583 & 1.253604e-13 & -0.1913938 & 0.84821705 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 20\n",
       "\n",
       "| chr &lt;int&gt; | molecular_id &lt;chr&gt; | TSS &lt;int&gt; | start &lt;int&gt; | end &lt;int&gt; | context &lt;chr&gt; | gwas_study &lt;chr&gt; | method &lt;chr&gt; | is_imputable &lt;lgl&gt; | is_selected_method &lt;lgl&gt; | rsq_cv &lt;dbl&gt; | pval_cv &lt;dbl&gt; | twas_z &lt;dbl&gt; | twas_pval &lt;dbl&gt; | type &lt;chr&gt; | block &lt;chr&gt; | method_selected_original &lt;lgl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | Inh_DeJager_eQTL | unadjusted_wp_2020 | enet  | TRUE | FALSE | -0.0002636756 | 3.460743e-01 | -0.8321645 | 0.40531612 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | Inh_DeJager_eQTL | unadjusted_wp_2020 | lasso | TRUE | FALSE | -0.0008220066 | 4.181955e-01 | -0.9241237 | 0.35542192 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | Inh_DeJager_eQTL | unadjusted_wp_2020 | mrash | TRUE | FALSE |  0.0074837805 | 4.222085e-02 | -2.4786840 | 0.01318681 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | Inh_DeJager_eQTL | unadjusted_wp_2020 | susie | TRUE |  TRUE |  0.0157468862 | 5.809734e-03 | -0.6523170 | 0.51419672 | eQTL | chr10_100331627_104378781 |  TRUE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | PCC_DeJager_eQTL | unadjusted_wp_2020 | enet  | TRUE | FALSE |  0.1176354683 | 7.716102e-14 | -0.2702251 | 0.78698710 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | PCC_DeJager_eQTL | unadjusted_wp_2020 | lasso | TRUE | FALSE |  0.1157138583 | 1.253604e-13 | -0.1913938 | 0.84821705 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "\n"
      ],
      "text/plain": [
       "  chr molecular_id    TSS       start    end       context         \n",
       "1 10  ENSG00000055950 100987515 99320000 102120000 Inh_DeJager_eQTL\n",
       "2 10  ENSG00000055950 100987515 99320000 102120000 Inh_DeJager_eQTL\n",
       "3 10  ENSG00000055950 100987515 99320000 102120000 Inh_DeJager_eQTL\n",
       "4 10  ENSG00000055950 100987515 99320000 102120000 Inh_DeJager_eQTL\n",
       "5 10  ENSG00000055950 100987515 99320000 102120000 PCC_DeJager_eQTL\n",
       "6 10  ENSG00000055950 100987515 99320000 102120000 PCC_DeJager_eQTL\n",
       "  gwas_study         method is_imputable is_selected_method rsq_cv       \n",
       "1 unadjusted_wp_2020 enet   TRUE         FALSE              -0.0002636756\n",
       "2 unadjusted_wp_2020 lasso  TRUE         FALSE              -0.0008220066\n",
       "3 unadjusted_wp_2020 mrash  TRUE         FALSE               0.0074837805\n",
       "4 unadjusted_wp_2020 susie  TRUE          TRUE               0.0157468862\n",
       "5 unadjusted_wp_2020 enet   TRUE         FALSE               0.1176354683\n",
       "6 unadjusted_wp_2020 lasso  TRUE         FALSE               0.1157138583\n",
       "  pval_cv      twas_z     twas_pval  type block                    \n",
       "1 3.460743e-01 -0.8321645 0.40531612 eQTL chr10_100331627_104378781\n",
       "2 4.181955e-01 -0.9241237 0.35542192 eQTL chr10_100331627_104378781\n",
       "3 4.222085e-02 -2.4786840 0.01318681 eQTL chr10_100331627_104378781\n",
       "4 5.809734e-03 -0.6523170 0.51419672 eQTL chr10_100331627_104378781\n",
       "5 7.716102e-14 -0.2702251 0.78698710 eQTL chr10_100331627_104378781\n",
       "6 1.253604e-13 -0.1913938 0.84821705 eQTL chr10_100331627_104378781\n",
       "  method_selected_original region                    study_context \n",
       "1 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "2 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "3 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "4  TRUE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "5 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "6 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "  source_file                                                    \n",
       "1 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "2 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "3 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "4 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "5 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "6 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>170518</li><li>20</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 170518\n",
       "\\item 20\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 170518\n",
       "2. 20\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 170518     20"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(twas)\n",
    "dim(twas)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "668b29f4-54df-4efb-b3bd-1f3417e11108",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Inh_DeJager_eQTL'</li><li>'PCC_DeJager_eQTL'</li><li>'Mic_DeJager_eQTL'</li><li>'Exc_DeJager_eQTL'</li><li>'Oli_DeJager_eQTL'</li><li>'Ast_DeJager_eQTL'</li><li>'OPC_DeJager_eQTL'</li><li>'AC_DeJager_eQTL'</li><li>'DLPFC_DeJager_eQTL'</li><li>'monocyte_ROSMAP_eQTL'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Inh\\_DeJager\\_eQTL'\n",
       "\\item 'PCC\\_DeJager\\_eQTL'\n",
       "\\item 'Mic\\_DeJager\\_eQTL'\n",
       "\\item 'Exc\\_DeJager\\_eQTL'\n",
       "\\item 'Oli\\_DeJager\\_eQTL'\n",
       "\\item 'Ast\\_DeJager\\_eQTL'\n",
       "\\item 'OPC\\_DeJager\\_eQTL'\n",
       "\\item 'AC\\_DeJager\\_eQTL'\n",
       "\\item 'DLPFC\\_DeJager\\_eQTL'\n",
       "\\item 'monocyte\\_ROSMAP\\_eQTL'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Inh_DeJager_eQTL'\n",
       "2. 'PCC_DeJager_eQTL'\n",
       "3. 'Mic_DeJager_eQTL'\n",
       "4. 'Exc_DeJager_eQTL'\n",
       "5. 'Oli_DeJager_eQTL'\n",
       "6. 'Ast_DeJager_eQTL'\n",
       "7. 'OPC_DeJager_eQTL'\n",
       "8. 'AC_DeJager_eQTL'\n",
       "9. 'DLPFC_DeJager_eQTL'\n",
       "10. 'monocyte_ROSMAP_eQTL'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Inh_DeJager_eQTL\"     \"PCC_DeJager_eQTL\"     \"Mic_DeJager_eQTL\"    \n",
       " [4] \"Exc_DeJager_eQTL\"     \"Oli_DeJager_eQTL\"     \"Ast_DeJager_eQTL\"    \n",
       " [7] \"OPC_DeJager_eQTL\"     \"AC_DeJager_eQTL\"      \"DLPFC_DeJager_eQTL\"  \n",
       "[10] \"monocyte_ROSMAP_eQTL\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "14107"
      ],
      "text/latex": [
       "14107"
      ],
      "text/markdown": [
       "14107"
      ],
      "text/plain": [
       "[1] 14107"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## TWAS analysis were performed across 10 contexts and 13872 genes\n",
    "twas %>% pull(context) |> unique()#%>%length\n",
    "\n",
    "twas %>% pull(molecular_id)%>%unique()%>%length\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1ef0fe8a",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 10 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>context</th><th scope=col>gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>AC_DeJager_eQTL     </td><td>   60</td></tr>\n",
       "\t<tr><td>Ast_DeJager_eQTL    </td><td> 5922</td></tr>\n",
       "\t<tr><td>DLPFC_DeJager_eQTL  </td><td>   57</td></tr>\n",
       "\t<tr><td>Exc_DeJager_eQTL    </td><td> 8273</td></tr>\n",
       "\t<tr><td>Inh_DeJager_eQTL    </td><td> 6450</td></tr>\n",
       "\t<tr><td>Mic_DeJager_eQTL    </td><td> 2770</td></tr>\n",
       "\t<tr><td>OPC_DeJager_eQTL    </td><td> 3054</td></tr>\n",
       "\t<tr><td>Oli_DeJager_eQTL    </td><td> 5450</td></tr>\n",
       "\t<tr><td>PCC_DeJager_eQTL    </td><td>10573</td></tr>\n",
       "\t<tr><td>monocyte_ROSMAP_eQTL</td><td>    3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 10 × 2\n",
       "\\begin{tabular}{ll}\n",
       " context & gene\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t AC\\_DeJager\\_eQTL      &    60\\\\\n",
       "\t Ast\\_DeJager\\_eQTL     &  5922\\\\\n",
       "\t DLPFC\\_DeJager\\_eQTL   &    57\\\\\n",
       "\t Exc\\_DeJager\\_eQTL     &  8273\\\\\n",
       "\t Inh\\_DeJager\\_eQTL     &  6450\\\\\n",
       "\t Mic\\_DeJager\\_eQTL     &  2770\\\\\n",
       "\t OPC\\_DeJager\\_eQTL     &  3054\\\\\n",
       "\t Oli\\_DeJager\\_eQTL     &  5450\\\\\n",
       "\t PCC\\_DeJager\\_eQTL     & 10573\\\\\n",
       "\t monocyte\\_ROSMAP\\_eQTL &     3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 10 × 2\n",
       "\n",
       "| context &lt;chr&gt; | gene &lt;int&gt; |\n",
       "|---|---|\n",
       "| AC_DeJager_eQTL      |    60 |\n",
       "| Ast_DeJager_eQTL     |  5922 |\n",
       "| DLPFC_DeJager_eQTL   |    57 |\n",
       "| Exc_DeJager_eQTL     |  8273 |\n",
       "| Inh_DeJager_eQTL     |  6450 |\n",
       "| Mic_DeJager_eQTL     |  2770 |\n",
       "| OPC_DeJager_eQTL     |  3054 |\n",
       "| Oli_DeJager_eQTL     |  5450 |\n",
       "| PCC_DeJager_eQTL     | 10573 |\n",
       "| monocyte_ROSMAP_eQTL |     3 |\n",
       "\n"
      ],
      "text/plain": [
       "   context              gene \n",
       "1  AC_DeJager_eQTL         60\n",
       "2  Ast_DeJager_eQTL      5922\n",
       "3  DLPFC_DeJager_eQTL      57\n",
       "4  Exc_DeJager_eQTL      8273\n",
       "5  Inh_DeJager_eQTL      6450\n",
       "6  Mic_DeJager_eQTL      2770\n",
       "7  OPC_DeJager_eQTL      3054\n",
       "8  Oli_DeJager_eQTL      5450\n",
       "9  PCC_DeJager_eQTL     10573\n",
       "10 monocyte_ROSMAP_eQTL     3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## available genes by contexts\n",
    "twas |> group_by(context) |> summarize(gene = n_distinct(molecular_id))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "65d8afca-6867-416a-8e9f-e39bf07064b5",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "4423"
      ],
      "text/latex": [
       "4423"
      ],
      "text/markdown": [
       "4423"
      ],
      "text/plain": [
       "[1] 4423"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "twas_res_sig <- twas %>% filter(is_selected_method==\"TRUE\")%>%filter(twas_pval<=0.05)#%>%pull(context)%>%unique()\n",
    "twas_res_sig|> pull(molecular_id)%>%unique()%>%length\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a06ae9b6-477c-4b77-97c9-8763ab5e0bb8",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "57"
      ],
      "text/latex": [
       "57"
      ],
      "text/markdown": [
       "57"
      ],
      "text/plain": [
       "[1] 57"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "twas_DLPFC_bulk = twas |> filter(context == 'DLPFC_DeJager_eQTL') |> pull(molecular_id)%>%unique()%>%length\n",
    "twas_DLPFC_bulk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3e94ecfc",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "11546"
      ],
      "text/latex": [
       "11546"
      ],
      "text/markdown": [
       "11546"
      ],
      "text/plain": [
       "[1] 11546"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "twas_DLPFC = twas|> filter(!context %in% c(\"monocyte_ROSMAP_eQTL\", \"PCC_DeJager_eQTL\", \"AC_DeJager_eQTL\")) |> pull(molecular_id)%>%unique()%>%length\n",
    "twas_DLPFC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "e3a9a9d5",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 0 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>molecular_id</th><th scope=col>TSS</th><th scope=col>start</th><th scope=col>end</th><th scope=col>context</th><th scope=col>gwas_study</th><th scope=col>method</th><th scope=col>is_imputable</th><th scope=col>is_selected_method</th><th scope=col>rsq_cv</th><th scope=col>pval_cv</th><th scope=col>twas_z</th><th scope=col>twas_pval</th><th scope=col>type</th><th scope=col>block</th><th scope=col>method_selected_original</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 0 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " chr & molecular\\_id & TSS & start & end & context & gwas\\_study & method & is\\_imputable & is\\_selected\\_method & rsq\\_cv & pval\\_cv & twas\\_z & twas\\_pval & type & block & method\\_selected\\_original & region & study\\_context & source\\_file\\\\\n",
       " <int> & <chr> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <lgl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 0 × 20\n",
       "\n",
       "| chr &lt;int&gt; | molecular_id &lt;chr&gt; | TSS &lt;int&gt; | start &lt;int&gt; | end &lt;int&gt; | context &lt;chr&gt; | gwas_study &lt;chr&gt; | method &lt;chr&gt; | is_imputable &lt;lgl&gt; | is_selected_method &lt;lgl&gt; | rsq_cv &lt;dbl&gt; | pval_cv &lt;dbl&gt; | twas_z &lt;dbl&gt; | twas_pval &lt;dbl&gt; | type &lt;chr&gt; | block &lt;chr&gt; | method_selected_original &lt;lgl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     chr molecular_id TSS start end context gwas_study method is_imputable\n",
       "     is_selected_method rsq_cv pval_cv twas_z twas_pval type block\n",
       "     method_selected_original region study_context source_file"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>molecular_id</th><th scope=col>TSS</th><th scope=col>start</th><th scope=col>end</th><th scope=col>context</th><th scope=col>gwas_study</th><th scope=col>method</th><th scope=col>is_imputable</th><th scope=col>is_selected_method</th><th scope=col>rsq_cv</th><th scope=col>pval_cv</th><th scope=col>twas_z</th><th scope=col>twas_pval</th><th scope=col>type</th><th scope=col>block</th><th scope=col>method_selected_original</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>3</td><td>ENSG00000044524</td><td>89107621</td><td>86600000</td><td>93720000</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>mrash</td><td>TRUE</td><td>FALSE</td><td>0.04285215</td><td>2.935038e-09</td><td>-5.764569</td><td>8.186698e-09</td><td>eQTL</td><td>chr3_88209300_94537986</td><td>FALSE</td><td>chr3_88209300_94537986</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr3_88209300_94537986.twas.tsv.gz</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " chr & molecular\\_id & TSS & start & end & context & gwas\\_study & method & is\\_imputable & is\\_selected\\_method & rsq\\_cv & pval\\_cv & twas\\_z & twas\\_pval & type & block & method\\_selected\\_original & region & study\\_context & source\\_file\\\\\n",
       " <int> & <chr> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <lgl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 3 & ENSG00000044524 & 89107621 & 86600000 & 93720000 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & mrash & TRUE & FALSE & 0.04285215 & 2.935038e-09 & -5.764569 & 8.186698e-09 & eQTL & chr3\\_88209300\\_94537986 & FALSE & chr3\\_88209300\\_94537986 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr3\\_88209300\\_94537986.twas.tsv.gz\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 20\n",
       "\n",
       "| chr &lt;int&gt; | molecular_id &lt;chr&gt; | TSS &lt;int&gt; | start &lt;int&gt; | end &lt;int&gt; | context &lt;chr&gt; | gwas_study &lt;chr&gt; | method &lt;chr&gt; | is_imputable &lt;lgl&gt; | is_selected_method &lt;lgl&gt; | rsq_cv &lt;dbl&gt; | pval_cv &lt;dbl&gt; | twas_z &lt;dbl&gt; | twas_pval &lt;dbl&gt; | type &lt;chr&gt; | block &lt;chr&gt; | method_selected_original &lt;lgl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 3 | ENSG00000044524 | 89107621 | 86600000 | 93720000 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | mrash | TRUE | FALSE | 0.04285215 | 2.935038e-09 | -5.764569 | 8.186698e-09 | eQTL | chr3_88209300_94537986 | FALSE | chr3_88209300_94537986 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr3_88209300_94537986.twas.tsv.gz |\n",
       "\n"
      ],
      "text/plain": [
       "  chr molecular_id    TSS      start    end      context           \n",
       "1 3   ENSG00000044524 89107621 86600000 93720000 DLPFC_DeJager_eQTL\n",
       "  gwas_study         method is_imputable is_selected_method rsq_cv    \n",
       "1 unadjusted_wp_2020 mrash  TRUE         FALSE              0.04285215\n",
       "  pval_cv      twas_z    twas_pval    type block                 \n",
       "1 2.935038e-09 -5.764569 8.186698e-09 eQTL chr3_88209300_94537986\n",
       "  method_selected_original region                 study_context \n",
       "1 FALSE                    chr3_88209300_94537986 ROSMAP_eQTL_wp\n",
       "  source_file                                                 \n",
       "1 ROSMAP_eQTL_wp_unadjusted.chr3_88209300_94537986.twas.tsv.gz"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## need to run after unadj_cutoff was calculated,  shown here for better review\n",
    "## Best model-0 gene significant after bonferroni correction\n",
    "twas |> filter(is_selected_method, twas_pval < unadj_cutoff, str_detect(context, \"DLPFC\"))\n",
    "## All models-1 gene(EPHA3) significant after bonferroni correction\n",
    "twas |> filter(twas_pval < unadj_cutoff, str_detect(context, \"DLPFC\"))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a64c2d9f",
   "metadata": {},
   "source": [
    "#### MR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "89201606-c5eb-4bfb-9aaf-b026aec56c0f",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "mr <- fread('/mnt/lustre/lab/ctcn/hklein/motor_qtl_project/unadjusted_wp_2020/susie_twas/noQC_nonimputed_result/twas/ROSMAP_eQTL_wp_unadjusted.combined_mr.tsv.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6303aa4a-76c7-4416-82d9-6fe4ec293aa1",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "mr_sig_unadj <- mr %>% .[\n",
    "  meta_pval < 0.05 / .N &          # Bonferroni correction\n",
    "  cpip >= 0.7 &                    # strong causal evidence\n",
    "  num_CS >= 1 &                    # as quested by Hans and Gao on Oct 24, 2025 to get a loose version\n",
    "  Q_pval > 0.01 &                  # no heterogeneity\n",
    "  I2 < 0.4                         # low heterogeneity\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f3f48a47-880f-4e9b-aa3e-ad5faf4a1e4b",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Q</th><th scope=col>Q_pval</th><th scope=col>I2</th><th scope=col>context</th><th scope=col>cpip</th><th scope=col>gene_name</th><th scope=col>gwas_study</th><th scope=col>meta_eff</th><th scope=col>meta_pval</th><th scope=col>num_CS</th><th scope=col>num_IV</th><th scope=col>se_meta_eff</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1.823</td><td>0.402</td><td>0</td><td>Exc_DeJager_eQTL</td><td>0.983</td><td>ENSG00000138111</td><td>unadjusted_wp_2020</td><td> 0.000</td><td>0</td><td>3</td><td> 21</td><td>0</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz</td></tr>\n",
       "\t<tr><td>0.000</td><td>1.000</td><td>0</td><td>Inh_DeJager_eQTL</td><td>0.956</td><td>ENSG00000138111</td><td>unadjusted_wp_2020</td><td> 0.000</td><td>0</td><td>1</td><td> 16</td><td>0</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz</td></tr>\n",
       "\t<tr><td>0.000</td><td>1.000</td><td>0</td><td>Ast_DeJager_eQTL</td><td>1.012</td><td>ENSG00000171206</td><td>unadjusted_wp_2020</td><td> 0.001</td><td>0</td><td>1</td><td> 59</td><td>0</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz</td></tr>\n",
       "\t<tr><td>0.001</td><td>0.979</td><td>0</td><td>Exc_DeJager_eQTL</td><td>1.026</td><td>ENSG00000171206</td><td>unadjusted_wp_2020</td><td> 0.001</td><td>0</td><td>2</td><td>144</td><td>0</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz</td></tr>\n",
       "\t<tr><td>0.000</td><td>1.000</td><td>0</td><td>Exc_DeJager_eQTL</td><td>0.960</td><td>ENSG00000078403</td><td>unadjusted_wp_2020</td><td>-0.001</td><td>0</td><td>1</td><td>  8</td><td>0</td><td>chr10_20625737_23053090  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_20625737_23053090.mr_result.tsv.gz  </td></tr>\n",
       "\t<tr><td>0.000</td><td>1.000</td><td>0</td><td>PCC_DeJager_eQTL</td><td>0.995</td><td>ENSG00000148572</td><td>unadjusted_wp_2020</td><td>-0.002</td><td>0</td><td>1</td><td> 92</td><td>0</td><td>chr10_62446953_64035328  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_62446953_64035328.mr_result.tsv.gz  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 15\n",
       "\\begin{tabular}{lllllllllllllll}\n",
       " Q & Q\\_pval & I2 & context & cpip & gene\\_name & gwas\\_study & meta\\_eff & meta\\_pval & num\\_CS & num\\_IV & se\\_meta\\_eff & region & study\\_context & source\\_file\\\\\n",
       " <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <chr> & <chr> & <dbl> & <dbl> & <int> & <int> & <dbl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1.823 & 0.402 & 0 & Exc\\_DeJager\\_eQTL & 0.983 & ENSG00000138111 & unadjusted\\_wp\\_2020 &  0.000 & 0 & 3 &  21 & 0 & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.mr\\_result.tsv.gz\\\\\n",
       "\t 0.000 & 1.000 & 0 & Inh\\_DeJager\\_eQTL & 0.956 & ENSG00000138111 & unadjusted\\_wp\\_2020 &  0.000 & 0 & 1 &  16 & 0 & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.mr\\_result.tsv.gz\\\\\n",
       "\t 0.000 & 1.000 & 0 & Ast\\_DeJager\\_eQTL & 1.012 & ENSG00000171206 & unadjusted\\_wp\\_2020 &  0.001 & 0 & 1 &  59 & 0 & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.mr\\_result.tsv.gz\\\\\n",
       "\t 0.001 & 0.979 & 0 & Exc\\_DeJager\\_eQTL & 1.026 & ENSG00000171206 & unadjusted\\_wp\\_2020 &  0.001 & 0 & 2 & 144 & 0 & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.mr\\_result.tsv.gz\\\\\n",
       "\t 0.000 & 1.000 & 0 & Exc\\_DeJager\\_eQTL & 0.960 & ENSG00000078403 & unadjusted\\_wp\\_2020 & -0.001 & 0 & 1 &   8 & 0 & chr10\\_20625737\\_23053090   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_20625737\\_23053090.mr\\_result.tsv.gz  \\\\\n",
       "\t 0.000 & 1.000 & 0 & PCC\\_DeJager\\_eQTL & 0.995 & ENSG00000148572 & unadjusted\\_wp\\_2020 & -0.002 & 0 & 1 &  92 & 0 & chr10\\_62446953\\_64035328   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_62446953\\_64035328.mr\\_result.tsv.gz  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 15\n",
       "\n",
       "| Q &lt;dbl&gt; | Q_pval &lt;dbl&gt; | I2 &lt;dbl&gt; | context &lt;chr&gt; | cpip &lt;dbl&gt; | gene_name &lt;chr&gt; | gwas_study &lt;chr&gt; | meta_eff &lt;dbl&gt; | meta_pval &lt;dbl&gt; | num_CS &lt;int&gt; | num_IV &lt;int&gt; | se_meta_eff &lt;dbl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1.823 | 0.402 | 0 | Exc_DeJager_eQTL | 0.983 | ENSG00000138111 | unadjusted_wp_2020 |  0.000 | 0 | 3 |  21 | 0 | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz |\n",
       "| 0.000 | 1.000 | 0 | Inh_DeJager_eQTL | 0.956 | ENSG00000138111 | unadjusted_wp_2020 |  0.000 | 0 | 1 |  16 | 0 | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz |\n",
       "| 0.000 | 1.000 | 0 | Ast_DeJager_eQTL | 1.012 | ENSG00000171206 | unadjusted_wp_2020 |  0.001 | 0 | 1 |  59 | 0 | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz |\n",
       "| 0.001 | 0.979 | 0 | Exc_DeJager_eQTL | 1.026 | ENSG00000171206 | unadjusted_wp_2020 |  0.001 | 0 | 2 | 144 | 0 | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz |\n",
       "| 0.000 | 1.000 | 0 | Exc_DeJager_eQTL | 0.960 | ENSG00000078403 | unadjusted_wp_2020 | -0.001 | 0 | 1 |   8 | 0 | chr10_20625737_23053090   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_20625737_23053090.mr_result.tsv.gz   |\n",
       "| 0.000 | 1.000 | 0 | PCC_DeJager_eQTL | 0.995 | ENSG00000148572 | unadjusted_wp_2020 | -0.002 | 0 | 1 |  92 | 0 | chr10_62446953_64035328   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_62446953_64035328.mr_result.tsv.gz   |\n",
       "\n"
      ],
      "text/plain": [
       "  Q     Q_pval I2 context          cpip  gene_name       gwas_study        \n",
       "1 1.823 0.402  0  Exc_DeJager_eQTL 0.983 ENSG00000138111 unadjusted_wp_2020\n",
       "2 0.000 1.000  0  Inh_DeJager_eQTL 0.956 ENSG00000138111 unadjusted_wp_2020\n",
       "3 0.000 1.000  0  Ast_DeJager_eQTL 1.012 ENSG00000171206 unadjusted_wp_2020\n",
       "4 0.001 0.979  0  Exc_DeJager_eQTL 1.026 ENSG00000171206 unadjusted_wp_2020\n",
       "5 0.000 1.000  0  Exc_DeJager_eQTL 0.960 ENSG00000078403 unadjusted_wp_2020\n",
       "6 0.000 1.000  0  PCC_DeJager_eQTL 0.995 ENSG00000148572 unadjusted_wp_2020\n",
       "  meta_eff meta_pval num_CS num_IV se_meta_eff region                   \n",
       "1  0.000   0         3       21    0           chr10_100331627_104378781\n",
       "2  0.000   0         1       16    0           chr10_100331627_104378781\n",
       "3  0.001   0         1       59    0           chr10_100331627_104378781\n",
       "4  0.001   0         2      144    0           chr10_100331627_104378781\n",
       "5 -0.001   0         1        8    0           chr10_20625737_23053090  \n",
       "6 -0.002   0         1       92    0           chr10_62446953_64035328  \n",
       "  study_context \n",
       "1 ROSMAP_eQTL_wp\n",
       "2 ROSMAP_eQTL_wp\n",
       "3 ROSMAP_eQTL_wp\n",
       "4 ROSMAP_eQTL_wp\n",
       "5 ROSMAP_eQTL_wp\n",
       "6 ROSMAP_eQTL_wp\n",
       "  source_file                                                         \n",
       "1 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz\n",
       "2 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz\n",
       "3 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz\n",
       "4 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz\n",
       "5 ROSMAP_eQTL_wp_unadjusted.chr10_20625737_23053090.mr_result.tsv.gz  \n",
       "6 ROSMAP_eQTL_wp_unadjusted.chr10_62446953_64035328.mr_result.tsv.gz  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>216</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 216\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 216\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 216  15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(mr_sig_unadj)\n",
    "dim(mr_sig_unadj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "da1af503",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "155"
      ],
      "text/latex": [
       "155"
      ],
      "text/markdown": [
       "155"
      ],
      "text/plain": [
       "[1] 155"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mr_sig_unadj |> pull(gene_name)%>%unique()%>%length"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "9c239e7f",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "92"
      ],
      "text/latex": [
       "92"
      ],
      "text/markdown": [
       "92"
      ],
      "text/plain": [
       "[1] 92"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mr_sig_unadj |> filter(!context %in% c(\"monocyte_ROSMAP_eQTL\", \"PCC_DeJager_eQTL\", \"AC_DeJager_eQTL\")) |> pull(gene_name)%>%unique()%>%length"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "37214743",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 7 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>context</th><th scope=col>gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Ast_DeJager_eQTL</td><td>22</td></tr>\n",
       "\t<tr><td>Exc_DeJager_eQTL</td><td>45</td></tr>\n",
       "\t<tr><td>Inh_DeJager_eQTL</td><td>29</td></tr>\n",
       "\t<tr><td>Mic_DeJager_eQTL</td><td> 6</td></tr>\n",
       "\t<tr><td>OPC_DeJager_eQTL</td><td> 7</td></tr>\n",
       "\t<tr><td>Oli_DeJager_eQTL</td><td>27</td></tr>\n",
       "\t<tr><td>PCC_DeJager_eQTL</td><td>80</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 7 × 2\n",
       "\\begin{tabular}{ll}\n",
       " context & gene\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Ast\\_DeJager\\_eQTL & 22\\\\\n",
       "\t Exc\\_DeJager\\_eQTL & 45\\\\\n",
       "\t Inh\\_DeJager\\_eQTL & 29\\\\\n",
       "\t Mic\\_DeJager\\_eQTL &  6\\\\\n",
       "\t OPC\\_DeJager\\_eQTL &  7\\\\\n",
       "\t Oli\\_DeJager\\_eQTL & 27\\\\\n",
       "\t PCC\\_DeJager\\_eQTL & 80\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 7 × 2\n",
       "\n",
       "| context &lt;chr&gt; | gene &lt;int&gt; |\n",
       "|---|---|\n",
       "| Ast_DeJager_eQTL | 22 |\n",
       "| Exc_DeJager_eQTL | 45 |\n",
       "| Inh_DeJager_eQTL | 29 |\n",
       "| Mic_DeJager_eQTL |  6 |\n",
       "| OPC_DeJager_eQTL |  7 |\n",
       "| Oli_DeJager_eQTL | 27 |\n",
       "| PCC_DeJager_eQTL | 80 |\n",
       "\n"
      ],
      "text/plain": [
       "  context          gene\n",
       "1 Ast_DeJager_eQTL 22  \n",
       "2 Exc_DeJager_eQTL 45  \n",
       "3 Inh_DeJager_eQTL 29  \n",
       "4 Mic_DeJager_eQTL  6  \n",
       "5 OPC_DeJager_eQTL  7  \n",
       "6 Oli_DeJager_eQTL 27  \n",
       "7 PCC_DeJager_eQTL 80  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mr_sig_unadj |> group_by(context) |> summarize(gene = n_distinct(gene_name))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "567e49d6",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "61"
      ],
      "text/latex": [
       "61"
      ],
      "text/markdown": [
       "61"
      ],
      "text/plain": [
       "[1] 61"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## Regardless of context, there are 60 genes have both TWAS significant and MR significant asscoaitions\n",
    "intersect(twas_unadj_sig$molecular_id, mr_sig_unadj$gene_name) |> unique()|> length()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c45de04b-ce4d-467a-b981-a1ec5b4511eb",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "### summarize significant TWAS & MR genes\n",
    "\n",
    "* Significant TWAS associations were defined using two criteria: (i) the most accurate prediction model yielded a TWAS p-value below the Bonferroni-corrected threshold, or (ii) at least half of the prediction models produced p-values below this threshold.\n",
    "* Among the 14,075 total genes, 11,538 are from DLPFC. Overall, 133 genes show significant TWAS associations, of which 84 are significant in DLPFC tissues\n",
    "\n",
    "* Mendelian randomization (MR) associations were considered significant if they met all of the following criteria: a meta-analysis p-value passing Bonferroni correction (meta-analysis p < 0.05 divided by the number of tested genes), strong causal evidence as indicated by a causal posterior inclusion probability (CPIP) ≥ 0.7, at least one credible set (num_CS ≥ 1), and no evidence of substantial heterogeneity (Cochran’s Q test p > 0.01 and I² < 0.4).\n",
    "* 154 genes show significant MR associations, of which 92 are significant in DLPFC tissues.\n",
    "* Regardless of context, 60 genes are both MR and TWAS significant. 55 genes are both MR and TWAS significant if matched by contexts, of which 23 are from DLPFC(both bulk and sn)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "51741b66-ce32-4b11-8989-71f41c6a4f0f",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "42612"
      ],
      "text/latex": [
       "42612"
      ],
      "text/markdown": [
       "42612"
      ],
      "text/plain": [
       "[1] 42612"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "n_imputable_genes <- twas %>%\n",
    "  filter(is_imputable) %>%                  # keep only imputable rows\n",
    "  distinct(molecular_id, context, gwas_study) %>%         # count unique gene-context pairs\n",
    "  nrow()\n",
    "n_imputable_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "607cbcd1-b856-4edb-95c3-fe521a8ff7a9",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "1.17337839106355e-06"
      ],
      "text/latex": [
       "1.17337839106355e-06"
      ],
      "text/markdown": [
       "1.17337839106355e-06"
      ],
      "text/plain": [
       "[1] 1.173378e-06"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Compute cutoff using Bonferroni-style correction\n",
    "unadj_cutoff <- 0.05 / n_imputable_genes\n",
    "unadj_cutoff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e78ef1f1-b8fe-42e8-973f-ad843b7357ed",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(molecular_id, context, gwas_study)`\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>139</li><li>25</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 139\n",
       "\\item 25\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 139\n",
       "2. 25\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 139  25"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "twas_unadj_sig <- twas %>%\n",
    "  mutate(sig = twas_pval < unadj_cutoff) %>%\n",
    "  group_by(molecular_id, context, gwas_study) %>%\n",
    "  summarise(\n",
    "    n_methods = n(),\n",
    "    n_sig = sum(sig, na.rm = T),\n",
    "    best_sig = any(sig & is_selected_method, na.rm = T),\n",
    "    .groups = \"drop\"\n",
    "  ) %>%\n",
    "  mutate(pass = (n_sig / n_methods >= 0.5) | best_sig) %>%\n",
    "  mutate(pass_strict = (n_sig / n_methods >= 0.5)) %>%\n",
    "  filter(pass) %>% left_join(twas) %>% arrange(twas_pval) %>% distinct(molecular_id, .keep_all = T)\n",
    "twas_unadj_sig %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "37bb6225",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "84"
      ],
      "text/latex": [
       "84"
      ],
      "text/markdown": [
       "84"
      ],
      "text/plain": [
       "[1] 84"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "twas_unadj_sig |> filter(!context %in% c(\"monocyte_ROSMAP_eQTL\", \"PCC_DeJager_eQTL\", \"AC_DeJager_eQTL\")) |> nrow()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "193c534f-c425-49a6-9b1d-97e364f6b9da",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "gene_ref <- fread('/mnt/lustre/home/rl3328/rl3328/resource/Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.region_list.2025')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "dd79014c-34fb-45ff-b051-bdf18c61ce38",
   "metadata": {
    "kernel": "R",
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# twas_unadj_sig |> left_join(gene_ref, by = c(\"molecular_id\" = \"gene_id\")) |> select(molecular_id, gene_name, everything())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33c9361d-3ab3-42a1-8cfd-2a1e2a5ae97f",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "## Figure data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "d2409cc8-d679-4919-8255-325ac354695f",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“\u001b[1m\u001b[22mThere were 214 warnings in `summarise()`.\n",
      "The first warning was:\n",
      "\u001b[1m\u001b[22m\u001b[36mℹ\u001b[39m In argument: `best_method_pval = min(twas_pval[is_selected_method], na.rm =\n",
      "  TRUE)`.\n",
      "\u001b[36mℹ\u001b[39m In group 202: `molecular_id = \"ENSG00000005189\"`, `context =\n",
      "  \"Exc_DeJager_eQTL\"`, `gwas_study = \"unadjusted_wp_2020\"`.\n",
      "Caused by warning in `min()`:\n",
      "\u001b[33m!\u001b[39m no non-missing arguments to min; returning Inf\n",
      "\u001b[1m\u001b[22m\u001b[36mℹ\u001b[39m Run `dplyr::last_dplyr_warnings()` to see the 213 remaining warnings.”\n"
     ]
    }
   ],
   "source": [
    "df <- twas %>%\n",
    "  mutate(sig = twas_pval < unadj_cutoff) %>%\n",
    "  group_by(molecular_id, context, gwas_study) %>%\n",
    "  summarise(\n",
    "    n_methods = n(),\n",
    "    n_sig = sum(sig, na.rm = TRUE),\n",
    "    best_pval = min(twas_pval, na.rm = TRUE),\n",
    "    best_method_pval = min(twas_pval[is_selected_method], na.rm = TRUE),\n",
    "    best_method_selected = any(sig & is_selected_method, na.rm = TRUE),\n",
    "    .groups = \"drop\"\n",
    "  ) %>%\n",
    "  mutate(\n",
    "    pass = (n_sig / n_methods >= 0.5) | best_method_selected, # pass defined by two threshold, best or half methods satisfied\n",
    "    final_pval = ifelse(n_sig / n_methods >= 0.5, best_pval, best_method_pval),\n",
    "    # fallback: if no selected method existed (Inf), use best_pval\n",
    "    final_pval = ifelse(is.infinite(final_pval), best_pval, final_pval)\n",
    "  ) %>%\n",
    "  left_join(twas, by = c(\"molecular_id\", \"context\", \"gwas_study\")) %>%\n",
    "  filter(twas_pval == final_pval) %>%\n",
    "  distinct(molecular_id, .keep_all = TRUE) %>%\n",
    "  mutate(\n",
    "    logp = -log10(twas_pval) * sign(twas_z),\n",
    "    pos = (start + end) / 2\n",
    "  ) %>%\n",
    "  merge(gene_ref, by.x = 'molecular_id', by.y = 'gene_id')\n",
    "\n",
    "# Order chromosomes\n",
    "df$chr <- factor(df$chr, levels = as.character(1:22))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "05d4225a",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>14107</li><li>33</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 14107\n",
       "\\item 33\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 14107\n",
       "2. 33\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 14107    33"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "bc6550ab",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>molecular_id</th><th scope=col>context</th><th scope=col>gwas_study</th><th scope=col>n_methods</th><th scope=col>n_sig</th><th scope=col>best_pval</th><th scope=col>best_method_pval</th><th scope=col>best_method_selected</th><th scope=col>pass</th><th scope=col>final_pval</th><th scope=col>⋯</th><th scope=col>method_selected_original</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th><th scope=col>logp</th><th scope=col>pos</th><th scope=col>#chr</th><th scope=col>TSS.y</th><th scope=col>TES</th><th scope=col>gene_name</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ENSG00000000419</td><td>Exc_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.004887676</td><td>0.634741658</td><td>FALSE</td><td>FALSE</td><td>0.634741658</td><td>⋯</td><td>TRUE</td><td>chr20_50319421_53436693 </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr20_50319421_53436693.twas.tsv.gz </td><td>-0.1974030</td><td> 51747434</td><td>chr20</td><td> 50958555</td><td> 50934867</td><td>DPM1    </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ENSG00000000457</td><td>Ast_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.314317590</td><td>0.666951750</td><td>FALSE</td><td>FALSE</td><td>0.666951750</td><td>⋯</td><td>TRUE</td><td>chr1_168438717_170228106</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz</td><td> 0.1759056</td><td>169540000</td><td>chr1 </td><td>169894267</td><td>169849631</td><td>SCYL3   </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>ENSG00000000460</td><td>Ast_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.550799973</td><td>0.681781965</td><td>FALSE</td><td>FALSE</td><td>0.681781965</td><td>⋯</td><td>TRUE</td><td>chr1_168438717_170228106</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz</td><td>-0.1663545</td><td>169540000</td><td>chr1 </td><td>169662007</td><td>169854080</td><td>C1orf112</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>ENSG00000000971</td><td>PCC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.006727627</td><td>0.006727627</td><td>FALSE</td><td>FALSE</td><td>0.006727627</td><td>⋯</td><td>TRUE</td><td>chr1_195599253_199271134</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr1_195599253_199271134.twas.tsv.gz</td><td> 2.1721381</td><td>195093752</td><td>chr1 </td><td>196652043</td><td>196747504</td><td>CFH     </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ENSG00000001036</td><td>Ast_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.125012981</td><td>0.125012981</td><td>FALSE</td><td>FALSE</td><td>0.125012981</td><td>⋯</td><td>TRUE</td><td>chr6_143399926_145632539</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr6_143399926_145632539.twas.tsv.gz</td><td>-0.9030449</td><td>143115860</td><td>chr6 </td><td>143511720</td><td>143494812</td><td>FUCA2   </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ENSG00000001084</td><td>Ast_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.166978326</td><td>0.851335412</td><td>FALSE</td><td>FALSE</td><td>0.851335412</td><td>⋯</td><td>TRUE</td><td>chr6_52730905_54027603  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr6_52730905_54027603.twas.tsv.gz  </td><td>-0.0698993</td><td> 53557156</td><td>chr6 </td><td> 53616970</td><td> 53497341</td><td>GCLC    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 33\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & molecular\\_id & context & gwas\\_study & n\\_methods & n\\_sig & best\\_pval & best\\_method\\_pval & best\\_method\\_selected & pass & final\\_pval & ⋯ & method\\_selected\\_original & region & study\\_context & source\\_file & logp & pos & \\#chr & TSS.y & TES & gene\\_name\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <int> & <dbl> & <dbl> & <lgl> & <lgl> & <dbl> & ⋯ & <lgl> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ENSG00000000419 & Exc\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.004887676 & 0.634741658 & FALSE & FALSE & 0.634741658 & ⋯ & TRUE & chr20\\_50319421\\_53436693  & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr20\\_50319421\\_53436693.twas.tsv.gz  & -0.1974030 &  51747434 & chr20 &  50958555 &  50934867 & DPM1    \\\\\n",
       "\t2 & ENSG00000000457 & Ast\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.314317590 & 0.666951750 & FALSE & FALSE & 0.666951750 & ⋯ & TRUE & chr1\\_168438717\\_170228106 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr1\\_168438717\\_170228106.twas.tsv.gz &  0.1759056 & 169540000 & chr1  & 169894267 & 169849631 & SCYL3   \\\\\n",
       "\t3 & ENSG00000000460 & Ast\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.550799973 & 0.681781965 & FALSE & FALSE & 0.681781965 & ⋯ & TRUE & chr1\\_168438717\\_170228106 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr1\\_168438717\\_170228106.twas.tsv.gz & -0.1663545 & 169540000 & chr1  & 169662007 & 169854080 & C1orf112\\\\\n",
       "\t4 & ENSG00000000971 & PCC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.006727627 & 0.006727627 & FALSE & FALSE & 0.006727627 & ⋯ & TRUE & chr1\\_195599253\\_199271134 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr1\\_195599253\\_199271134.twas.tsv.gz &  2.1721381 & 195093752 & chr1  & 196652043 & 196747504 & CFH     \\\\\n",
       "\t5 & ENSG00000001036 & Ast\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.125012981 & 0.125012981 & FALSE & FALSE & 0.125012981 & ⋯ & TRUE & chr6\\_143399926\\_145632539 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr6\\_143399926\\_145632539.twas.tsv.gz & -0.9030449 & 143115860 & chr6  & 143511720 & 143494812 & FUCA2   \\\\\n",
       "\t6 & ENSG00000001084 & Ast\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.166978326 & 0.851335412 & FALSE & FALSE & 0.851335412 & ⋯ & TRUE & chr6\\_52730905\\_54027603   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr6\\_52730905\\_54027603.twas.tsv.gz   & -0.0698993 &  53557156 & chr6  &  53616970 &  53497341 & GCLC    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 33\n",
       "\n",
       "| <!--/--> | molecular_id &lt;chr&gt; | context &lt;chr&gt; | gwas_study &lt;chr&gt; | n_methods &lt;int&gt; | n_sig &lt;int&gt; | best_pval &lt;dbl&gt; | best_method_pval &lt;dbl&gt; | best_method_selected &lt;lgl&gt; | pass &lt;lgl&gt; | final_pval &lt;dbl&gt; | ⋯ ⋯ | method_selected_original &lt;lgl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; | logp &lt;dbl&gt; | pos &lt;dbl&gt; | #chr &lt;chr&gt; | TSS.y &lt;int&gt; | TES &lt;int&gt; | gene_name &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ENSG00000000419 | Exc_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.004887676 | 0.634741658 | FALSE | FALSE | 0.634741658 | ⋯ | TRUE | chr20_50319421_53436693  | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr20_50319421_53436693.twas.tsv.gz  | -0.1974030 |  51747434 | chr20 |  50958555 |  50934867 | DPM1     |\n",
       "| 2 | ENSG00000000457 | Ast_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.314317590 | 0.666951750 | FALSE | FALSE | 0.666951750 | ⋯ | TRUE | chr1_168438717_170228106 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz |  0.1759056 | 169540000 | chr1  | 169894267 | 169849631 | SCYL3    |\n",
       "| 3 | ENSG00000000460 | Ast_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.550799973 | 0.681781965 | FALSE | FALSE | 0.681781965 | ⋯ | TRUE | chr1_168438717_170228106 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz | -0.1663545 | 169540000 | chr1  | 169662007 | 169854080 | C1orf112 |\n",
       "| 4 | ENSG00000000971 | PCC_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.006727627 | 0.006727627 | FALSE | FALSE | 0.006727627 | ⋯ | TRUE | chr1_195599253_199271134 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr1_195599253_199271134.twas.tsv.gz |  2.1721381 | 195093752 | chr1  | 196652043 | 196747504 | CFH      |\n",
       "| 5 | ENSG00000001036 | Ast_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.125012981 | 0.125012981 | FALSE | FALSE | 0.125012981 | ⋯ | TRUE | chr6_143399926_145632539 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr6_143399926_145632539.twas.tsv.gz | -0.9030449 | 143115860 | chr6  | 143511720 | 143494812 | FUCA2    |\n",
       "| 6 | ENSG00000001084 | Ast_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.166978326 | 0.851335412 | FALSE | FALSE | 0.851335412 | ⋯ | TRUE | chr6_52730905_54027603   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr6_52730905_54027603.twas.tsv.gz   | -0.0698993 |  53557156 | chr6  |  53616970 |  53497341 | GCLC     |\n",
       "\n"
      ],
      "text/plain": [
       "  molecular_id    context          gwas_study         n_methods n_sig\n",
       "1 ENSG00000000419 Exc_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "2 ENSG00000000457 Ast_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "3 ENSG00000000460 Ast_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "4 ENSG00000000971 PCC_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "5 ENSG00000001036 Ast_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "6 ENSG00000001084 Ast_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "  best_pval   best_method_pval best_method_selected pass  final_pval  ⋯\n",
       "1 0.004887676 0.634741658      FALSE                FALSE 0.634741658 ⋯\n",
       "2 0.314317590 0.666951750      FALSE                FALSE 0.666951750 ⋯\n",
       "3 0.550799973 0.681781965      FALSE                FALSE 0.681781965 ⋯\n",
       "4 0.006727627 0.006727627      FALSE                FALSE 0.006727627 ⋯\n",
       "5 0.125012981 0.125012981      FALSE                FALSE 0.125012981 ⋯\n",
       "6 0.166978326 0.851335412      FALSE                FALSE 0.851335412 ⋯\n",
       "  method_selected_original region                   study_context \n",
       "1 TRUE                     chr20_50319421_53436693  ROSMAP_eQTL_wp\n",
       "2 TRUE                     chr1_168438717_170228106 ROSMAP_eQTL_wp\n",
       "3 TRUE                     chr1_168438717_170228106 ROSMAP_eQTL_wp\n",
       "4 TRUE                     chr1_195599253_199271134 ROSMAP_eQTL_wp\n",
       "5 TRUE                     chr6_143399926_145632539 ROSMAP_eQTL_wp\n",
       "6 TRUE                     chr6_52730905_54027603   ROSMAP_eQTL_wp\n",
       "  source_file                                                    logp      \n",
       "1 ROSMAP_eQTL_wp_unadjusted.chr20_50319421_53436693.twas.tsv.gz  -0.1974030\n",
       "2 ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz  0.1759056\n",
       "3 ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz -0.1663545\n",
       "4 ROSMAP_eQTL_wp_unadjusted.chr1_195599253_199271134.twas.tsv.gz  2.1721381\n",
       "5 ROSMAP_eQTL_wp_unadjusted.chr6_143399926_145632539.twas.tsv.gz -0.9030449\n",
       "6 ROSMAP_eQTL_wp_unadjusted.chr6_52730905_54027603.twas.tsv.gz   -0.0698993\n",
       "  pos       #chr  TSS.y     TES       gene_name\n",
       "1  51747434 chr20  50958555  50934867 DPM1     \n",
       "2 169540000 chr1  169894267 169849631 SCYL3    \n",
       "3 169540000 chr1  169662007 169854080 C1orf112 \n",
       "4 195093752 chr1  196652043 196747504 CFH      \n",
       "5 143115860 chr6  143511720 143494812 FUCA2    \n",
       "6  53557156 chr6   53616970  53497341 GCLC     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "240a37d2-b50f-42d9-a8ba-177d0ea16f00",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "chr_info <- df %>%\n",
    "  group_by(chr) %>%\n",
    "  summarise(chr_len = max(pos)) %>%\n",
    "  mutate(chr_start = lag(cumsum(chr_len), default = 0))\n",
    "\n",
    "df <- df %>%\n",
    "  left_join(chr_info, by = \"chr\") %>%\n",
    "  mutate(cum_pos = pos + chr_start)\n",
    "  \n",
    "axis_df <- chr_info %>%\n",
    "  mutate(center = chr_start + chr_len/2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "f884ab9d-068c-4ee3-bcf9-67182c3e2b0a",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "df <- df %>%\n",
    "  left_join(\n",
    "    mr_sig_unadj %>% rename(molecular_id = gene_name),\n",
    "    by = c(\"molecular_id\", \"context\")\n",
    "  )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "7a99be0a-499f-4ee3-adad-d0a564927c3e",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 56 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>molecular_id</th><th scope=col>context</th><th scope=col>gene_name</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000004534</td><td>Ast_DeJager_eQTL</td><td>RBM6          </td></tr>\n",
       "\t<tr><td>ENSG00000014123</td><td>Ast_DeJager_eQTL</td><td>UFL1          </td></tr>\n",
       "\t<tr><td>ENSG00000023516</td><td>Exc_DeJager_eQTL</td><td>AKAP11        </td></tr>\n",
       "\t<tr><td>ENSG00000030066</td><td>Exc_DeJager_eQTL</td><td>NUP160        </td></tr>\n",
       "\t<tr><td>ENSG00000073711</td><td>Ast_DeJager_eQTL</td><td>PPP2R3A       </td></tr>\n",
       "\t<tr><td>ENSG00000075213</td><td>Ast_DeJager_eQTL</td><td>SEMA3A        </td></tr>\n",
       "\t<tr><td>ENSG00000075413</td><td>Ast_DeJager_eQTL</td><td>MARK3         </td></tr>\n",
       "\t<tr><td>ENSG00000078487</td><td>Ast_DeJager_eQTL</td><td>ZCWPW1        </td></tr>\n",
       "\t<tr><td>ENSG00000081377</td><td>Ast_DeJager_eQTL</td><td>CDC14B        </td></tr>\n",
       "\t<tr><td>ENSG00000090621</td><td>PCC_DeJager_eQTL</td><td>PABPC4        </td></tr>\n",
       "\t<tr><td>ENSG00000099331</td><td>Exc_DeJager_eQTL</td><td>MYO9B         </td></tr>\n",
       "\t<tr><td>ENSG00000101000</td><td>Ast_DeJager_eQTL</td><td>PROCR         </td></tr>\n",
       "\t<tr><td>ENSG00000101639</td><td>Ast_DeJager_eQTL</td><td>CEP192        </td></tr>\n",
       "\t<tr><td>ENSG00000111863</td><td>Exc_DeJager_eQTL</td><td>ADTRP         </td></tr>\n",
       "\t<tr><td>ENSG00000116754</td><td>Exc_DeJager_eQTL</td><td>SRSF11        </td></tr>\n",
       "\t<tr><td>ENSG00000119878</td><td>Ast_DeJager_eQTL</td><td>CRIPT         </td></tr>\n",
       "\t<tr><td>ENSG00000121716</td><td>Ast_DeJager_eQTL</td><td>PILRB         </td></tr>\n",
       "\t<tr><td>ENSG00000125965</td><td>PCC_DeJager_eQTL</td><td>GDF5          </td></tr>\n",
       "\t<tr><td>ENSG00000135473</td><td>Exc_DeJager_eQTL</td><td>PAN2          </td></tr>\n",
       "\t<tr><td>ENSG00000138193</td><td>Ast_DeJager_eQTL</td><td>PLCE1         </td></tr>\n",
       "\t<tr><td>ENSG00000139899</td><td>PCC_DeJager_eQTL</td><td>CBLN3         </td></tr>\n",
       "\t<tr><td>ENSG00000155111</td><td>Exc_DeJager_eQTL</td><td>CDK19         </td></tr>\n",
       "\t<tr><td>ENSG00000157322</td><td>PCC_DeJager_eQTL</td><td>CLEC18A       </td></tr>\n",
       "\t<tr><td>ENSG00000158092</td><td>Ast_DeJager_eQTL</td><td>NCK1          </td></tr>\n",
       "\t<tr><td>ENSG00000158373</td><td>PCC_DeJager_eQTL</td><td>H2BC5         </td></tr>\n",
       "\t<tr><td>ENSG00000162631</td><td>Ast_DeJager_eQTL</td><td>NTNG1         </td></tr>\n",
       "\t<tr><td>ENSG00000163935</td><td>Ast_DeJager_eQTL</td><td>SFMBT1        </td></tr>\n",
       "\t<tr><td>ENSG00000164078</td><td>PCC_DeJager_eQTL</td><td>MST1R         </td></tr>\n",
       "\t<tr><td>ENSG00000165476</td><td>Oli_DeJager_eQTL</td><td>REEP3         </td></tr>\n",
       "\t<tr><td>ENSG00000165915</td><td>PCC_DeJager_eQTL</td><td>SLC39A13      </td></tr>\n",
       "\t<tr><td>ENSG00000167394</td><td>Exc_DeJager_eQTL</td><td>ZNF668        </td></tr>\n",
       "\t<tr><td>ENSG00000167395</td><td>PCC_DeJager_eQTL</td><td>ZNF646        </td></tr>\n",
       "\t<tr><td>ENSG00000168917</td><td>Ast_DeJager_eQTL</td><td>SLC35G2       </td></tr>\n",
       "\t<tr><td>ENSG00000171206</td><td>Ast_DeJager_eQTL</td><td>TRIM8         </td></tr>\n",
       "\t<tr><td>ENSG00000172247</td><td>Exc_DeJager_eQTL</td><td>C1QTNF4       </td></tr>\n",
       "\t<tr><td>ENSG00000176020</td><td>PCC_DeJager_eQTL</td><td>AMIGO3        </td></tr>\n",
       "\t<tr><td>ENSG00000176046</td><td>PCC_DeJager_eQTL</td><td>NUPR1         </td></tr>\n",
       "\t<tr><td>ENSG00000176383</td><td>PCC_DeJager_eQTL</td><td>B3GNT4        </td></tr>\n",
       "\t<tr><td>ENSG00000178700</td><td>Ast_DeJager_eQTL</td><td>DHFR2         </td></tr>\n",
       "\t<tr><td>ENSG00000181722</td><td>Ast_DeJager_eQTL</td><td>ZBTB20        </td></tr>\n",
       "\t<tr><td>ENSG00000184110</td><td>Exc_DeJager_eQTL</td><td>EIF3C         </td></tr>\n",
       "\t<tr><td>ENSG00000188338</td><td>Ast_DeJager_eQTL</td><td>SLC38A3       </td></tr>\n",
       "\t<tr><td>ENSG00000196993</td><td>PCC_DeJager_eQTL</td><td>NPIPB9        </td></tr>\n",
       "\t<tr><td>ENSG00000197181</td><td>PCC_DeJager_eQTL</td><td>PIWIL2        </td></tr>\n",
       "\t<tr><td>ENSG00000204642</td><td>Ast_DeJager_eQTL</td><td>HLA-F         </td></tr>\n",
       "\t<tr><td>ENSG00000213658</td><td>Exc_DeJager_eQTL</td><td>LAT           </td></tr>\n",
       "\t<tr><td>ENSG00000231852</td><td>PCC_DeJager_eQTL</td><td>CYP21A2       </td></tr>\n",
       "\t<tr><td>ENSG00000237441</td><td>Exc_DeJager_eQTL</td><td>RGL2          </td></tr>\n",
       "\t<tr><td>ENSG00000243477</td><td>PCC_DeJager_eQTL</td><td>NAA80         </td></tr>\n",
       "\t<tr><td>ENSG00000243696</td><td>PCC_DeJager_eQTL</td><td>AC006254.1    </td></tr>\n",
       "\t<tr><td>ENSG00000248592</td><td>PCC_DeJager_eQTL</td><td>STIMATE-MUSTN1</td></tr>\n",
       "\t<tr><td>ENSG00000254858</td><td>PCC_DeJager_eQTL</td><td>MPV17L2       </td></tr>\n",
       "\t<tr><td>ENSG00000255439</td><td>PCC_DeJager_eQTL</td><td>AC135050.2    </td></tr>\n",
       "\t<tr><td>ENSG00000258653</td><td>PCC_DeJager_eQTL</td><td>AC005520.1    </td></tr>\n",
       "\t<tr><td>ENSG00000272305</td><td>PCC_DeJager_eQTL</td><td>AC096887.1    </td></tr>\n",
       "\t<tr><td>ENSG00000284917</td><td>PCC_DeJager_eQTL</td><td>EEF1AKMT4-ECE2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 56 × 3\n",
       "\\begin{tabular}{lll}\n",
       " molecular\\_id & context & gene\\_name\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000004534 & Ast\\_DeJager\\_eQTL & RBM6          \\\\\n",
       "\t ENSG00000014123 & Ast\\_DeJager\\_eQTL & UFL1          \\\\\n",
       "\t ENSG00000023516 & Exc\\_DeJager\\_eQTL & AKAP11        \\\\\n",
       "\t ENSG00000030066 & Exc\\_DeJager\\_eQTL & NUP160        \\\\\n",
       "\t ENSG00000073711 & Ast\\_DeJager\\_eQTL & PPP2R3A       \\\\\n",
       "\t ENSG00000075213 & Ast\\_DeJager\\_eQTL & SEMA3A        \\\\\n",
       "\t ENSG00000075413 & Ast\\_DeJager\\_eQTL & MARK3         \\\\\n",
       "\t ENSG00000078487 & Ast\\_DeJager\\_eQTL & ZCWPW1        \\\\\n",
       "\t ENSG00000081377 & Ast\\_DeJager\\_eQTL & CDC14B        \\\\\n",
       "\t ENSG00000090621 & PCC\\_DeJager\\_eQTL & PABPC4        \\\\\n",
       "\t ENSG00000099331 & Exc\\_DeJager\\_eQTL & MYO9B         \\\\\n",
       "\t ENSG00000101000 & Ast\\_DeJager\\_eQTL & PROCR         \\\\\n",
       "\t ENSG00000101639 & Ast\\_DeJager\\_eQTL & CEP192        \\\\\n",
       "\t ENSG00000111863 & Exc\\_DeJager\\_eQTL & ADTRP         \\\\\n",
       "\t ENSG00000116754 & Exc\\_DeJager\\_eQTL & SRSF11        \\\\\n",
       "\t ENSG00000119878 & Ast\\_DeJager\\_eQTL & CRIPT         \\\\\n",
       "\t ENSG00000121716 & Ast\\_DeJager\\_eQTL & PILRB         \\\\\n",
       "\t ENSG00000125965 & PCC\\_DeJager\\_eQTL & GDF5          \\\\\n",
       "\t ENSG00000135473 & Exc\\_DeJager\\_eQTL & PAN2          \\\\\n",
       "\t ENSG00000138193 & Ast\\_DeJager\\_eQTL & PLCE1         \\\\\n",
       "\t ENSG00000139899 & PCC\\_DeJager\\_eQTL & CBLN3         \\\\\n",
       "\t ENSG00000155111 & Exc\\_DeJager\\_eQTL & CDK19         \\\\\n",
       "\t ENSG00000157322 & PCC\\_DeJager\\_eQTL & CLEC18A       \\\\\n",
       "\t ENSG00000158092 & Ast\\_DeJager\\_eQTL & NCK1          \\\\\n",
       "\t ENSG00000158373 & PCC\\_DeJager\\_eQTL & H2BC5         \\\\\n",
       "\t ENSG00000162631 & Ast\\_DeJager\\_eQTL & NTNG1         \\\\\n",
       "\t ENSG00000163935 & Ast\\_DeJager\\_eQTL & SFMBT1        \\\\\n",
       "\t ENSG00000164078 & PCC\\_DeJager\\_eQTL & MST1R         \\\\\n",
       "\t ENSG00000165476 & Oli\\_DeJager\\_eQTL & REEP3         \\\\\n",
       "\t ENSG00000165915 & PCC\\_DeJager\\_eQTL & SLC39A13      \\\\\n",
       "\t ENSG00000167394 & Exc\\_DeJager\\_eQTL & ZNF668        \\\\\n",
       "\t ENSG00000167395 & PCC\\_DeJager\\_eQTL & ZNF646        \\\\\n",
       "\t ENSG00000168917 & Ast\\_DeJager\\_eQTL & SLC35G2       \\\\\n",
       "\t ENSG00000171206 & Ast\\_DeJager\\_eQTL & TRIM8         \\\\\n",
       "\t ENSG00000172247 & Exc\\_DeJager\\_eQTL & C1QTNF4       \\\\\n",
       "\t ENSG00000176020 & PCC\\_DeJager\\_eQTL & AMIGO3        \\\\\n",
       "\t ENSG00000176046 & PCC\\_DeJager\\_eQTL & NUPR1         \\\\\n",
       "\t ENSG00000176383 & PCC\\_DeJager\\_eQTL & B3GNT4        \\\\\n",
       "\t ENSG00000178700 & Ast\\_DeJager\\_eQTL & DHFR2         \\\\\n",
       "\t ENSG00000181722 & Ast\\_DeJager\\_eQTL & ZBTB20        \\\\\n",
       "\t ENSG00000184110 & Exc\\_DeJager\\_eQTL & EIF3C         \\\\\n",
       "\t ENSG00000188338 & Ast\\_DeJager\\_eQTL & SLC38A3       \\\\\n",
       "\t ENSG00000196993 & PCC\\_DeJager\\_eQTL & NPIPB9        \\\\\n",
       "\t ENSG00000197181 & PCC\\_DeJager\\_eQTL & PIWIL2        \\\\\n",
       "\t ENSG00000204642 & Ast\\_DeJager\\_eQTL & HLA-F         \\\\\n",
       "\t ENSG00000213658 & Exc\\_DeJager\\_eQTL & LAT           \\\\\n",
       "\t ENSG00000231852 & PCC\\_DeJager\\_eQTL & CYP21A2       \\\\\n",
       "\t ENSG00000237441 & Exc\\_DeJager\\_eQTL & RGL2          \\\\\n",
       "\t ENSG00000243477 & PCC\\_DeJager\\_eQTL & NAA80         \\\\\n",
       "\t ENSG00000243696 & PCC\\_DeJager\\_eQTL & AC006254.1    \\\\\n",
       "\t ENSG00000248592 & PCC\\_DeJager\\_eQTL & STIMATE-MUSTN1\\\\\n",
       "\t ENSG00000254858 & PCC\\_DeJager\\_eQTL & MPV17L2       \\\\\n",
       "\t ENSG00000255439 & PCC\\_DeJager\\_eQTL & AC135050.2    \\\\\n",
       "\t ENSG00000258653 & PCC\\_DeJager\\_eQTL & AC005520.1    \\\\\n",
       "\t ENSG00000272305 & PCC\\_DeJager\\_eQTL & AC096887.1    \\\\\n",
       "\t ENSG00000284917 & PCC\\_DeJager\\_eQTL & EEF1AKMT4-ECE2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 56 × 3\n",
       "\n",
       "| molecular_id &lt;chr&gt; | context &lt;chr&gt; | gene_name &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000004534 | Ast_DeJager_eQTL | RBM6           |\n",
       "| ENSG00000014123 | Ast_DeJager_eQTL | UFL1           |\n",
       "| ENSG00000023516 | Exc_DeJager_eQTL | AKAP11         |\n",
       "| ENSG00000030066 | Exc_DeJager_eQTL | NUP160         |\n",
       "| ENSG00000073711 | Ast_DeJager_eQTL | PPP2R3A        |\n",
       "| ENSG00000075213 | Ast_DeJager_eQTL | SEMA3A         |\n",
       "| ENSG00000075413 | Ast_DeJager_eQTL | MARK3          |\n",
       "| ENSG00000078487 | Ast_DeJager_eQTL | ZCWPW1         |\n",
       "| ENSG00000081377 | Ast_DeJager_eQTL | CDC14B         |\n",
       "| ENSG00000090621 | PCC_DeJager_eQTL | PABPC4         |\n",
       "| ENSG00000099331 | Exc_DeJager_eQTL | MYO9B          |\n",
       "| ENSG00000101000 | Ast_DeJager_eQTL | PROCR          |\n",
       "| ENSG00000101639 | Ast_DeJager_eQTL | CEP192         |\n",
       "| ENSG00000111863 | Exc_DeJager_eQTL | ADTRP          |\n",
       "| ENSG00000116754 | Exc_DeJager_eQTL | SRSF11         |\n",
       "| ENSG00000119878 | Ast_DeJager_eQTL | CRIPT          |\n",
       "| ENSG00000121716 | Ast_DeJager_eQTL | PILRB          |\n",
       "| ENSG00000125965 | PCC_DeJager_eQTL | GDF5           |\n",
       "| ENSG00000135473 | Exc_DeJager_eQTL | PAN2           |\n",
       "| ENSG00000138193 | Ast_DeJager_eQTL | PLCE1          |\n",
       "| ENSG00000139899 | PCC_DeJager_eQTL | CBLN3          |\n",
       "| ENSG00000155111 | Exc_DeJager_eQTL | CDK19          |\n",
       "| ENSG00000157322 | PCC_DeJager_eQTL | CLEC18A        |\n",
       "| ENSG00000158092 | Ast_DeJager_eQTL | NCK1           |\n",
       "| ENSG00000158373 | PCC_DeJager_eQTL | H2BC5          |\n",
       "| ENSG00000162631 | Ast_DeJager_eQTL | NTNG1          |\n",
       "| ENSG00000163935 | Ast_DeJager_eQTL | SFMBT1         |\n",
       "| ENSG00000164078 | PCC_DeJager_eQTL | MST1R          |\n",
       "| ENSG00000165476 | Oli_DeJager_eQTL | REEP3          |\n",
       "| ENSG00000165915 | PCC_DeJager_eQTL | SLC39A13       |\n",
       "| ENSG00000167394 | Exc_DeJager_eQTL | ZNF668         |\n",
       "| ENSG00000167395 | PCC_DeJager_eQTL | ZNF646         |\n",
       "| ENSG00000168917 | Ast_DeJager_eQTL | SLC35G2        |\n",
       "| ENSG00000171206 | Ast_DeJager_eQTL | TRIM8          |\n",
       "| ENSG00000172247 | Exc_DeJager_eQTL | C1QTNF4        |\n",
       "| ENSG00000176020 | PCC_DeJager_eQTL | AMIGO3         |\n",
       "| ENSG00000176046 | PCC_DeJager_eQTL | NUPR1          |\n",
       "| ENSG00000176383 | PCC_DeJager_eQTL | B3GNT4         |\n",
       "| ENSG00000178700 | Ast_DeJager_eQTL | DHFR2          |\n",
       "| ENSG00000181722 | Ast_DeJager_eQTL | ZBTB20         |\n",
       "| ENSG00000184110 | Exc_DeJager_eQTL | EIF3C          |\n",
       "| ENSG00000188338 | Ast_DeJager_eQTL | SLC38A3        |\n",
       "| ENSG00000196993 | PCC_DeJager_eQTL | NPIPB9         |\n",
       "| ENSG00000197181 | PCC_DeJager_eQTL | PIWIL2         |\n",
       "| ENSG00000204642 | Ast_DeJager_eQTL | HLA-F          |\n",
       "| ENSG00000213658 | Exc_DeJager_eQTL | LAT            |\n",
       "| ENSG00000231852 | PCC_DeJager_eQTL | CYP21A2        |\n",
       "| ENSG00000237441 | Exc_DeJager_eQTL | RGL2           |\n",
       "| ENSG00000243477 | PCC_DeJager_eQTL | NAA80          |\n",
       "| ENSG00000243696 | PCC_DeJager_eQTL | AC006254.1     |\n",
       "| ENSG00000248592 | PCC_DeJager_eQTL | STIMATE-MUSTN1 |\n",
       "| ENSG00000254858 | PCC_DeJager_eQTL | MPV17L2        |\n",
       "| ENSG00000255439 | PCC_DeJager_eQTL | AC135050.2     |\n",
       "| ENSG00000258653 | PCC_DeJager_eQTL | AC005520.1     |\n",
       "| ENSG00000272305 | PCC_DeJager_eQTL | AC096887.1     |\n",
       "| ENSG00000284917 | PCC_DeJager_eQTL | EEF1AKMT4-ECE2 |\n",
       "\n"
      ],
      "text/plain": [
       "   molecular_id    context          gene_name     \n",
       "1  ENSG00000004534 Ast_DeJager_eQTL RBM6          \n",
       "2  ENSG00000014123 Ast_DeJager_eQTL UFL1          \n",
       "3  ENSG00000023516 Exc_DeJager_eQTL AKAP11        \n",
       "4  ENSG00000030066 Exc_DeJager_eQTL NUP160        \n",
       "5  ENSG00000073711 Ast_DeJager_eQTL PPP2R3A       \n",
       "6  ENSG00000075213 Ast_DeJager_eQTL SEMA3A        \n",
       "7  ENSG00000075413 Ast_DeJager_eQTL MARK3         \n",
       "8  ENSG00000078487 Ast_DeJager_eQTL ZCWPW1        \n",
       "9  ENSG00000081377 Ast_DeJager_eQTL CDC14B        \n",
       "10 ENSG00000090621 PCC_DeJager_eQTL PABPC4        \n",
       "11 ENSG00000099331 Exc_DeJager_eQTL MYO9B         \n",
       "12 ENSG00000101000 Ast_DeJager_eQTL PROCR         \n",
       "13 ENSG00000101639 Ast_DeJager_eQTL CEP192        \n",
       "14 ENSG00000111863 Exc_DeJager_eQTL ADTRP         \n",
       "15 ENSG00000116754 Exc_DeJager_eQTL SRSF11        \n",
       "16 ENSG00000119878 Ast_DeJager_eQTL CRIPT         \n",
       "17 ENSG00000121716 Ast_DeJager_eQTL PILRB         \n",
       "18 ENSG00000125965 PCC_DeJager_eQTL GDF5          \n",
       "19 ENSG00000135473 Exc_DeJager_eQTL PAN2          \n",
       "20 ENSG00000138193 Ast_DeJager_eQTL PLCE1         \n",
       "21 ENSG00000139899 PCC_DeJager_eQTL CBLN3         \n",
       "22 ENSG00000155111 Exc_DeJager_eQTL CDK19         \n",
       "23 ENSG00000157322 PCC_DeJager_eQTL CLEC18A       \n",
       "24 ENSG00000158092 Ast_DeJager_eQTL NCK1          \n",
       "25 ENSG00000158373 PCC_DeJager_eQTL H2BC5         \n",
       "26 ENSG00000162631 Ast_DeJager_eQTL NTNG1         \n",
       "27 ENSG00000163935 Ast_DeJager_eQTL SFMBT1        \n",
       "28 ENSG00000164078 PCC_DeJager_eQTL MST1R         \n",
       "29 ENSG00000165476 Oli_DeJager_eQTL REEP3         \n",
       "30 ENSG00000165915 PCC_DeJager_eQTL SLC39A13      \n",
       "31 ENSG00000167394 Exc_DeJager_eQTL ZNF668        \n",
       "32 ENSG00000167395 PCC_DeJager_eQTL ZNF646        \n",
       "33 ENSG00000168917 Ast_DeJager_eQTL SLC35G2       \n",
       "34 ENSG00000171206 Ast_DeJager_eQTL TRIM8         \n",
       "35 ENSG00000172247 Exc_DeJager_eQTL C1QTNF4       \n",
       "36 ENSG00000176020 PCC_DeJager_eQTL AMIGO3        \n",
       "37 ENSG00000176046 PCC_DeJager_eQTL NUPR1         \n",
       "38 ENSG00000176383 PCC_DeJager_eQTL B3GNT4        \n",
       "39 ENSG00000178700 Ast_DeJager_eQTL DHFR2         \n",
       "40 ENSG00000181722 Ast_DeJager_eQTL ZBTB20        \n",
       "41 ENSG00000184110 Exc_DeJager_eQTL EIF3C         \n",
       "42 ENSG00000188338 Ast_DeJager_eQTL SLC38A3       \n",
       "43 ENSG00000196993 PCC_DeJager_eQTL NPIPB9        \n",
       "44 ENSG00000197181 PCC_DeJager_eQTL PIWIL2        \n",
       "45 ENSG00000204642 Ast_DeJager_eQTL HLA-F         \n",
       "46 ENSG00000213658 Exc_DeJager_eQTL LAT           \n",
       "47 ENSG00000231852 PCC_DeJager_eQTL CYP21A2       \n",
       "48 ENSG00000237441 Exc_DeJager_eQTL RGL2          \n",
       "49 ENSG00000243477 PCC_DeJager_eQTL NAA80         \n",
       "50 ENSG00000243696 PCC_DeJager_eQTL AC006254.1    \n",
       "51 ENSG00000248592 PCC_DeJager_eQTL STIMATE-MUSTN1\n",
       "52 ENSG00000254858 PCC_DeJager_eQTL MPV17L2       \n",
       "53 ENSG00000255439 PCC_DeJager_eQTL AC135050.2    \n",
       "54 ENSG00000258653 PCC_DeJager_eQTL AC005520.1    \n",
       "55 ENSG00000272305 PCC_DeJager_eQTL AC096887.1    \n",
       "56 ENSG00000284917 PCC_DeJager_eQTL EEF1AKMT4-ECE2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df %>% filter(!is.na(Q_pval)) %>% select(molecular_id, context, gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "6b60d619",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 49</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>molecular_id</th><th scope=col>context</th><th scope=col>gwas_study.x</th><th scope=col>n_methods</th><th scope=col>n_sig</th><th scope=col>best_pval</th><th scope=col>best_method_pval</th><th scope=col>best_method_selected</th><th scope=col>pass</th><th scope=col>final_pval</th><th scope=col>⋯</th><th scope=col>cpip</th><th scope=col>gwas_study.y</th><th scope=col>meta_eff</th><th scope=col>meta_pval</th><th scope=col>num_CS</th><th scope=col>num_IV</th><th scope=col>se_meta_eff</th><th scope=col>region.y</th><th scope=col>study_context.y</th><th scope=col>source_file.y</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ENSG00000000419</td><td>Exc_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.004887676</td><td>0.634741658</td><td>FALSE</td><td>FALSE</td><td>0.634741658</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ENSG00000000457</td><td>Ast_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.314317590</td><td>0.666951750</td><td>FALSE</td><td>FALSE</td><td>0.666951750</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>ENSG00000000460</td><td>Ast_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.550799973</td><td>0.681781965</td><td>FALSE</td><td>FALSE</td><td>0.681781965</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>ENSG00000000971</td><td>PCC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.006727627</td><td>0.006727627</td><td>FALSE</td><td>FALSE</td><td>0.006727627</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ENSG00000001036</td><td>Ast_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.125012981</td><td>0.125012981</td><td>FALSE</td><td>FALSE</td><td>0.125012981</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ENSG00000001084</td><td>Ast_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.166978326</td><td>0.851335412</td><td>FALSE</td><td>FALSE</td><td>0.851335412</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 49\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & molecular\\_id & context & gwas\\_study.x & n\\_methods & n\\_sig & best\\_pval & best\\_method\\_pval & best\\_method\\_selected & pass & final\\_pval & ⋯ & cpip & gwas\\_study.y & meta\\_eff & meta\\_pval & num\\_CS & num\\_IV & se\\_meta\\_eff & region.y & study\\_context.y & source\\_file.y\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <int> & <dbl> & <dbl> & <lgl> & <lgl> & <dbl> & ⋯ & <dbl> & <chr> & <dbl> & <dbl> & <int> & <int> & <dbl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ENSG00000000419 & Exc\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.004887676 & 0.634741658 & FALSE & FALSE & 0.634741658 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t2 & ENSG00000000457 & Ast\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.314317590 & 0.666951750 & FALSE & FALSE & 0.666951750 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t3 & ENSG00000000460 & Ast\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.550799973 & 0.681781965 & FALSE & FALSE & 0.681781965 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t4 & ENSG00000000971 & PCC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.006727627 & 0.006727627 & FALSE & FALSE & 0.006727627 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t5 & ENSG00000001036 & Ast\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.125012981 & 0.125012981 & FALSE & FALSE & 0.125012981 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t6 & ENSG00000001084 & Ast\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.166978326 & 0.851335412 & FALSE & FALSE & 0.851335412 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 49\n",
       "\n",
       "| <!--/--> | molecular_id &lt;chr&gt; | context &lt;chr&gt; | gwas_study.x &lt;chr&gt; | n_methods &lt;int&gt; | n_sig &lt;int&gt; | best_pval &lt;dbl&gt; | best_method_pval &lt;dbl&gt; | best_method_selected &lt;lgl&gt; | pass &lt;lgl&gt; | final_pval &lt;dbl&gt; | ⋯ ⋯ | cpip &lt;dbl&gt; | gwas_study.y &lt;chr&gt; | meta_eff &lt;dbl&gt; | meta_pval &lt;dbl&gt; | num_CS &lt;int&gt; | num_IV &lt;int&gt; | se_meta_eff &lt;dbl&gt; | region.y &lt;chr&gt; | study_context.y &lt;chr&gt; | source_file.y &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ENSG00000000419 | Exc_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.004887676 | 0.634741658 | FALSE | FALSE | 0.634741658 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 2 | ENSG00000000457 | Ast_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.314317590 | 0.666951750 | FALSE | FALSE | 0.666951750 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 3 | ENSG00000000460 | Ast_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.550799973 | 0.681781965 | FALSE | FALSE | 0.681781965 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 4 | ENSG00000000971 | PCC_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.006727627 | 0.006727627 | FALSE | FALSE | 0.006727627 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 5 | ENSG00000001036 | Ast_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.125012981 | 0.125012981 | FALSE | FALSE | 0.125012981 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 6 | ENSG00000001084 | Ast_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.166978326 | 0.851335412 | FALSE | FALSE | 0.851335412 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  molecular_id    context          gwas_study.x       n_methods n_sig\n",
       "1 ENSG00000000419 Exc_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "2 ENSG00000000457 Ast_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "3 ENSG00000000460 Ast_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "4 ENSG00000000971 PCC_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "5 ENSG00000001036 Ast_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "6 ENSG00000001084 Ast_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "  best_pval   best_method_pval best_method_selected pass  final_pval  ⋯ cpip\n",
       "1 0.004887676 0.634741658      FALSE                FALSE 0.634741658 ⋯ NA  \n",
       "2 0.314317590 0.666951750      FALSE                FALSE 0.666951750 ⋯ NA  \n",
       "3 0.550799973 0.681781965      FALSE                FALSE 0.681781965 ⋯ NA  \n",
       "4 0.006727627 0.006727627      FALSE                FALSE 0.006727627 ⋯ NA  \n",
       "5 0.125012981 0.125012981      FALSE                FALSE 0.125012981 ⋯ NA  \n",
       "6 0.166978326 0.851335412      FALSE                FALSE 0.851335412 ⋯ NA  \n",
       "  gwas_study.y meta_eff meta_pval num_CS num_IV se_meta_eff region.y\n",
       "1 NA           NA       NA        NA     NA     NA          NA      \n",
       "2 NA           NA       NA        NA     NA     NA          NA      \n",
       "3 NA           NA       NA        NA     NA     NA          NA      \n",
       "4 NA           NA       NA        NA     NA     NA          NA      \n",
       "5 NA           NA       NA        NA     NA     NA          NA      \n",
       "6 NA           NA       NA        NA     NA     NA          NA      \n",
       "  study_context.y source_file.y\n",
       "1 NA              NA           \n",
       "2 NA              NA           \n",
       "3 NA              NA           \n",
       "4 NA              NA           \n",
       "5 NA              NA           \n",
       "6 NA              NA           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>14107</li><li>49</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 14107\n",
       "\\item 49\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 14107\n",
       "2. 49\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 14107    49"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)\n",
    "dim(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "d5e7f215-7297-4242-97c3-329cb21953f7",
   "metadata": {
    "kernel": "R",
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "\n",
    "# Add a column for MR significance (example: FDR < 0.05)\n",
    "df <- df %>%\n",
    "  mutate(category = case_when(\n",
    "    !is.na(Q_pval) & pass ~ \"TWAS & MR\",        # Both significant\n",
    "    is.na(Q_pval) & pass ~ \"TWAS only\",         # Only TWAS significant\n",
    "    !is.na(Q_pval) & !pass ~ \"MR only\",         # Only MR significant\n",
    "    is.na(Q_pval) & !pass ~ \"Not Significant\",  # Neither significant\n",
    "    TRUE ~ \"Others\"                              # Catch-all for remaining cases\n",
    "  ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "d1018956",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>category</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>MR only        </td><td>   32</td></tr>\n",
       "\t<tr><td>Not Significant</td><td>13997</td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>TWAS &amp; MR      </span></td><td><span style=white-space:pre-wrap>   24</span></td></tr>\n",
       "\t<tr><td>TWAS only      </td><td>   54</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4 × 2\n",
       "\\begin{tabular}{ll}\n",
       " category & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t MR only         &    32\\\\\n",
       "\t Not Significant & 13997\\\\\n",
       "\t TWAS \\& MR       &    24\\\\\n",
       "\t TWAS only       &    54\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4 × 2\n",
       "\n",
       "| category &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| MR only         |    32 |\n",
       "| Not Significant | 13997 |\n",
       "| TWAS &amp; MR       |    24 |\n",
       "| TWAS only       |    54 |\n",
       "\n"
      ],
      "text/plain": [
       "  category        n    \n",
       "1 MR only            32\n",
       "2 Not Significant 13997\n",
       "3 TWAS & MR          24\n",
       "4 TWAS only          54"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df |> count(category)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "46571a3d-98d3-4def-953b-ea9879608e28",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# dir.create('data')\n",
    "saveRDS(list(df = df, unadj_cutoff = unadj_cutoff, axis_df= axis_df), 'data/TWAS_manhattan_plot_data.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0456b84a-5e08-423a-be45-5b64d32e6861",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.3"
  },
  "sos": {
   "kernels": [
    [
     "R",
     "ir",
     "R",
     "",
     ""
    ]
   ],
   "version": ""
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
