{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "7d50c71c-d2a2-4677-a763-88c6bdd3123a",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "# TWAS\n",
    "a) Manhattan plot of glyco-TWAS results (unadjusted data); annotate points to indicate MR significance\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8bbe5663-26c0-4106-ad47-8d95bc26233c",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "## Combine Data\n",
    "TWAS and MR results generaated by pipelines were combined using the following commands\n",
    "```\n",
    "python /mnt/lustre/home/rl3328/rl3328/motor_qtl/combine_twas.py /mnt/lustre/lab/ctcn/hklein/motor_qtl_project/unadjusted_wp_2020/susie_twas/noQC_nonimputed_result/twas --type both --output ROSMAP_eQTL_wp_unadjusted\n",
    "```\n",
    "### Input\n",
    "Input files are all on the new AWS-based HPC\n",
    "* Combined TWAS file:`/mnt/lustre/lab/ctcn/hklein/motor_qtl_project/wp_2020/susie_twas/noQC_nonimputed_result/twas/ROSMAP_eQTL_wp_unadjusted.combined_twas.tsv.gz`\n",
    "* Combined MR file:`/mnt/lustre/lab/ctcn/hklein/motor_qtl_project/unadjusted_wp_2020/susie_twas/noQC_nonimputed_result/twas/ROSMAP_eQTL_wp_unadjusted.combined_mr.tsv.gz`\n",
    "* reference file for gene annotation: `/mnt/lustre/home/rl3328/rl3328/resource/Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.region_list.2025`\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01773a49",
   "metadata": {},
   "source": [
    "### Landscape\n",
    "\n",
    "* The TWAS results show substantial variation in the number of available genes across contexts. Monocyte, DLPFC, and AC have notably fewer genes available for analysis. In particular, the DLPFC bulk eQTL dataset includes only 57 genes, which likely explains why no genes reach significance under the most accurate predictive model for DLPFC bulk eQTL.\n",
    "* Consistently, the MR analysis also yields no significant genes for monocyte, DLPFC, or AC.\n",
    "\n",
    "* **Given these limitations, it may be more appropriate to combine bulk and single-nucleus (sn) data for DLPFC in downstream analyses.**\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09fa2c4a",
   "metadata": {},
   "source": [
    "#### TWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d55a570c-74e3-4642-86c7-64fa97d3cf82",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "library(data.table)\n",
    "library(tidyverse)\n",
    "twas <- fread('/mnt/lustre/lab/ctcn/hklein/motor_qtl_project/unadjusted_wp_2020/susie_twas/noQC_nonimputed_result/twas/ROSMAP_eQTL_wp_unadjusted.combined_twas.tsv.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5ce0b8c3",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>molecular_id</th><th scope=col>TSS</th><th scope=col>start</th><th scope=col>end</th><th scope=col>context</th><th scope=col>gwas_study</th><th scope=col>method</th><th scope=col>is_imputable</th><th scope=col>is_selected_method</th><th scope=col>rsq_cv</th><th scope=col>pval_cv</th><th scope=col>twas_z</th><th scope=col>twas_pval</th><th scope=col>type</th><th scope=col>block</th><th scope=col>method_selected_original</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>bayes_l</td><td>TRUE</td><td>FALSE</td><td>0.1507800</td><td>1.287865e-29</td><td>-0.5663673</td><td>0.5711441</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>bayes_r</td><td>TRUE</td><td>FALSE</td><td>0.1523213</td><td>6.298261e-30</td><td>-0.5729437</td><td>0.5666828</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>enet   </td><td>TRUE</td><td> TRUE</td><td>0.1586583</td><td>3.283208e-31</td><td>-0.1316813</td><td>0.8952364</td><td>eQTL</td><td>chr10_100331627_104378781</td><td> TRUE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>lasso  </td><td>TRUE</td><td>FALSE</td><td>0.1470135</td><td>7.357674e-29</td><td>-0.1149415</td><td>0.9084915</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>mrash  </td><td>TRUE</td><td>FALSE</td><td>0.1283372</td><td>3.746242e-25</td><td>-0.4621670</td><td>0.6439616</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>susie  </td><td>TRUE</td><td>FALSE</td><td>0.1577097</td><td>5.116002e-31</td><td>-0.1690379</td><td>0.8657668</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " chr & molecular\\_id & TSS & start & end & context & gwas\\_study & method & is\\_imputable & is\\_selected\\_method & rsq\\_cv & pval\\_cv & twas\\_z & twas\\_pval & type & block & method\\_selected\\_original & region & study\\_context & source\\_file\\\\\n",
       " <int> & <chr> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <lgl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & bayes\\_l & TRUE & FALSE & 0.1507800 & 1.287865e-29 & -0.5663673 & 0.5711441 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & bayes\\_r & TRUE & FALSE & 0.1523213 & 6.298261e-30 & -0.5729437 & 0.5666828 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & enet    & TRUE &  TRUE & 0.1586583 & 3.283208e-31 & -0.1316813 & 0.8952364 & eQTL & chr10\\_100331627\\_104378781 &  TRUE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & lasso   & TRUE & FALSE & 0.1470135 & 7.357674e-29 & -0.1149415 & 0.9084915 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & mrash   & TRUE & FALSE & 0.1283372 & 3.746242e-25 & -0.4621670 & 0.6439616 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & susie   & TRUE & FALSE & 0.1577097 & 5.116002e-31 & -0.1690379 & 0.8657668 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 20\n",
       "\n",
       "| chr &lt;int&gt; | molecular_id &lt;chr&gt; | TSS &lt;int&gt; | start &lt;int&gt; | end &lt;int&gt; | context &lt;chr&gt; | gwas_study &lt;chr&gt; | method &lt;chr&gt; | is_imputable &lt;lgl&gt; | is_selected_method &lt;lgl&gt; | rsq_cv &lt;dbl&gt; | pval_cv &lt;dbl&gt; | twas_z &lt;dbl&gt; | twas_pval &lt;dbl&gt; | type &lt;chr&gt; | block &lt;chr&gt; | method_selected_original &lt;lgl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | bayes_l | TRUE | FALSE | 0.1507800 | 1.287865e-29 | -0.5663673 | 0.5711441 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | bayes_r | TRUE | FALSE | 0.1523213 | 6.298261e-30 | -0.5729437 | 0.5666828 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | enet    | TRUE |  TRUE | 0.1586583 | 3.283208e-31 | -0.1316813 | 0.8952364 | eQTL | chr10_100331627_104378781 |  TRUE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | lasso   | TRUE | FALSE | 0.1470135 | 7.357674e-29 | -0.1149415 | 0.9084915 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | mrash   | TRUE | FALSE | 0.1283372 | 3.746242e-25 | -0.4621670 | 0.6439616 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | susie   | TRUE | FALSE | 0.1577097 | 5.116002e-31 | -0.1690379 | 0.8657668 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "\n"
      ],
      "text/plain": [
       "  chr molecular_id    TSS       start    end       context           \n",
       "1 10  ENSG00000055950 100987515 99320000 102120000 DLPFC_DeJager_eQTL\n",
       "2 10  ENSG00000055950 100987515 99320000 102120000 DLPFC_DeJager_eQTL\n",
       "3 10  ENSG00000055950 100987515 99320000 102120000 DLPFC_DeJager_eQTL\n",
       "4 10  ENSG00000055950 100987515 99320000 102120000 DLPFC_DeJager_eQTL\n",
       "5 10  ENSG00000055950 100987515 99320000 102120000 DLPFC_DeJager_eQTL\n",
       "6 10  ENSG00000055950 100987515 99320000 102120000 DLPFC_DeJager_eQTL\n",
       "  gwas_study         method  is_imputable is_selected_method rsq_cv   \n",
       "1 unadjusted_wp_2020 bayes_l TRUE         FALSE              0.1507800\n",
       "2 unadjusted_wp_2020 bayes_r TRUE         FALSE              0.1523213\n",
       "3 unadjusted_wp_2020 enet    TRUE          TRUE              0.1586583\n",
       "4 unadjusted_wp_2020 lasso   TRUE         FALSE              0.1470135\n",
       "5 unadjusted_wp_2020 mrash   TRUE         FALSE              0.1283372\n",
       "6 unadjusted_wp_2020 susie   TRUE         FALSE              0.1577097\n",
       "  pval_cv      twas_z     twas_pval type block                    \n",
       "1 1.287865e-29 -0.5663673 0.5711441 eQTL chr10_100331627_104378781\n",
       "2 6.298261e-30 -0.5729437 0.5666828 eQTL chr10_100331627_104378781\n",
       "3 3.283208e-31 -0.1316813 0.8952364 eQTL chr10_100331627_104378781\n",
       "4 7.357674e-29 -0.1149415 0.9084915 eQTL chr10_100331627_104378781\n",
       "5 3.746242e-25 -0.4621670 0.6439616 eQTL chr10_100331627_104378781\n",
       "6 5.116002e-31 -0.1690379 0.8657668 eQTL chr10_100331627_104378781\n",
       "  method_selected_original region                    study_context \n",
       "1 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "2 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "3  TRUE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "4 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "5 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "6 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "  source_file                                                    \n",
       "1 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "2 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "3 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "4 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "5 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "6 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>44820</li><li>20</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 44820\n",
       "\\item 20\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 44820\n",
       "2. 20\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 44820    20"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(twas)\n",
    "dim(twas)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "668b29f4-54df-4efb-b3bd-1f3417e11108",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'DLPFC_DeJager_eQTL'"
      ],
      "text/latex": [
       "'DLPFC\\_DeJager\\_eQTL'"
      ],
      "text/markdown": [
       "'DLPFC_DeJager_eQTL'"
      ],
      "text/plain": [
       "[1] \"DLPFC_DeJager_eQTL\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7443"
      ],
      "text/latex": [
       "7443"
      ],
      "text/markdown": [
       "7443"
      ],
      "text/plain": [
       "[1] 7443"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## TWAS analysis were performed across 10 contexts and 13872 genes\n",
    "twas %>% pull(context) |> unique()#%>%length\n",
    "\n",
    "twas %>% pull(molecular_id)%>%unique()%>%length\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1ef0fe8a",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>context</th><th scope=col>gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>DLPFC_DeJager_eQTL</td><td>7443</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 2\n",
       "\\begin{tabular}{ll}\n",
       " context & gene\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t DLPFC\\_DeJager\\_eQTL & 7443\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 2\n",
       "\n",
       "| context &lt;chr&gt; | gene &lt;int&gt; |\n",
       "|---|---|\n",
       "| DLPFC_DeJager_eQTL | 7443 |\n",
       "\n"
      ],
      "text/plain": [
       "  context            gene\n",
       "1 DLPFC_DeJager_eQTL 7443"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## available genes by contexts\n",
    "twas |> group_by(context) |> summarize(gene = n_distinct(molecular_id))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "65d8afca-6867-416a-8e9f-e39bf07064b5",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "1213"
      ],
      "text/latex": [
       "1213"
      ],
      "text/markdown": [
       "1213"
      ],
      "text/plain": [
       "[1] 1213"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "twas_res_sig <- twas %>% filter(is_selected_method==\"TRUE\")%>%filter(twas_pval<=0.05)#%>%pull(context)%>%unique()\n",
    "twas_res_sig|> pull(molecular_id)%>%unique()%>%length\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a06ae9b6-477c-4b77-97c9-8763ab5e0bb8",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "7443"
      ],
      "text/latex": [
       "7443"
      ],
      "text/markdown": [
       "7443"
      ],
      "text/plain": [
       "[1] 7443"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "twas_DLPFC_bulk = twas |> filter(context == 'DLPFC_DeJager_eQTL') |> pull(molecular_id)%>%unique()%>%length\n",
    "twas_DLPFC_bulk"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a64c2d9f",
   "metadata": {},
   "source": [
    "#### MR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "89201606-c5eb-4bfb-9aaf-b026aec56c0f",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "mr <- fread('/mnt/lustre/lab/ctcn/hklein/motor_qtl_project/unadjusted_wp_2020/susie_twas/noQC_nonimputed_result/twas/ROSMAP_eQTL_wp_unadjusted.combined_mr.tsv.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "6303aa4a-76c7-4416-82d9-6fe4ec293aa1",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "mr_sig_unadj <- mr %>% .[\n",
    "  meta_pval < 0.05 / .N &          # Bonferroni correction\n",
    "  cpip >= 0.7 &                    # strong causal evidence\n",
    "  num_CS >= 1 &                    # as quested by Hans and Gao on Oct 24, 2025 to get a loose version\n",
    "  Q_pval > 0.01 &                  # no heterogeneity\n",
    "  I2 < 0.4                         # low heterogeneity\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "f3f48a47-880f-4e9b-aa3e-ad5faf4a1e4b",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Q</th><th scope=col>Q_pval</th><th scope=col>I2</th><th scope=col>context</th><th scope=col>cpip</th><th scope=col>gene_name</th><th scope=col>gwas_study</th><th scope=col>meta_eff</th><th scope=col>meta_pval</th><th scope=col>num_CS</th><th scope=col>num_IV</th><th scope=col>se_meta_eff</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>0.134</td><td>0.715</td><td>0</td><td>DLPFC_DeJager_eQTL</td><td>0.960</td><td>ENSG00000171206</td><td>unadjusted_wp_2020</td><td> 0.001</td><td>0</td><td>2</td><td>71</td><td>0</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz</td></tr>\n",
       "\t<tr><td>0.530</td><td>0.467</td><td>0</td><td>DLPFC_DeJager_eQTL</td><td>0.989</td><td>ENSG00000203780</td><td>unadjusted_wp_2020</td><td> 0.000</td><td>0</td><td>2</td><td>19</td><td>0</td><td>chr10_125506866_127341896</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_125506866_127341896.mr_result.tsv.gz</td></tr>\n",
       "\t<tr><td>0.000</td><td>1.000</td><td>0</td><td>DLPFC_DeJager_eQTL</td><td>1.024</td><td>ENSG00000148572</td><td>unadjusted_wp_2020</td><td>-0.001</td><td>0</td><td>1</td><td>89</td><td>0</td><td>chr10_62446953_64035328  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_62446953_64035328.mr_result.tsv.gz  </td></tr>\n",
       "\t<tr><td>0.000</td><td>1.000</td><td>0</td><td>DLPFC_DeJager_eQTL</td><td>0.950</td><td>ENSG00000165644</td><td>unadjusted_wp_2020</td><td> 0.002</td><td>0</td><td>1</td><td>18</td><td>0</td><td>chr10_74769575_77264695  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_74769575_77264695.mr_result.tsv.gz  </td></tr>\n",
       "\t<tr><td>0.313</td><td>0.855</td><td>0</td><td>DLPFC_DeJager_eQTL</td><td>0.954</td><td>ENSG00000188199</td><td>unadjusted_wp_2020</td><td> 0.000</td><td>0</td><td>3</td><td> 9</td><td>0</td><td>chr10_79309511_80126158  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_79309511_80126158.mr_result.tsv.gz  </td></tr>\n",
       "\t<tr><td>1.383</td><td>0.847</td><td>0</td><td>DLPFC_DeJager_eQTL</td><td>0.965</td><td>ENSG00000228570</td><td>unadjusted_wp_2020</td><td> 0.000</td><td>0</td><td>5</td><td>21</td><td>0</td><td>chr10_79309511_80126158  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_79309511_80126158.mr_result.tsv.gz  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 15\n",
       "\\begin{tabular}{lllllllllllllll}\n",
       " Q & Q\\_pval & I2 & context & cpip & gene\\_name & gwas\\_study & meta\\_eff & meta\\_pval & num\\_CS & num\\_IV & se\\_meta\\_eff & region & study\\_context & source\\_file\\\\\n",
       " <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <chr> & <chr> & <dbl> & <dbl> & <int> & <int> & <dbl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 0.134 & 0.715 & 0 & DLPFC\\_DeJager\\_eQTL & 0.960 & ENSG00000171206 & unadjusted\\_wp\\_2020 &  0.001 & 0 & 2 & 71 & 0 & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.mr\\_result.tsv.gz\\\\\n",
       "\t 0.530 & 0.467 & 0 & DLPFC\\_DeJager\\_eQTL & 0.989 & ENSG00000203780 & unadjusted\\_wp\\_2020 &  0.000 & 0 & 2 & 19 & 0 & chr10\\_125506866\\_127341896 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_125506866\\_127341896.mr\\_result.tsv.gz\\\\\n",
       "\t 0.000 & 1.000 & 0 & DLPFC\\_DeJager\\_eQTL & 1.024 & ENSG00000148572 & unadjusted\\_wp\\_2020 & -0.001 & 0 & 1 & 89 & 0 & chr10\\_62446953\\_64035328   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_62446953\\_64035328.mr\\_result.tsv.gz  \\\\\n",
       "\t 0.000 & 1.000 & 0 & DLPFC\\_DeJager\\_eQTL & 0.950 & ENSG00000165644 & unadjusted\\_wp\\_2020 &  0.002 & 0 & 1 & 18 & 0 & chr10\\_74769575\\_77264695   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_74769575\\_77264695.mr\\_result.tsv.gz  \\\\\n",
       "\t 0.313 & 0.855 & 0 & DLPFC\\_DeJager\\_eQTL & 0.954 & ENSG00000188199 & unadjusted\\_wp\\_2020 &  0.000 & 0 & 3 &  9 & 0 & chr10\\_79309511\\_80126158   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_79309511\\_80126158.mr\\_result.tsv.gz  \\\\\n",
       "\t 1.383 & 0.847 & 0 & DLPFC\\_DeJager\\_eQTL & 0.965 & ENSG00000228570 & unadjusted\\_wp\\_2020 &  0.000 & 0 & 5 & 21 & 0 & chr10\\_79309511\\_80126158   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_79309511\\_80126158.mr\\_result.tsv.gz  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 15\n",
       "\n",
       "| Q &lt;dbl&gt; | Q_pval &lt;dbl&gt; | I2 &lt;dbl&gt; | context &lt;chr&gt; | cpip &lt;dbl&gt; | gene_name &lt;chr&gt; | gwas_study &lt;chr&gt; | meta_eff &lt;dbl&gt; | meta_pval &lt;dbl&gt; | num_CS &lt;int&gt; | num_IV &lt;int&gt; | se_meta_eff &lt;dbl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 0.134 | 0.715 | 0 | DLPFC_DeJager_eQTL | 0.960 | ENSG00000171206 | unadjusted_wp_2020 |  0.001 | 0 | 2 | 71 | 0 | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz |\n",
       "| 0.530 | 0.467 | 0 | DLPFC_DeJager_eQTL | 0.989 | ENSG00000203780 | unadjusted_wp_2020 |  0.000 | 0 | 2 | 19 | 0 | chr10_125506866_127341896 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_125506866_127341896.mr_result.tsv.gz |\n",
       "| 0.000 | 1.000 | 0 | DLPFC_DeJager_eQTL | 1.024 | ENSG00000148572 | unadjusted_wp_2020 | -0.001 | 0 | 1 | 89 | 0 | chr10_62446953_64035328   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_62446953_64035328.mr_result.tsv.gz   |\n",
       "| 0.000 | 1.000 | 0 | DLPFC_DeJager_eQTL | 0.950 | ENSG00000165644 | unadjusted_wp_2020 |  0.002 | 0 | 1 | 18 | 0 | chr10_74769575_77264695   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_74769575_77264695.mr_result.tsv.gz   |\n",
       "| 0.313 | 0.855 | 0 | DLPFC_DeJager_eQTL | 0.954 | ENSG00000188199 | unadjusted_wp_2020 |  0.000 | 0 | 3 |  9 | 0 | chr10_79309511_80126158   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_79309511_80126158.mr_result.tsv.gz   |\n",
       "| 1.383 | 0.847 | 0 | DLPFC_DeJager_eQTL | 0.965 | ENSG00000228570 | unadjusted_wp_2020 |  0.000 | 0 | 5 | 21 | 0 | chr10_79309511_80126158   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_79309511_80126158.mr_result.tsv.gz   |\n",
       "\n"
      ],
      "text/plain": [
       "  Q     Q_pval I2 context            cpip  gene_name       gwas_study        \n",
       "1 0.134 0.715  0  DLPFC_DeJager_eQTL 0.960 ENSG00000171206 unadjusted_wp_2020\n",
       "2 0.530 0.467  0  DLPFC_DeJager_eQTL 0.989 ENSG00000203780 unadjusted_wp_2020\n",
       "3 0.000 1.000  0  DLPFC_DeJager_eQTL 1.024 ENSG00000148572 unadjusted_wp_2020\n",
       "4 0.000 1.000  0  DLPFC_DeJager_eQTL 0.950 ENSG00000165644 unadjusted_wp_2020\n",
       "5 0.313 0.855  0  DLPFC_DeJager_eQTL 0.954 ENSG00000188199 unadjusted_wp_2020\n",
       "6 1.383 0.847  0  DLPFC_DeJager_eQTL 0.965 ENSG00000228570 unadjusted_wp_2020\n",
       "  meta_eff meta_pval num_CS num_IV se_meta_eff region                   \n",
       "1  0.001   0         2      71     0           chr10_100331627_104378781\n",
       "2  0.000   0         2      19     0           chr10_125506866_127341896\n",
       "3 -0.001   0         1      89     0           chr10_62446953_64035328  \n",
       "4  0.002   0         1      18     0           chr10_74769575_77264695  \n",
       "5  0.000   0         3       9     0           chr10_79309511_80126158  \n",
       "6  0.000   0         5      21     0           chr10_79309511_80126158  \n",
       "  study_context \n",
       "1 ROSMAP_eQTL_wp\n",
       "2 ROSMAP_eQTL_wp\n",
       "3 ROSMAP_eQTL_wp\n",
       "4 ROSMAP_eQTL_wp\n",
       "5 ROSMAP_eQTL_wp\n",
       "6 ROSMAP_eQTL_wp\n",
       "  source_file                                                         \n",
       "1 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz\n",
       "2 ROSMAP_eQTL_wp_unadjusted.chr10_125506866_127341896.mr_result.tsv.gz\n",
       "3 ROSMAP_eQTL_wp_unadjusted.chr10_62446953_64035328.mr_result.tsv.gz  \n",
       "4 ROSMAP_eQTL_wp_unadjusted.chr10_74769575_77264695.mr_result.tsv.gz  \n",
       "5 ROSMAP_eQTL_wp_unadjusted.chr10_79309511_80126158.mr_result.tsv.gz  \n",
       "6 ROSMAP_eQTL_wp_unadjusted.chr10_79309511_80126158.mr_result.tsv.gz  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>80</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 80\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 80\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 80 15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(mr_sig_unadj)\n",
    "dim(mr_sig_unadj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "da1af503",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "80"
      ],
      "text/latex": [
       "80"
      ],
      "text/markdown": [
       "80"
      ],
      "text/plain": [
       "[1] 80"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mr_sig_unadj |> pull(gene_name)%>%unique()%>%length"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "37214743",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>context</th><th scope=col>gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>DLPFC_DeJager_eQTL</td><td>80</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 2\n",
       "\\begin{tabular}{ll}\n",
       " context & gene\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t DLPFC\\_DeJager\\_eQTL & 80\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 2\n",
       "\n",
       "| context &lt;chr&gt; | gene &lt;int&gt; |\n",
       "|---|---|\n",
       "| DLPFC_DeJager_eQTL | 80 |\n",
       "\n"
      ],
      "text/plain": [
       "  context            gene\n",
       "1 DLPFC_DeJager_eQTL 80  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mr_sig_unadj |> group_by(context) |> summarize(gene = n_distinct(gene_name))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "567e49d6",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "36"
      ],
      "text/latex": [
       "36"
      ],
      "text/markdown": [
       "36"
      ],
      "text/plain": [
       "[1] 36"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## Regardless of context, there are 60 genes have both TWAS significant and MR significant asscoaitions\n",
    "intersect(twas_unadj_sig$molecular_id, mr_sig_unadj$gene_name) |> unique()|> length()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c45de04b-ce4d-467a-b981-a1ec5b4511eb",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "### summarize significant TWAS & MR genes\n",
    "\n",
    "* Significant TWAS associations were defined using two criteria: (i) the most accurate prediction model yielded a TWAS p-value below the Bonferroni-corrected threshold, or (ii) at least half of the prediction models produced p-values below this threshold.\n",
    "* Among the 7,443 total genes from DLPFC bulk eQTL, 67 genes show significant TWAS associations\n",
    "\n",
    "* Mendelian randomization (MR) associations were considered significant if they met all of the following criteria: a meta-analysis p-value passing Bonferroni correction (meta-analysis p < 0.05 divided by the number of tested genes), strong causal evidence as indicated by a causal posterior inclusion probability (CPIP) ≥ 0.7, at least one credible set (num_CS ≥ 1), and no evidence of substantial heterogeneity (Cochran’s Q test p > 0.01 and I² < 0.4).\n",
    "* For DLPFC bulk eQTL, 80 genes show significant MR associations.\n",
    "* For DLPFC bulk eQTL, 36 genes are both MR and TWAS significant. \n",
    "* KAT8, SLC39A13 and ZNF668 are significant TWAS gene for both gait speed and AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "51741b66-ce32-4b11-8989-71f41c6a4f0f",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "7443"
      ],
      "text/latex": [
       "7443"
      ],
      "text/markdown": [
       "7443"
      ],
      "text/plain": [
       "[1] 7443"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "n_imputable_genes <- twas %>%\n",
    "  filter(is_imputable) %>%                  # keep only imputable rows\n",
    "  distinct(molecular_id, context, gwas_study) %>%         # count unique gene-context pairs\n",
    "  nrow()\n",
    "n_imputable_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "607cbcd1-b856-4edb-95c3-fe521a8ff7a9",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "6.71772134891845e-06"
      ],
      "text/latex": [
       "6.71772134891845e-06"
      ],
      "text/markdown": [
       "6.71772134891845e-06"
      ],
      "text/plain": [
       "[1] 6.717721e-06"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Compute cutoff using Bonferroni-style correction\n",
    "unadj_cutoff <- 0.05 / n_imputable_genes\n",
    "unadj_cutoff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "e78ef1f1-b8fe-42e8-973f-ad843b7357ed",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(molecular_id, context, gwas_study)`\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>67</li><li>25</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 67\n",
       "\\item 25\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 67\n",
       "2. 25\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 67 25"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "twas_unadj_sig <- twas %>%\n",
    "  mutate(sig = twas_pval < unadj_cutoff) %>%\n",
    "  group_by(molecular_id, context, gwas_study) %>%\n",
    "  summarise(\n",
    "    n_methods = n(),\n",
    "    n_sig = sum(sig, na.rm = T),\n",
    "    best_sig = any(sig & is_selected_method, na.rm = T),\n",
    "    .groups = \"drop\"\n",
    "  ) %>%\n",
    "  mutate(pass = (n_sig / n_methods >= 0.5) | best_sig) %>%\n",
    "  mutate(pass_strict = (n_sig / n_methods >= 0.5)) %>%\n",
    "  filter(pass) %>% left_join(twas) %>% arrange(twas_pval) %>% distinct(molecular_id, .keep_all = T)\n",
    "twas_unadj_sig %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "193c534f-c425-49a6-9b1d-97e364f6b9da",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "gene_ref <- fread('/mnt/lustre/home/rl3328/rl3328/resource/Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.region_list.2025')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "dd79014c-34fb-45ff-b051-bdf18c61ce38",
   "metadata": {
    "kernel": "R",
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# twas_unadj_sig |> left_join(gene_ref, by = c(\"molecular_id\" = \"gene_id\")) |> select(molecular_id, gene_name, everything())\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33c9361d-3ab3-42a1-8cfd-2a1e2a5ae97f",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "## Figure data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "d2409cc8-d679-4919-8255-325ac354695f",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“\u001b[1m\u001b[22mThere were 34 warnings in `summarise()`.\n",
      "The first warning was:\n",
      "\u001b[1m\u001b[22m\u001b[36mℹ\u001b[39m In argument: `best_method_pval = min(twas_pval[is_selected_method], na.rm =\n",
      "  TRUE)`.\n",
      "\u001b[36mℹ\u001b[39m In group 113: `molecular_id = \"ENSG00000011465\"`, `context =\n",
      "  \"DLPFC_DeJager_eQTL\"`, `gwas_study = \"unadjusted_wp_2020\"`.\n",
      "Caused by warning in `min()`:\n",
      "\u001b[33m!\u001b[39m no non-missing arguments to min; returning Inf\n",
      "\u001b[1m\u001b[22m\u001b[36mℹ\u001b[39m Run `dplyr::last_dplyr_warnings()` to see the 33 remaining warnings.”\n"
     ]
    }
   ],
   "source": [
    "df <- twas %>%\n",
    "  mutate(sig = twas_pval < unadj_cutoff) %>%\n",
    "  group_by(molecular_id, context, gwas_study) %>%\n",
    "  summarise(\n",
    "    n_methods = n(),\n",
    "    n_sig = sum(sig, na.rm = TRUE),\n",
    "    best_pval = min(twas_pval, na.rm = TRUE),\n",
    "    best_method_pval = min(twas_pval[is_selected_method], na.rm = TRUE),\n",
    "    best_method_selected = any(sig & is_selected_method, na.rm = TRUE),\n",
    "    .groups = \"drop\"\n",
    "  ) %>%\n",
    "  mutate(\n",
    "    pass = (n_sig / n_methods >= 0.5) | best_method_selected, # pass defined by two threshold, best or half methods satisfied\n",
    "    final_pval = ifelse(n_sig / n_methods >= 0.5, best_pval, best_method_pval),\n",
    "    # fallback: if no selected method existed (Inf), use best_pval\n",
    "    final_pval = ifelse(is.infinite(final_pval), best_pval, final_pval)\n",
    "  ) %>%\n",
    "  left_join(twas, by = c(\"molecular_id\", \"context\", \"gwas_study\")) %>%\n",
    "  filter(twas_pval == final_pval) %>%\n",
    "  distinct(molecular_id, .keep_all = TRUE) %>%\n",
    "  mutate(\n",
    "    logp = -log10(twas_pval) * sign(twas_z),\n",
    "    pos = (start + end) / 2\n",
    "  ) %>%\n",
    "  merge(gene_ref, by.x = 'molecular_id', by.y = 'gene_id')\n",
    "\n",
    "# Order chromosomes\n",
    "df$chr <- factor(df$chr, levels = as.character(1:22))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "05d4225a",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>7443</li><li>33</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 7443\n",
       "\\item 33\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 7443\n",
       "2. 33\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 7443   33"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "bc6550ab",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>molecular_id</th><th scope=col>context</th><th scope=col>gwas_study</th><th scope=col>n_methods</th><th scope=col>n_sig</th><th scope=col>best_pval</th><th scope=col>best_method_pval</th><th scope=col>best_method_selected</th><th scope=col>pass</th><th scope=col>final_pval</th><th scope=col>⋯</th><th scope=col>method_selected_original</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th><th scope=col>logp</th><th scope=col>pos</th><th scope=col>#chr</th><th scope=col>TSS.y</th><th scope=col>TES</th><th scope=col>gene_name</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ENSG00000000457</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>6</td><td>0</td><td>0.506484945</td><td>0.506484945</td><td>FALSE</td><td>FALSE</td><td>0.506484945</td><td>⋯</td><td>TRUE</td><td>chr1_168438717_170228106</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz</td><td> 0.2954335</td><td>169540000</td><td>chr1</td><td>169894267</td><td>169849631</td><td>SCYL3   </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ENSG00000000460</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>6</td><td>0</td><td>0.400370493</td><td>0.633623327</td><td>FALSE</td><td>FALSE</td><td>0.633623327</td><td>⋯</td><td>TRUE</td><td>chr1_168438717_170228106</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz</td><td> 0.1981688</td><td>169540000</td><td>chr1</td><td>169662007</td><td>169854080</td><td>C1orf112</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>ENSG00000000971</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>6</td><td>0</td><td>0.006181974</td><td>0.007051131</td><td>FALSE</td><td>FALSE</td><td>0.007051131</td><td>⋯</td><td>TRUE</td><td>chr1_195599253_199271134</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr1_195599253_199271134.twas.tsv.gz</td><td> 2.1517412</td><td>195093752</td><td>chr1</td><td>196652043</td><td>196747504</td><td>CFH     </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>ENSG00000001084</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>6</td><td>0</td><td>0.147556313</td><td>0.502194806</td><td>FALSE</td><td>FALSE</td><td>0.502194806</td><td>⋯</td><td>TRUE</td><td>chr6_52730905_54027603  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr6_52730905_54027603.twas.tsv.gz  </td><td> 0.2991278</td><td> 53557156</td><td>chr6</td><td> 53616970</td><td> 53497341</td><td>GCLC    </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ENSG00000001460</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>6</td><td>0</td><td>0.659517185</td><td>0.659517185</td><td>FALSE</td><td>FALSE</td><td>0.659517185</td><td>⋯</td><td>TRUE</td><td>chr1_24199848_26390306  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr1_24199848_26390306.twas.tsv.gz  </td><td>-0.1807739</td><td> 24386966</td><td>chr1</td><td> 24416934</td><td> 24356999</td><td>STPG1   </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ENSG00000001461</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>6</td><td>0</td><td>0.414763094</td><td>0.778955410</td><td>FALSE</td><td>FALSE</td><td>0.778955410</td><td>⋯</td><td>TRUE</td><td>chr1_24199848_26390306  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr1_24199848_26390306.twas.tsv.gz  </td><td>-0.1084874</td><td> 24444388</td><td>chr1</td><td> 24415802</td><td> 24472976</td><td>NIPAL3  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 33\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & molecular\\_id & context & gwas\\_study & n\\_methods & n\\_sig & best\\_pval & best\\_method\\_pval & best\\_method\\_selected & pass & final\\_pval & ⋯ & method\\_selected\\_original & region & study\\_context & source\\_file & logp & pos & \\#chr & TSS.y & TES & gene\\_name\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <int> & <dbl> & <dbl> & <lgl> & <lgl> & <dbl> & ⋯ & <lgl> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ENSG00000000457 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 6 & 0 & 0.506484945 & 0.506484945 & FALSE & FALSE & 0.506484945 & ⋯ & TRUE & chr1\\_168438717\\_170228106 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr1\\_168438717\\_170228106.twas.tsv.gz &  0.2954335 & 169540000 & chr1 & 169894267 & 169849631 & SCYL3   \\\\\n",
       "\t2 & ENSG00000000460 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 6 & 0 & 0.400370493 & 0.633623327 & FALSE & FALSE & 0.633623327 & ⋯ & TRUE & chr1\\_168438717\\_170228106 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr1\\_168438717\\_170228106.twas.tsv.gz &  0.1981688 & 169540000 & chr1 & 169662007 & 169854080 & C1orf112\\\\\n",
       "\t3 & ENSG00000000971 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 6 & 0 & 0.006181974 & 0.007051131 & FALSE & FALSE & 0.007051131 & ⋯ & TRUE & chr1\\_195599253\\_199271134 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr1\\_195599253\\_199271134.twas.tsv.gz &  2.1517412 & 195093752 & chr1 & 196652043 & 196747504 & CFH     \\\\\n",
       "\t4 & ENSG00000001084 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 6 & 0 & 0.147556313 & 0.502194806 & FALSE & FALSE & 0.502194806 & ⋯ & TRUE & chr6\\_52730905\\_54027603   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr6\\_52730905\\_54027603.twas.tsv.gz   &  0.2991278 &  53557156 & chr6 &  53616970 &  53497341 & GCLC    \\\\\n",
       "\t5 & ENSG00000001460 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 6 & 0 & 0.659517185 & 0.659517185 & FALSE & FALSE & 0.659517185 & ⋯ & TRUE & chr1\\_24199848\\_26390306   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr1\\_24199848\\_26390306.twas.tsv.gz   & -0.1807739 &  24386966 & chr1 &  24416934 &  24356999 & STPG1   \\\\\n",
       "\t6 & ENSG00000001461 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 6 & 0 & 0.414763094 & 0.778955410 & FALSE & FALSE & 0.778955410 & ⋯ & TRUE & chr1\\_24199848\\_26390306   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr1\\_24199848\\_26390306.twas.tsv.gz   & -0.1084874 &  24444388 & chr1 &  24415802 &  24472976 & NIPAL3  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 33\n",
       "\n",
       "| <!--/--> | molecular_id &lt;chr&gt; | context &lt;chr&gt; | gwas_study &lt;chr&gt; | n_methods &lt;int&gt; | n_sig &lt;int&gt; | best_pval &lt;dbl&gt; | best_method_pval &lt;dbl&gt; | best_method_selected &lt;lgl&gt; | pass &lt;lgl&gt; | final_pval &lt;dbl&gt; | ⋯ ⋯ | method_selected_original &lt;lgl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; | logp &lt;dbl&gt; | pos &lt;dbl&gt; | #chr &lt;chr&gt; | TSS.y &lt;int&gt; | TES &lt;int&gt; | gene_name &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ENSG00000000457 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | 6 | 0 | 0.506484945 | 0.506484945 | FALSE | FALSE | 0.506484945 | ⋯ | TRUE | chr1_168438717_170228106 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz |  0.2954335 | 169540000 | chr1 | 169894267 | 169849631 | SCYL3    |\n",
       "| 2 | ENSG00000000460 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | 6 | 0 | 0.400370493 | 0.633623327 | FALSE | FALSE | 0.633623327 | ⋯ | TRUE | chr1_168438717_170228106 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz |  0.1981688 | 169540000 | chr1 | 169662007 | 169854080 | C1orf112 |\n",
       "| 3 | ENSG00000000971 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | 6 | 0 | 0.006181974 | 0.007051131 | FALSE | FALSE | 0.007051131 | ⋯ | TRUE | chr1_195599253_199271134 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr1_195599253_199271134.twas.tsv.gz |  2.1517412 | 195093752 | chr1 | 196652043 | 196747504 | CFH      |\n",
       "| 4 | ENSG00000001084 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | 6 | 0 | 0.147556313 | 0.502194806 | FALSE | FALSE | 0.502194806 | ⋯ | TRUE | chr6_52730905_54027603   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr6_52730905_54027603.twas.tsv.gz   |  0.2991278 |  53557156 | chr6 |  53616970 |  53497341 | GCLC     |\n",
       "| 5 | ENSG00000001460 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | 6 | 0 | 0.659517185 | 0.659517185 | FALSE | FALSE | 0.659517185 | ⋯ | TRUE | chr1_24199848_26390306   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr1_24199848_26390306.twas.tsv.gz   | -0.1807739 |  24386966 | chr1 |  24416934 |  24356999 | STPG1    |\n",
       "| 6 | ENSG00000001461 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | 6 | 0 | 0.414763094 | 0.778955410 | FALSE | FALSE | 0.778955410 | ⋯ | TRUE | chr1_24199848_26390306   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr1_24199848_26390306.twas.tsv.gz   | -0.1084874 |  24444388 | chr1 |  24415802 |  24472976 | NIPAL3   |\n",
       "\n"
      ],
      "text/plain": [
       "  molecular_id    context            gwas_study         n_methods n_sig\n",
       "1 ENSG00000000457 DLPFC_DeJager_eQTL unadjusted_wp_2020 6         0    \n",
       "2 ENSG00000000460 DLPFC_DeJager_eQTL unadjusted_wp_2020 6         0    \n",
       "3 ENSG00000000971 DLPFC_DeJager_eQTL unadjusted_wp_2020 6         0    \n",
       "4 ENSG00000001084 DLPFC_DeJager_eQTL unadjusted_wp_2020 6         0    \n",
       "5 ENSG00000001460 DLPFC_DeJager_eQTL unadjusted_wp_2020 6         0    \n",
       "6 ENSG00000001461 DLPFC_DeJager_eQTL unadjusted_wp_2020 6         0    \n",
       "  best_pval   best_method_pval best_method_selected pass  final_pval  ⋯\n",
       "1 0.506484945 0.506484945      FALSE                FALSE 0.506484945 ⋯\n",
       "2 0.400370493 0.633623327      FALSE                FALSE 0.633623327 ⋯\n",
       "3 0.006181974 0.007051131      FALSE                FALSE 0.007051131 ⋯\n",
       "4 0.147556313 0.502194806      FALSE                FALSE 0.502194806 ⋯\n",
       "5 0.659517185 0.659517185      FALSE                FALSE 0.659517185 ⋯\n",
       "6 0.414763094 0.778955410      FALSE                FALSE 0.778955410 ⋯\n",
       "  method_selected_original region                   study_context \n",
       "1 TRUE                     chr1_168438717_170228106 ROSMAP_eQTL_wp\n",
       "2 TRUE                     chr1_168438717_170228106 ROSMAP_eQTL_wp\n",
       "3 TRUE                     chr1_195599253_199271134 ROSMAP_eQTL_wp\n",
       "4 TRUE                     chr6_52730905_54027603   ROSMAP_eQTL_wp\n",
       "5 TRUE                     chr1_24199848_26390306   ROSMAP_eQTL_wp\n",
       "6 TRUE                     chr1_24199848_26390306   ROSMAP_eQTL_wp\n",
       "  source_file                                                    logp      \n",
       "1 ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz  0.2954335\n",
       "2 ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz  0.1981688\n",
       "3 ROSMAP_eQTL_wp_unadjusted.chr1_195599253_199271134.twas.tsv.gz  2.1517412\n",
       "4 ROSMAP_eQTL_wp_unadjusted.chr6_52730905_54027603.twas.tsv.gz    0.2991278\n",
       "5 ROSMAP_eQTL_wp_unadjusted.chr1_24199848_26390306.twas.tsv.gz   -0.1807739\n",
       "6 ROSMAP_eQTL_wp_unadjusted.chr1_24199848_26390306.twas.tsv.gz   -0.1084874\n",
       "  pos       #chr TSS.y     TES       gene_name\n",
       "1 169540000 chr1 169894267 169849631 SCYL3    \n",
       "2 169540000 chr1 169662007 169854080 C1orf112 \n",
       "3 195093752 chr1 196652043 196747504 CFH      \n",
       "4  53557156 chr6  53616970  53497341 GCLC     \n",
       "5  24386966 chr1  24416934  24356999 STPG1    \n",
       "6  24444388 chr1  24415802  24472976 NIPAL3   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "240a37d2-b50f-42d9-a8ba-177d0ea16f00",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "chr_info <- df %>%\n",
    "  group_by(chr) %>%\n",
    "  summarise(chr_len = max(pos)) %>%\n",
    "  mutate(chr_start = lag(cumsum(chr_len), default = 0))\n",
    "\n",
    "df <- df %>%\n",
    "  left_join(chr_info, by = \"chr\") %>%\n",
    "  mutate(cum_pos = pos + chr_start)\n",
    "  \n",
    "axis_df <- chr_info %>%\n",
    "  mutate(center = chr_start + chr_len/2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "f884ab9d-068c-4ee3-bcf9-67182c3e2b0a",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "df <- df %>%\n",
    "  left_join(\n",
    "    mr_sig_unadj %>% rename(molecular_id = gene_name),\n",
    "    by = c(\"molecular_id\", \"context\")\n",
    "  )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "7a99be0a-499f-4ee3-adad-d0a564927c3e",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# df %>% filter(!is.na(Q_pval)) %>% select(molecular_id, context, gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "6b60d619",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 49</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>molecular_id</th><th scope=col>context</th><th scope=col>gwas_study.x</th><th scope=col>n_methods</th><th scope=col>n_sig</th><th scope=col>best_pval</th><th scope=col>best_method_pval</th><th scope=col>best_method_selected</th><th scope=col>pass</th><th scope=col>final_pval</th><th scope=col>⋯</th><th scope=col>cpip</th><th scope=col>gwas_study.y</th><th scope=col>meta_eff</th><th scope=col>meta_pval</th><th scope=col>num_CS</th><th scope=col>num_IV</th><th scope=col>se_meta_eff</th><th scope=col>region.y</th><th scope=col>study_context.y</th><th scope=col>source_file.y</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ENSG00000000457</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>6</td><td>0</td><td>0.506484945</td><td>0.506484945</td><td>FALSE</td><td>FALSE</td><td>0.506484945</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ENSG00000000460</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>6</td><td>0</td><td>0.400370493</td><td>0.633623327</td><td>FALSE</td><td>FALSE</td><td>0.633623327</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>ENSG00000000971</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>6</td><td>0</td><td>0.006181974</td><td>0.007051131</td><td>FALSE</td><td>FALSE</td><td>0.007051131</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>ENSG00000001084</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>6</td><td>0</td><td>0.147556313</td><td>0.502194806</td><td>FALSE</td><td>FALSE</td><td>0.502194806</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ENSG00000001460</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>6</td><td>0</td><td>0.659517185</td><td>0.659517185</td><td>FALSE</td><td>FALSE</td><td>0.659517185</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ENSG00000001461</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>6</td><td>0</td><td>0.414763094</td><td>0.778955410</td><td>FALSE</td><td>FALSE</td><td>0.778955410</td><td>⋯</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 49\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & molecular\\_id & context & gwas\\_study.x & n\\_methods & n\\_sig & best\\_pval & best\\_method\\_pval & best\\_method\\_selected & pass & final\\_pval & ⋯ & cpip & gwas\\_study.y & meta\\_eff & meta\\_pval & num\\_CS & num\\_IV & se\\_meta\\_eff & region.y & study\\_context.y & source\\_file.y\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <int> & <dbl> & <dbl> & <lgl> & <lgl> & <dbl> & ⋯ & <dbl> & <chr> & <dbl> & <dbl> & <int> & <int> & <dbl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ENSG00000000457 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 6 & 0 & 0.506484945 & 0.506484945 & FALSE & FALSE & 0.506484945 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t2 & ENSG00000000460 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 6 & 0 & 0.400370493 & 0.633623327 & FALSE & FALSE & 0.633623327 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t3 & ENSG00000000971 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 6 & 0 & 0.006181974 & 0.007051131 & FALSE & FALSE & 0.007051131 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t4 & ENSG00000001084 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 6 & 0 & 0.147556313 & 0.502194806 & FALSE & FALSE & 0.502194806 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t5 & ENSG00000001460 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 6 & 0 & 0.659517185 & 0.659517185 & FALSE & FALSE & 0.659517185 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\t6 & ENSG00000001461 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 6 & 0 & 0.414763094 & 0.778955410 & FALSE & FALSE & 0.778955410 & ⋯ & NA & NA & NA & NA & NA & NA & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 49\n",
       "\n",
       "| <!--/--> | molecular_id &lt;chr&gt; | context &lt;chr&gt; | gwas_study.x &lt;chr&gt; | n_methods &lt;int&gt; | n_sig &lt;int&gt; | best_pval &lt;dbl&gt; | best_method_pval &lt;dbl&gt; | best_method_selected &lt;lgl&gt; | pass &lt;lgl&gt; | final_pval &lt;dbl&gt; | ⋯ ⋯ | cpip &lt;dbl&gt; | gwas_study.y &lt;chr&gt; | meta_eff &lt;dbl&gt; | meta_pval &lt;dbl&gt; | num_CS &lt;int&gt; | num_IV &lt;int&gt; | se_meta_eff &lt;dbl&gt; | region.y &lt;chr&gt; | study_context.y &lt;chr&gt; | source_file.y &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ENSG00000000457 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | 6 | 0 | 0.506484945 | 0.506484945 | FALSE | FALSE | 0.506484945 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 2 | ENSG00000000460 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | 6 | 0 | 0.400370493 | 0.633623327 | FALSE | FALSE | 0.633623327 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 3 | ENSG00000000971 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | 6 | 0 | 0.006181974 | 0.007051131 | FALSE | FALSE | 0.007051131 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 4 | ENSG00000001084 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | 6 | 0 | 0.147556313 | 0.502194806 | FALSE | FALSE | 0.502194806 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 5 | ENSG00000001460 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | 6 | 0 | 0.659517185 | 0.659517185 | FALSE | FALSE | 0.659517185 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "| 6 | ENSG00000001461 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | 6 | 0 | 0.414763094 | 0.778955410 | FALSE | FALSE | 0.778955410 | ⋯ | NA | NA | NA | NA | NA | NA | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  molecular_id    context            gwas_study.x       n_methods n_sig\n",
       "1 ENSG00000000457 DLPFC_DeJager_eQTL unadjusted_wp_2020 6         0    \n",
       "2 ENSG00000000460 DLPFC_DeJager_eQTL unadjusted_wp_2020 6         0    \n",
       "3 ENSG00000000971 DLPFC_DeJager_eQTL unadjusted_wp_2020 6         0    \n",
       "4 ENSG00000001084 DLPFC_DeJager_eQTL unadjusted_wp_2020 6         0    \n",
       "5 ENSG00000001460 DLPFC_DeJager_eQTL unadjusted_wp_2020 6         0    \n",
       "6 ENSG00000001461 DLPFC_DeJager_eQTL unadjusted_wp_2020 6         0    \n",
       "  best_pval   best_method_pval best_method_selected pass  final_pval  ⋯ cpip\n",
       "1 0.506484945 0.506484945      FALSE                FALSE 0.506484945 ⋯ NA  \n",
       "2 0.400370493 0.633623327      FALSE                FALSE 0.633623327 ⋯ NA  \n",
       "3 0.006181974 0.007051131      FALSE                FALSE 0.007051131 ⋯ NA  \n",
       "4 0.147556313 0.502194806      FALSE                FALSE 0.502194806 ⋯ NA  \n",
       "5 0.659517185 0.659517185      FALSE                FALSE 0.659517185 ⋯ NA  \n",
       "6 0.414763094 0.778955410      FALSE                FALSE 0.778955410 ⋯ NA  \n",
       "  gwas_study.y meta_eff meta_pval num_CS num_IV se_meta_eff region.y\n",
       "1 NA           NA       NA        NA     NA     NA          NA      \n",
       "2 NA           NA       NA        NA     NA     NA          NA      \n",
       "3 NA           NA       NA        NA     NA     NA          NA      \n",
       "4 NA           NA       NA        NA     NA     NA          NA      \n",
       "5 NA           NA       NA        NA     NA     NA          NA      \n",
       "6 NA           NA       NA        NA     NA     NA          NA      \n",
       "  study_context.y source_file.y\n",
       "1 NA              NA           \n",
       "2 NA              NA           \n",
       "3 NA              NA           \n",
       "4 NA              NA           \n",
       "5 NA              NA           \n",
       "6 NA              NA           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>7443</li><li>49</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 7443\n",
       "\\item 49\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 7443\n",
       "2. 49\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 7443   49"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)\n",
    "dim(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "d5e7f215-7297-4242-97c3-329cb21953f7",
   "metadata": {
    "kernel": "R",
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "\n",
    "# Add a column for MR significance (example: FDR < 0.05)\n",
    "df <- df %>%\n",
    "  mutate(category = case_when(\n",
    "    !is.na(Q_pval) & pass ~ \"TWAS & MR\",        # Both significant\n",
    "    is.na(Q_pval) & pass ~ \"TWAS only\",         # Only TWAS significant\n",
    "    !is.na(Q_pval) & !pass ~ \"MR only\",         # Only MR significant\n",
    "    is.na(Q_pval) & !pass ~ \"Not Significant\",  # Neither significant\n",
    "    TRUE ~ \"Others\"                              # Catch-all for remaining cases\n",
    "  ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "d1018956",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>category</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>MR only        </td><td>  44</td></tr>\n",
       "\t<tr><td>Not Significant</td><td>7332</td></tr>\n",
       "\t<tr><td><span style=white-space:pre-wrap>TWAS &amp; MR      </span></td><td><span style=white-space:pre-wrap>  36</span></td></tr>\n",
       "\t<tr><td>TWAS only      </td><td>  31</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4 × 2\n",
       "\\begin{tabular}{ll}\n",
       " category & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t MR only         &   44\\\\\n",
       "\t Not Significant & 7332\\\\\n",
       "\t TWAS \\& MR       &   36\\\\\n",
       "\t TWAS only       &   31\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4 × 2\n",
       "\n",
       "| category &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| MR only         |   44 |\n",
       "| Not Significant | 7332 |\n",
       "| TWAS &amp; MR       |   36 |\n",
       "| TWAS only       |   31 |\n",
       "\n"
      ],
      "text/plain": [
       "  category        n   \n",
       "1 MR only           44\n",
       "2 Not Significant 7332\n",
       "3 TWAS & MR         36\n",
       "4 TWAS only         31"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df |> count(category)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "46571a3d-98d3-4def-953b-ea9879608e28",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# dir.create('data')\n",
    "saveRDS(list(df = df, unadj_cutoff = unadj_cutoff, axis_df= axis_df), 'data/TWAS_manhattan_plot_data.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0456b84a-5e08-423a-be45-5b64d32e6861",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.3"
  },
  "sos": {
   "kernels": [
    [
     "R",
     "ir",
     "R",
     "",
     ""
    ]
   ],
   "version": ""
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
