{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "7d50c71c-d2a2-4677-a763-88c6bdd3123a",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "# TWAS\n",
    "a) Manhattan plot of glyco-TWAS results (unadjusted data); annotate points to indicate MR significance\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8bbe5663-26c0-4106-ad47-8d95bc26233c",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "## prepare data\n",
    "TWAS and MR results generaated by pipelines were combined using the following commands\n",
    "```\n",
    "python /mnt/lustre/home/rl3328/rl3328/motor_qtl/combine_twas.py /mnt/lustre/lab/ctcn/hklein/motor_qtl_project/unadjusted_wp_2020/susie_twas/noQC_nonimputed_result/twas --type both --output ROSMAP_eQTL_wp_unadjusted\n",
    "```\n",
    "### Input\n",
    "Input files are all on the new AWS-based HPC\n",
    "* Combined TWAS file:`/mnt/lustre/lab/ctcn/hklein/motor_qtl_project/wp_2020/susie_twas/noQC_nonimputed_result/twas/ROSMAP_eQTL_wp_unadjusted.combined_twas.tsv.gz`\n",
    "* Combined MR file:`/mnt/lustre/lab/ctcn/hklein/motor_qtl_project/unadjusted_wp_2020/susie_twas/noQC_nonimputed_result/twas/ROSMAP_eQTL_wp_unadjusted.combined_mr.tsv.gz`\n",
    "* reference file for gene annotation: `/mnt/lustre/home/rl3328/rl3328/resource/Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.region_list.2025`\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01773a49",
   "metadata": {},
   "source": [
    "### Landscape\n",
    "\n",
    "* The TWAS results show substantial variation in the number of available genes across contexts. Monocyte, DLPFC, and AC have notably fewer genes available for analysis. In particular, the DLPFC bulk eQTL dataset includes only 58 genes, which likely explains why no genes reach significance under the most accurate predictive model for DLPFC.\n",
    "* Consistently, the MR analysis also yields no significant genes for monocyte, DLPFC, or AC.\n",
    "\n",
    "* **Given these limitations, it may be more appropriate to combine bulk and single-nucleus (sn) data for DLPFC in downstream analyses.**\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09fa2c4a",
   "metadata": {},
   "source": [
    "#### TWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "d55a570c-74e3-4642-86c7-64fa97d3cf82",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "library(data.table)\n",
    "library(tidyverse)\n",
    "twas <- fread('/mnt/lustre/lab/ctcn/hklein/motor_qtl_project/unadjusted_wp_2020/susie_twas/noQC_nonimputed_result/twas/ROSMAP_eQTL_wp_unadjusted.combined_twas.tsv.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "5ce0b8c3",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>molecular_id</th><th scope=col>TSS</th><th scope=col>start</th><th scope=col>end</th><th scope=col>context</th><th scope=col>gwas_study</th><th scope=col>method</th><th scope=col>is_imputable</th><th scope=col>is_selected_method</th><th scope=col>rsq_cv</th><th scope=col>pval_cv</th><th scope=col>twas_z</th><th scope=col>twas_pval</th><th scope=col>type</th><th scope=col>block</th><th scope=col>method_selected_original</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>Inh_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>enet </td><td>TRUE</td><td>FALSE</td><td>-0.0002636756</td><td>3.460743e-01</td><td>-0.8321645</td><td>0.40531612</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>Inh_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>lasso</td><td>TRUE</td><td>FALSE</td><td>-0.0008220066</td><td>4.181955e-01</td><td>-0.9241237</td><td>0.35542192</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>Inh_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>mrash</td><td>TRUE</td><td>FALSE</td><td> 0.0074837805</td><td>4.222085e-02</td><td>-2.4786840</td><td>0.01318681</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>Inh_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>susie</td><td>TRUE</td><td> TRUE</td><td> 0.0157468862</td><td>5.809734e-03</td><td>-0.6523170</td><td>0.51419672</td><td>eQTL</td><td>chr10_100331627_104378781</td><td> TRUE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>PCC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>enet </td><td>TRUE</td><td>FALSE</td><td> 0.1176354683</td><td>7.716102e-14</td><td>-0.2702251</td><td>0.78698710</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "\t<tr><td>10</td><td>ENSG00000055950</td><td>100987515</td><td>99320000</td><td>102120000</td><td>PCC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>lasso</td><td>TRUE</td><td>FALSE</td><td> 0.1157138583</td><td>1.253604e-13</td><td>-0.1913938</td><td>0.84821705</td><td>eQTL</td><td>chr10_100331627_104378781</td><td>FALSE</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " chr & molecular\\_id & TSS & start & end & context & gwas\\_study & method & is\\_imputable & is\\_selected\\_method & rsq\\_cv & pval\\_cv & twas\\_z & twas\\_pval & type & block & method\\_selected\\_original & region & study\\_context & source\\_file\\\\\n",
       " <int> & <chr> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <lgl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & Inh\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & enet  & TRUE & FALSE & -0.0002636756 & 3.460743e-01 & -0.8321645 & 0.40531612 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & Inh\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & lasso & TRUE & FALSE & -0.0008220066 & 4.181955e-01 & -0.9241237 & 0.35542192 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & Inh\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & mrash & TRUE & FALSE &  0.0074837805 & 4.222085e-02 & -2.4786840 & 0.01318681 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & Inh\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & susie & TRUE &  TRUE &  0.0157468862 & 5.809734e-03 & -0.6523170 & 0.51419672 & eQTL & chr10\\_100331627\\_104378781 &  TRUE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & PCC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & enet  & TRUE & FALSE &  0.1176354683 & 7.716102e-14 & -0.2702251 & 0.78698710 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\t 10 & ENSG00000055950 & 100987515 & 99320000 & 102120000 & PCC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & lasso & TRUE & FALSE &  0.1157138583 & 1.253604e-13 & -0.1913938 & 0.84821705 & eQTL & chr10\\_100331627\\_104378781 & FALSE & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.twas.tsv.gz\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 20\n",
       "\n",
       "| chr &lt;int&gt; | molecular_id &lt;chr&gt; | TSS &lt;int&gt; | start &lt;int&gt; | end &lt;int&gt; | context &lt;chr&gt; | gwas_study &lt;chr&gt; | method &lt;chr&gt; | is_imputable &lt;lgl&gt; | is_selected_method &lt;lgl&gt; | rsq_cv &lt;dbl&gt; | pval_cv &lt;dbl&gt; | twas_z &lt;dbl&gt; | twas_pval &lt;dbl&gt; | type &lt;chr&gt; | block &lt;chr&gt; | method_selected_original &lt;lgl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | Inh_DeJager_eQTL | unadjusted_wp_2020 | enet  | TRUE | FALSE | -0.0002636756 | 3.460743e-01 | -0.8321645 | 0.40531612 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | Inh_DeJager_eQTL | unadjusted_wp_2020 | lasso | TRUE | FALSE | -0.0008220066 | 4.181955e-01 | -0.9241237 | 0.35542192 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | Inh_DeJager_eQTL | unadjusted_wp_2020 | mrash | TRUE | FALSE |  0.0074837805 | 4.222085e-02 | -2.4786840 | 0.01318681 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | Inh_DeJager_eQTL | unadjusted_wp_2020 | susie | TRUE |  TRUE |  0.0157468862 | 5.809734e-03 | -0.6523170 | 0.51419672 | eQTL | chr10_100331627_104378781 |  TRUE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | PCC_DeJager_eQTL | unadjusted_wp_2020 | enet  | TRUE | FALSE |  0.1176354683 | 7.716102e-14 | -0.2702251 | 0.78698710 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "| 10 | ENSG00000055950 | 100987515 | 99320000 | 102120000 | PCC_DeJager_eQTL | unadjusted_wp_2020 | lasso | TRUE | FALSE |  0.1157138583 | 1.253604e-13 | -0.1913938 | 0.84821705 | eQTL | chr10_100331627_104378781 | FALSE | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz |\n",
       "\n"
      ],
      "text/plain": [
       "  chr molecular_id    TSS       start    end       context         \n",
       "1 10  ENSG00000055950 100987515 99320000 102120000 Inh_DeJager_eQTL\n",
       "2 10  ENSG00000055950 100987515 99320000 102120000 Inh_DeJager_eQTL\n",
       "3 10  ENSG00000055950 100987515 99320000 102120000 Inh_DeJager_eQTL\n",
       "4 10  ENSG00000055950 100987515 99320000 102120000 Inh_DeJager_eQTL\n",
       "5 10  ENSG00000055950 100987515 99320000 102120000 PCC_DeJager_eQTL\n",
       "6 10  ENSG00000055950 100987515 99320000 102120000 PCC_DeJager_eQTL\n",
       "  gwas_study         method is_imputable is_selected_method rsq_cv       \n",
       "1 unadjusted_wp_2020 enet   TRUE         FALSE              -0.0002636756\n",
       "2 unadjusted_wp_2020 lasso  TRUE         FALSE              -0.0008220066\n",
       "3 unadjusted_wp_2020 mrash  TRUE         FALSE               0.0074837805\n",
       "4 unadjusted_wp_2020 susie  TRUE          TRUE               0.0157468862\n",
       "5 unadjusted_wp_2020 enet   TRUE         FALSE               0.1176354683\n",
       "6 unadjusted_wp_2020 lasso  TRUE         FALSE               0.1157138583\n",
       "  pval_cv      twas_z     twas_pval  type block                    \n",
       "1 3.460743e-01 -0.8321645 0.40531612 eQTL chr10_100331627_104378781\n",
       "2 4.181955e-01 -0.9241237 0.35542192 eQTL chr10_100331627_104378781\n",
       "3 4.222085e-02 -2.4786840 0.01318681 eQTL chr10_100331627_104378781\n",
       "4 5.809734e-03 -0.6523170 0.51419672 eQTL chr10_100331627_104378781\n",
       "5 7.716102e-14 -0.2702251 0.78698710 eQTL chr10_100331627_104378781\n",
       "6 1.253604e-13 -0.1913938 0.84821705 eQTL chr10_100331627_104378781\n",
       "  method_selected_original region                    study_context \n",
       "1 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "2 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "3 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "4  TRUE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "5 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "6 FALSE                    chr10_100331627_104378781 ROSMAP_eQTL_wp\n",
       "  source_file                                                    \n",
       "1 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "2 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "3 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "4 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "5 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz\n",
       "6 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.twas.tsv.gz"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>168334</li><li>20</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 168334\n",
       "\\item 20\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 168334\n",
       "2. 20\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 168334     20"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(twas)\n",
    "dim(twas)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "668b29f4-54df-4efb-b3bd-1f3417e11108",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Inh_DeJager_eQTL'</li><li>'PCC_DeJager_eQTL'</li><li>'Mic_DeJager_eQTL'</li><li>'Exc_DeJager_eQTL'</li><li>'Oli_DeJager_eQTL'</li><li>'Ast_DeJager_eQTL'</li><li>'OPC_DeJager_eQTL'</li><li>'AC_DeJager_eQTL'</li><li>'DLPFC_DeJager_eQTL'</li><li>'monocyte_ROSMAP_eQTL'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Inh\\_DeJager\\_eQTL'\n",
       "\\item 'PCC\\_DeJager\\_eQTL'\n",
       "\\item 'Mic\\_DeJager\\_eQTL'\n",
       "\\item 'Exc\\_DeJager\\_eQTL'\n",
       "\\item 'Oli\\_DeJager\\_eQTL'\n",
       "\\item 'Ast\\_DeJager\\_eQTL'\n",
       "\\item 'OPC\\_DeJager\\_eQTL'\n",
       "\\item 'AC\\_DeJager\\_eQTL'\n",
       "\\item 'DLPFC\\_DeJager\\_eQTL'\n",
       "\\item 'monocyte\\_ROSMAP\\_eQTL'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Inh_DeJager_eQTL'\n",
       "2. 'PCC_DeJager_eQTL'\n",
       "3. 'Mic_DeJager_eQTL'\n",
       "4. 'Exc_DeJager_eQTL'\n",
       "5. 'Oli_DeJager_eQTL'\n",
       "6. 'Ast_DeJager_eQTL'\n",
       "7. 'OPC_DeJager_eQTL'\n",
       "8. 'AC_DeJager_eQTL'\n",
       "9. 'DLPFC_DeJager_eQTL'\n",
       "10. 'monocyte_ROSMAP_eQTL'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Inh_DeJager_eQTL\"     \"PCC_DeJager_eQTL\"     \"Mic_DeJager_eQTL\"    \n",
       " [4] \"Exc_DeJager_eQTL\"     \"Oli_DeJager_eQTL\"     \"Ast_DeJager_eQTL\"    \n",
       " [7] \"OPC_DeJager_eQTL\"     \"AC_DeJager_eQTL\"      \"DLPFC_DeJager_eQTL\"  \n",
       "[10] \"monocyte_ROSMAP_eQTL\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "13872"
      ],
      "text/latex": [
       "13872"
      ],
      "text/markdown": [
       "13872"
      ],
      "text/plain": [
       "[1] 13872"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## TWAS analysis were performed across 10 contexts and 13872 genes\n",
    "twas %>% pull(context) |> unique()#%>%length\n",
    "\n",
    "twas %>% pull(molecular_id)%>%unique()%>%length\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "7d558397",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "10778"
      ],
      "text/latex": [
       "10778"
      ],
      "text/markdown": [
       "10778"
      ],
      "text/plain": [
       "[1] 10778"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "13872-60-3031-3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ef0fe8a",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 10 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>context</th><th scope=col>gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>AC_DeJager_eQTL     </td><td>   60</td></tr>\n",
       "\t<tr><td>Ast_DeJager_eQTL    </td><td> 5854</td></tr>\n",
       "\t<tr><td>DLPFC_DeJager_eQTL  </td><td>   57</td></tr>\n",
       "\t<tr><td>Exc_DeJager_eQTL    </td><td> 8187</td></tr>\n",
       "\t<tr><td>Inh_DeJager_eQTL    </td><td> 6377</td></tr>\n",
       "\t<tr><td>Mic_DeJager_eQTL    </td><td> 2729</td></tr>\n",
       "\t<tr><td>OPC_DeJager_eQTL    </td><td> 3031</td></tr>\n",
       "\t<tr><td>Oli_DeJager_eQTL    </td><td> 5392</td></tr>\n",
       "\t<tr><td>PCC_DeJager_eQTL    </td><td>10376</td></tr>\n",
       "\t<tr><td>monocyte_ROSMAP_eQTL</td><td>    3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 10 × 2\n",
       "\\begin{tabular}{ll}\n",
       " context & gene\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t AC\\_DeJager\\_eQTL      &    60\\\\\n",
       "\t Ast\\_DeJager\\_eQTL     &  5854\\\\\n",
       "\t DLPFC\\_DeJager\\_eQTL   &    57\\\\\n",
       "\t Exc\\_DeJager\\_eQTL     &  8187\\\\\n",
       "\t Inh\\_DeJager\\_eQTL     &  6377\\\\\n",
       "\t Mic\\_DeJager\\_eQTL     &  2729\\\\\n",
       "\t OPC\\_DeJager\\_eQTL     &  3031\\\\\n",
       "\t Oli\\_DeJager\\_eQTL     &  5392\\\\\n",
       "\t PCC\\_DeJager\\_eQTL     & 10376\\\\\n",
       "\t monocyte\\_ROSMAP\\_eQTL &     3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 10 × 2\n",
       "\n",
       "| context &lt;chr&gt; | gene &lt;int&gt; |\n",
       "|---|---|\n",
       "| AC_DeJager_eQTL      |    60 |\n",
       "| Ast_DeJager_eQTL     |  5854 |\n",
       "| DLPFC_DeJager_eQTL   |    57 |\n",
       "| Exc_DeJager_eQTL     |  8187 |\n",
       "| Inh_DeJager_eQTL     |  6377 |\n",
       "| Mic_DeJager_eQTL     |  2729 |\n",
       "| OPC_DeJager_eQTL     |  3031 |\n",
       "| Oli_DeJager_eQTL     |  5392 |\n",
       "| PCC_DeJager_eQTL     | 10376 |\n",
       "| monocyte_ROSMAP_eQTL |     3 |\n",
       "\n"
      ],
      "text/plain": [
       "   context              gene \n",
       "1  AC_DeJager_eQTL         60\n",
       "2  Ast_DeJager_eQTL      5854\n",
       "3  DLPFC_DeJager_eQTL      57\n",
       "4  Exc_DeJager_eQTL      8187\n",
       "5  Inh_DeJager_eQTL      6377\n",
       "6  Mic_DeJager_eQTL      2729\n",
       "7  OPC_DeJager_eQTL      3031\n",
       "8  Oli_DeJager_eQTL      5392\n",
       "9  PCC_DeJager_eQTL     10376\n",
       "10 monocyte_ROSMAP_eQTL     3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## 13872 total genes, 10778 are from DLPFC\n",
    "twas |> group_by(context) |> summarize(gene = n_distinct(molecular_id))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "65d8afca-6867-416a-8e9f-e39bf07064b5",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "4325"
      ],
      "text/latex": [
       "4325"
      ],
      "text/markdown": [
       "4325"
      ],
      "text/plain": [
       "[1] 4325"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "twas_res_sig <- twas %>% filter(is_selected_method==\"TRUE\")%>%filter(twas_pval<=0.05)#%>%pull(context)%>%unique()\n",
    "twas_res_sig|> pull(molecular_id)%>%unique()%>%length\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "a06ae9b6-477c-4b77-97c9-8763ab5e0bb8",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>232</li><li>20</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 232\n",
       "\\item 20\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 232\n",
       "2. 20\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 232  20"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "twas_DLPFC = twas |> filter(context == 'DLPFC_DeJager_eQTL')\n",
    "dim(twas_DLPFC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e3a9a9d5",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 0 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>molecular_id</th><th scope=col>TSS</th><th scope=col>start</th><th scope=col>end</th><th scope=col>context</th><th scope=col>gwas_study</th><th scope=col>method</th><th scope=col>is_imputable</th><th scope=col>is_selected_method</th><th scope=col>rsq_cv</th><th scope=col>pval_cv</th><th scope=col>twas_z</th><th scope=col>twas_pval</th><th scope=col>type</th><th scope=col>block</th><th scope=col>method_selected_original</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 0 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " chr & molecular\\_id & TSS & start & end & context & gwas\\_study & method & is\\_imputable & is\\_selected\\_method & rsq\\_cv & pval\\_cv & twas\\_z & twas\\_pval & type & block & method\\_selected\\_original & region & study\\_context & source\\_file\\\\\n",
       " <int> & <chr> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <lgl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 0 × 20\n",
       "\n",
       "| chr &lt;int&gt; | molecular_id &lt;chr&gt; | TSS &lt;int&gt; | start &lt;int&gt; | end &lt;int&gt; | context &lt;chr&gt; | gwas_study &lt;chr&gt; | method &lt;chr&gt; | is_imputable &lt;lgl&gt; | is_selected_method &lt;lgl&gt; | rsq_cv &lt;dbl&gt; | pval_cv &lt;dbl&gt; | twas_z &lt;dbl&gt; | twas_pval &lt;dbl&gt; | type &lt;chr&gt; | block &lt;chr&gt; | method_selected_original &lt;lgl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     chr molecular_id TSS start end context gwas_study method is_imputable\n",
       "     is_selected_method rsq_cv pval_cv twas_z twas_pval type block\n",
       "     method_selected_original region study_context source_file"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 20</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>molecular_id</th><th scope=col>TSS</th><th scope=col>start</th><th scope=col>end</th><th scope=col>context</th><th scope=col>gwas_study</th><th scope=col>method</th><th scope=col>is_imputable</th><th scope=col>is_selected_method</th><th scope=col>rsq_cv</th><th scope=col>pval_cv</th><th scope=col>twas_z</th><th scope=col>twas_pval</th><th scope=col>type</th><th scope=col>block</th><th scope=col>method_selected_original</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>3</td><td>ENSG00000044524</td><td>89107621</td><td>86600000</td><td>93720000</td><td>DLPFC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>mrash</td><td>TRUE</td><td>FALSE</td><td>0.04285215</td><td>2.935038e-09</td><td>-5.764569</td><td>8.186698e-09</td><td>eQTL</td><td>chr3_88209300_94537986</td><td>FALSE</td><td>chr3_88209300_94537986</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr3_88209300_94537986.twas.tsv.gz</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 20\n",
       "\\begin{tabular}{llllllllllllllllllll}\n",
       " chr & molecular\\_id & TSS & start & end & context & gwas\\_study & method & is\\_imputable & is\\_selected\\_method & rsq\\_cv & pval\\_cv & twas\\_z & twas\\_pval & type & block & method\\_selected\\_original & region & study\\_context & source\\_file\\\\\n",
       " <int> & <chr> & <int> & <int> & <int> & <chr> & <chr> & <chr> & <lgl> & <lgl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <lgl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 3 & ENSG00000044524 & 89107621 & 86600000 & 93720000 & DLPFC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & mrash & TRUE & FALSE & 0.04285215 & 2.935038e-09 & -5.764569 & 8.186698e-09 & eQTL & chr3\\_88209300\\_94537986 & FALSE & chr3\\_88209300\\_94537986 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr3\\_88209300\\_94537986.twas.tsv.gz\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 20\n",
       "\n",
       "| chr &lt;int&gt; | molecular_id &lt;chr&gt; | TSS &lt;int&gt; | start &lt;int&gt; | end &lt;int&gt; | context &lt;chr&gt; | gwas_study &lt;chr&gt; | method &lt;chr&gt; | is_imputable &lt;lgl&gt; | is_selected_method &lt;lgl&gt; | rsq_cv &lt;dbl&gt; | pval_cv &lt;dbl&gt; | twas_z &lt;dbl&gt; | twas_pval &lt;dbl&gt; | type &lt;chr&gt; | block &lt;chr&gt; | method_selected_original &lt;lgl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 3 | ENSG00000044524 | 89107621 | 86600000 | 93720000 | DLPFC_DeJager_eQTL | unadjusted_wp_2020 | mrash | TRUE | FALSE | 0.04285215 | 2.935038e-09 | -5.764569 | 8.186698e-09 | eQTL | chr3_88209300_94537986 | FALSE | chr3_88209300_94537986 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr3_88209300_94537986.twas.tsv.gz |\n",
       "\n"
      ],
      "text/plain": [
       "  chr molecular_id    TSS      start    end      context           \n",
       "1 3   ENSG00000044524 89107621 86600000 93720000 DLPFC_DeJager_eQTL\n",
       "  gwas_study         method is_imputable is_selected_method rsq_cv    \n",
       "1 unadjusted_wp_2020 mrash  TRUE         FALSE              0.04285215\n",
       "  pval_cv      twas_z    twas_pval    type block                 \n",
       "1 2.935038e-09 -5.764569 8.186698e-09 eQTL chr3_88209300_94537986\n",
       "  method_selected_original region                 study_context \n",
       "1 FALSE                    chr3_88209300_94537986 ROSMAP_eQTL_wp\n",
       "  source_file                                                 \n",
       "1 ROSMAP_eQTL_wp_unadjusted.chr3_88209300_94537986.twas.tsv.gz"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## Best model-0 gene significant after bonferroni correction\n",
    "twas |> filter(is_selected_method, twas_pval < unadj_cutoff, str_detect(context, \"DLPFC\"))\n",
    "## All models-1 gene(EPHA3) significant after bonferroni correction\n",
    "twas |> filter(twas_pval < unadj_cutoff, str_detect(context, \"DLPFC\"))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a64c2d9f",
   "metadata": {},
   "source": [
    "#### MR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "89201606-c5eb-4bfb-9aaf-b026aec56c0f",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "mr <- fread('/mnt/lustre/lab/ctcn/hklein/motor_qtl_project/unadjusted_wp_2020/susie_twas/noQC_nonimputed_result/twas/ROSMAP_eQTL_wp_unadjusted.combined_mr.tsv.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "6303aa4a-76c7-4416-82d9-6fe4ec293aa1",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "mr_sig_unadj <- mr %>% .[\n",
    "  meta_pval < 0.05 / .N &          # Bonferroni correction\n",
    "  cpip >= 0.7 &                    # strong causal evidence\n",
    "  num_CS >= 1 &                    # as quested by Hans and Gao on Oct 24, 2025 to get a loose version\n",
    "  Q_pval > 0.01 &                  # no heterogeneity\n",
    "  I2 < 0.4                         # low heterogeneity\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "da1af503",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 0 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Q</th><th scope=col>Q_pval</th><th scope=col>I2</th><th scope=col>context</th><th scope=col>cpip</th><th scope=col>gene_name</th><th scope=col>gwas_study</th><th scope=col>meta_eff</th><th scope=col>meta_pval</th><th scope=col>num_CS</th><th scope=col>num_IV</th><th scope=col>se_meta_eff</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 0 × 15\n",
       "\\begin{tabular}{lllllllllllllll}\n",
       " Q & Q\\_pval & I2 & context & cpip & gene\\_name & gwas\\_study & meta\\_eff & meta\\_pval & num\\_CS & num\\_IV & se\\_meta\\_eff & region & study\\_context & source\\_file\\\\\n",
       " <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <chr> & <chr> & <dbl> & <dbl> & <int> & <int> & <dbl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 0 × 15\n",
       "\n",
       "| Q &lt;dbl&gt; | Q_pval &lt;dbl&gt; | I2 &lt;dbl&gt; | context &lt;chr&gt; | cpip &lt;dbl&gt; | gene_name &lt;chr&gt; | gwas_study &lt;chr&gt; | meta_eff &lt;dbl&gt; | meta_pval &lt;dbl&gt; | num_CS &lt;int&gt; | num_IV &lt;int&gt; | se_meta_eff &lt;dbl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     Q Q_pval I2 context cpip gene_name gwas_study meta_eff meta_pval num_CS\n",
       "     num_IV se_meta_eff region study_context source_file"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mr_sig_unadj |> filter(gene_name == 'ENSG00000044524')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "f3f48a47-880f-4e9b-aa3e-ad5faf4a1e4b",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Q</th><th scope=col>Q_pval</th><th scope=col>I2</th><th scope=col>context</th><th scope=col>cpip</th><th scope=col>gene_name</th><th scope=col>gwas_study</th><th scope=col>meta_eff</th><th scope=col>meta_pval</th><th scope=col>num_CS</th><th scope=col>num_IV</th><th scope=col>se_meta_eff</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1.823</td><td>0.402</td><td>0</td><td>Exc_DeJager_eQTL</td><td>0.983</td><td>ENSG00000138111</td><td>unadjusted_wp_2020</td><td> 0.000</td><td>0</td><td>3</td><td> 21</td><td>0</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz</td></tr>\n",
       "\t<tr><td>0.000</td><td>1.000</td><td>0</td><td>Inh_DeJager_eQTL</td><td>0.956</td><td>ENSG00000138111</td><td>unadjusted_wp_2020</td><td> 0.000</td><td>0</td><td>1</td><td> 16</td><td>0</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz</td></tr>\n",
       "\t<tr><td>0.000</td><td>1.000</td><td>0</td><td>Ast_DeJager_eQTL</td><td>1.012</td><td>ENSG00000171206</td><td>unadjusted_wp_2020</td><td> 0.001</td><td>0</td><td>1</td><td> 59</td><td>0</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz</td></tr>\n",
       "\t<tr><td>0.001</td><td>0.979</td><td>0</td><td>Exc_DeJager_eQTL</td><td>1.026</td><td>ENSG00000171206</td><td>unadjusted_wp_2020</td><td> 0.001</td><td>0</td><td>2</td><td>144</td><td>0</td><td>chr10_100331627_104378781</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz</td></tr>\n",
       "\t<tr><td>0.000</td><td>1.000</td><td>0</td><td>Exc_DeJager_eQTL</td><td>0.960</td><td>ENSG00000078403</td><td>unadjusted_wp_2020</td><td>-0.001</td><td>0</td><td>1</td><td>  8</td><td>0</td><td>chr10_20625737_23053090  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_20625737_23053090.mr_result.tsv.gz  </td></tr>\n",
       "\t<tr><td>0.000</td><td>1.000</td><td>0</td><td>PCC_DeJager_eQTL</td><td>0.995</td><td>ENSG00000148572</td><td>unadjusted_wp_2020</td><td>-0.002</td><td>0</td><td>1</td><td> 92</td><td>0</td><td>chr10_62446953_64035328  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr10_62446953_64035328.mr_result.tsv.gz  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 15\n",
       "\\begin{tabular}{lllllllllllllll}\n",
       " Q & Q\\_pval & I2 & context & cpip & gene\\_name & gwas\\_study & meta\\_eff & meta\\_pval & num\\_CS & num\\_IV & se\\_meta\\_eff & region & study\\_context & source\\_file\\\\\n",
       " <dbl> & <dbl> & <dbl> & <chr> & <dbl> & <chr> & <chr> & <dbl> & <dbl> & <int> & <int> & <dbl> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1.823 & 0.402 & 0 & Exc\\_DeJager\\_eQTL & 0.983 & ENSG00000138111 & unadjusted\\_wp\\_2020 &  0.000 & 0 & 3 &  21 & 0 & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.mr\\_result.tsv.gz\\\\\n",
       "\t 0.000 & 1.000 & 0 & Inh\\_DeJager\\_eQTL & 0.956 & ENSG00000138111 & unadjusted\\_wp\\_2020 &  0.000 & 0 & 1 &  16 & 0 & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.mr\\_result.tsv.gz\\\\\n",
       "\t 0.000 & 1.000 & 0 & Ast\\_DeJager\\_eQTL & 1.012 & ENSG00000171206 & unadjusted\\_wp\\_2020 &  0.001 & 0 & 1 &  59 & 0 & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.mr\\_result.tsv.gz\\\\\n",
       "\t 0.001 & 0.979 & 0 & Exc\\_DeJager\\_eQTL & 1.026 & ENSG00000171206 & unadjusted\\_wp\\_2020 &  0.001 & 0 & 2 & 144 & 0 & chr10\\_100331627\\_104378781 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_100331627\\_104378781.mr\\_result.tsv.gz\\\\\n",
       "\t 0.000 & 1.000 & 0 & Exc\\_DeJager\\_eQTL & 0.960 & ENSG00000078403 & unadjusted\\_wp\\_2020 & -0.001 & 0 & 1 &   8 & 0 & chr10\\_20625737\\_23053090   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_20625737\\_23053090.mr\\_result.tsv.gz  \\\\\n",
       "\t 0.000 & 1.000 & 0 & PCC\\_DeJager\\_eQTL & 0.995 & ENSG00000148572 & unadjusted\\_wp\\_2020 & -0.002 & 0 & 1 &  92 & 0 & chr10\\_62446953\\_64035328   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr10\\_62446953\\_64035328.mr\\_result.tsv.gz  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 15\n",
       "\n",
       "| Q &lt;dbl&gt; | Q_pval &lt;dbl&gt; | I2 &lt;dbl&gt; | context &lt;chr&gt; | cpip &lt;dbl&gt; | gene_name &lt;chr&gt; | gwas_study &lt;chr&gt; | meta_eff &lt;dbl&gt; | meta_pval &lt;dbl&gt; | num_CS &lt;int&gt; | num_IV &lt;int&gt; | se_meta_eff &lt;dbl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1.823 | 0.402 | 0 | Exc_DeJager_eQTL | 0.983 | ENSG00000138111 | unadjusted_wp_2020 |  0.000 | 0 | 3 |  21 | 0 | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz |\n",
       "| 0.000 | 1.000 | 0 | Inh_DeJager_eQTL | 0.956 | ENSG00000138111 | unadjusted_wp_2020 |  0.000 | 0 | 1 |  16 | 0 | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz |\n",
       "| 0.000 | 1.000 | 0 | Ast_DeJager_eQTL | 1.012 | ENSG00000171206 | unadjusted_wp_2020 |  0.001 | 0 | 1 |  59 | 0 | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz |\n",
       "| 0.001 | 0.979 | 0 | Exc_DeJager_eQTL | 1.026 | ENSG00000171206 | unadjusted_wp_2020 |  0.001 | 0 | 2 | 144 | 0 | chr10_100331627_104378781 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz |\n",
       "| 0.000 | 1.000 | 0 | Exc_DeJager_eQTL | 0.960 | ENSG00000078403 | unadjusted_wp_2020 | -0.001 | 0 | 1 |   8 | 0 | chr10_20625737_23053090   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_20625737_23053090.mr_result.tsv.gz   |\n",
       "| 0.000 | 1.000 | 0 | PCC_DeJager_eQTL | 0.995 | ENSG00000148572 | unadjusted_wp_2020 | -0.002 | 0 | 1 |  92 | 0 | chr10_62446953_64035328   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr10_62446953_64035328.mr_result.tsv.gz   |\n",
       "\n"
      ],
      "text/plain": [
       "  Q     Q_pval I2 context          cpip  gene_name       gwas_study        \n",
       "1 1.823 0.402  0  Exc_DeJager_eQTL 0.983 ENSG00000138111 unadjusted_wp_2020\n",
       "2 0.000 1.000  0  Inh_DeJager_eQTL 0.956 ENSG00000138111 unadjusted_wp_2020\n",
       "3 0.000 1.000  0  Ast_DeJager_eQTL 1.012 ENSG00000171206 unadjusted_wp_2020\n",
       "4 0.001 0.979  0  Exc_DeJager_eQTL 1.026 ENSG00000171206 unadjusted_wp_2020\n",
       "5 0.000 1.000  0  Exc_DeJager_eQTL 0.960 ENSG00000078403 unadjusted_wp_2020\n",
       "6 0.000 1.000  0  PCC_DeJager_eQTL 0.995 ENSG00000148572 unadjusted_wp_2020\n",
       "  meta_eff meta_pval num_CS num_IV se_meta_eff region                   \n",
       "1  0.000   0         3       21    0           chr10_100331627_104378781\n",
       "2  0.000   0         1       16    0           chr10_100331627_104378781\n",
       "3  0.001   0         1       59    0           chr10_100331627_104378781\n",
       "4  0.001   0         2      144    0           chr10_100331627_104378781\n",
       "5 -0.001   0         1        8    0           chr10_20625737_23053090  \n",
       "6 -0.002   0         1       92    0           chr10_62446953_64035328  \n",
       "  study_context \n",
       "1 ROSMAP_eQTL_wp\n",
       "2 ROSMAP_eQTL_wp\n",
       "3 ROSMAP_eQTL_wp\n",
       "4 ROSMAP_eQTL_wp\n",
       "5 ROSMAP_eQTL_wp\n",
       "6 ROSMAP_eQTL_wp\n",
       "  source_file                                                         \n",
       "1 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz\n",
       "2 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz\n",
       "3 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz\n",
       "4 ROSMAP_eQTL_wp_unadjusted.chr10_100331627_104378781.mr_result.tsv.gz\n",
       "5 ROSMAP_eQTL_wp_unadjusted.chr10_20625737_23053090.mr_result.tsv.gz  \n",
       "6 ROSMAP_eQTL_wp_unadjusted.chr10_62446953_64035328.mr_result.tsv.gz  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>206</li><li>15</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 206\n",
       "\\item 15\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 206\n",
       "2. 15\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 206  15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(mr_sig_unadj)\n",
    "dim(mr_sig_unadj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "37214743",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 7 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>context</th><th scope=col>gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Ast_DeJager_eQTL</td><td>21</td></tr>\n",
       "\t<tr><td>Exc_DeJager_eQTL</td><td>41</td></tr>\n",
       "\t<tr><td>Inh_DeJager_eQTL</td><td>29</td></tr>\n",
       "\t<tr><td>Mic_DeJager_eQTL</td><td> 6</td></tr>\n",
       "\t<tr><td>OPC_DeJager_eQTL</td><td> 7</td></tr>\n",
       "\t<tr><td>Oli_DeJager_eQTL</td><td>27</td></tr>\n",
       "\t<tr><td>PCC_DeJager_eQTL</td><td>75</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 7 × 2\n",
       "\\begin{tabular}{ll}\n",
       " context & gene\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Ast\\_DeJager\\_eQTL & 21\\\\\n",
       "\t Exc\\_DeJager\\_eQTL & 41\\\\\n",
       "\t Inh\\_DeJager\\_eQTL & 29\\\\\n",
       "\t Mic\\_DeJager\\_eQTL &  6\\\\\n",
       "\t OPC\\_DeJager\\_eQTL &  7\\\\\n",
       "\t Oli\\_DeJager\\_eQTL & 27\\\\\n",
       "\t PCC\\_DeJager\\_eQTL & 75\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 7 × 2\n",
       "\n",
       "| context &lt;chr&gt; | gene &lt;int&gt; |\n",
       "|---|---|\n",
       "| Ast_DeJager_eQTL | 21 |\n",
       "| Exc_DeJager_eQTL | 41 |\n",
       "| Inh_DeJager_eQTL | 29 |\n",
       "| Mic_DeJager_eQTL |  6 |\n",
       "| OPC_DeJager_eQTL |  7 |\n",
       "| Oli_DeJager_eQTL | 27 |\n",
       "| PCC_DeJager_eQTL | 75 |\n",
       "\n"
      ],
      "text/plain": [
       "  context          gene\n",
       "1 Ast_DeJager_eQTL 21  \n",
       "2 Exc_DeJager_eQTL 41  \n",
       "3 Inh_DeJager_eQTL 29  \n",
       "4 Mic_DeJager_eQTL  6  \n",
       "5 OPC_DeJager_eQTL  7  \n",
       "6 Oli_DeJager_eQTL 27  \n",
       "7 PCC_DeJager_eQTL 75  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mr_sig_unadj |> group_by(context) |> summarize(gene = n_distinct(gene_name))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c45de04b-ce4d-467a-b981-a1ec5b4511eb",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "### summarize significant TWAS & MR genes\n",
    "\n",
    "* Significant TWAS associations were defined using two criteria: (i) the most accurate prediction model yielded a TWAS p-value below the Bonferroni-corrected threshold, or (ii) at least half of the prediction models produced p-values below this threshold.\n",
    "* Among the 13,872 total genes, 10,778 are from DLPFC. Overall, 126 genes show significant TWAS associations, of which 80 are significant in DLPFC tissues\n",
    "\n",
    "* Mendelian randomization (MR) associations were considered significant if they met all of the following criteria: a meta-analysis p-value passing Bonferroni correction (meta-analysis p < 0.05 divided by the number of tested genes), strong causal evidence as indicated by a causal posterior inclusion probability (CPIP) ≥ 0.7, at least one credible set (num_CS ≥ 1), and no evidence of substantial heterogeneity (Cochran’s Q test p > 0.01 and I² < 0.4).\n",
    "* 206 genes show significant MR associations, of which 131 are significant in DLPFC tissues."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "51741b66-ce32-4b11-8989-71f41c6a4f0f",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "42066"
      ],
      "text/latex": [
       "42066"
      ],
      "text/markdown": [
       "42066"
      ],
      "text/plain": [
       "[1] 42066"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "n_imputable_genes <- twas %>%\n",
    "  filter(is_imputable) %>%                  # keep only imputable rows\n",
    "  distinct(molecular_id, context, gwas_study) %>%         # count unique gene-context pairs\n",
    "  nrow()\n",
    "n_imputable_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "607cbcd1-b856-4edb-95c3-fe521a8ff7a9",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "1.18860837731184e-06"
      ],
      "text/latex": [
       "1.18860837731184e-06"
      ],
      "text/markdown": [
       "1.18860837731184e-06"
      ],
      "text/plain": [
       "[1] 1.188608e-06"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Compute cutoff using Bonferroni-style correction\n",
    "unadj_cutoff <- 0.05 / n_imputable_genes\n",
    "unadj_cutoff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "e78ef1f1-b8fe-42e8-973f-ad843b7357ed",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[22mJoining with `by = join_by(molecular_id, context, gwas_study)`\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>126</li><li>25</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 126\n",
       "\\item 25\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 126\n",
       "2. 25\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 126  25"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "twas_unadj_sig <- twas %>%\n",
    "  mutate(sig = twas_pval < unadj_cutoff) %>%\n",
    "  group_by(molecular_id, context, gwas_study) %>%\n",
    "  summarise(\n",
    "    n_methods = n(),\n",
    "    n_sig = sum(sig, na.rm = T),\n",
    "    best_sig = any(sig & is_selected_method, na.rm = T),\n",
    "    .groups = \"drop\"\n",
    "  ) %>%\n",
    "  mutate(pass = (n_sig / n_methods >= 0.5) | best_sig) %>%\n",
    "  mutate(pass_strict = (n_sig / n_methods >= 0.5)) %>%\n",
    "  filter(pass) %>% left_join(twas) %>% arrange(twas_pval) %>% distinct(molecular_id, .keep_all = T)\n",
    "twas_unadj_sig %>% dim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "37bb6225",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "80"
      ],
      "text/latex": [
       "80"
      ],
      "text/markdown": [
       "80"
      ],
      "text/plain": [
       "[1] 80"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "twas_unadj_sig |> filter(!context %in% c(\"monocyte_ROSMAP_eQTL\", \"PCC_DeJager_eQTL\", \"AC_DeJager_eQTL\")) |> nrow()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "193c534f-c425-49a6-9b1d-97e364f6b9da",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "gene_ref <- fread('/mnt/lustre/home/rl3328/rl3328/resource/Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.region_list.2025')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "dd79014c-34fb-45ff-b051-bdf18c61ce38",
   "metadata": {
    "kernel": "R",
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# twas_unadj_sig |> left_join(gene_ref, by = c(\"molecular_id\" = \"gene_id\")) |> select(molecular_id, gene_name, everything())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33c9361d-3ab3-42a1-8cfd-2a1e2a5ae97f",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "## Figure data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "d2409cc8-d679-4919-8255-325ac354695f",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“\u001b[1m\u001b[22mThere were 211 warnings in `summarise()`.\n",
      "The first warning was:\n",
      "\u001b[1m\u001b[22m\u001b[36mℹ\u001b[39m In argument: `best_method_pval = min(twas_pval[is_selected_method], na.rm =\n",
      "  TRUE)`.\n",
      "\u001b[36mℹ\u001b[39m In group 202: `molecular_id = \"ENSG00000005189\"`, `context =\n",
      "  \"Exc_DeJager_eQTL\"`, `gwas_study = \"unadjusted_wp_2020\"`.\n",
      "Caused by warning in `min()`:\n",
      "\u001b[33m!\u001b[39m no non-missing arguments to min; returning Inf\n",
      "\u001b[1m\u001b[22m\u001b[36mℹ\u001b[39m Run `dplyr::last_dplyr_warnings()` to see the 210 remaining warnings.”\n"
     ]
    }
   ],
   "source": [
    "df <- twas %>%\n",
    "  mutate(sig = twas_pval < unadj_cutoff) %>%\n",
    "  group_by(molecular_id, context, gwas_study) %>%\n",
    "  summarise(\n",
    "    n_methods = n(),\n",
    "    n_sig = sum(sig, na.rm = TRUE),\n",
    "    best_pval = min(twas_pval, na.rm = TRUE),\n",
    "    best_method_pval = min(twas_pval[is_selected_method], na.rm = TRUE),\n",
    "    best_method_selected = any(sig & is_selected_method, na.rm = TRUE),\n",
    "    .groups = \"drop\"\n",
    "  ) %>%\n",
    "  mutate(\n",
    "    pass = (n_sig / n_methods >= 0.5) | best_method_selected, # pass defined by two threshold, best or half methods satisfied\n",
    "    final_pval = ifelse(n_sig / n_methods >= 0.5, best_pval, best_method_pval),\n",
    "    # fallback: if no selected method existed (Inf), use best_pval\n",
    "    final_pval = ifelse(is.infinite(final_pval), best_pval, final_pval)\n",
    "  ) %>%\n",
    "  left_join(twas, by = c(\"molecular_id\", \"context\", \"gwas_study\")) %>%\n",
    "  filter(twas_pval == final_pval) %>%\n",
    "  distinct(molecular_id, .keep_all = TRUE) %>%\n",
    "  mutate(\n",
    "    logp = -log10(twas_pval) * sign(twas_z),\n",
    "    pos = (start + end) / 2\n",
    "  ) %>%\n",
    "  merge(gene_ref, by.x = 'molecular_id', by.y = 'gene_id')\n",
    "\n",
    "# Order chromosomes\n",
    "df$chr <- factor(df$chr, levels = as.character(1:22))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "05d4225a",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>13872</li><li>33</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 13872\n",
       "\\item 33\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 13872\n",
       "2. 33\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 13872    33"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "bc6550ab",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 33</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>molecular_id</th><th scope=col>context</th><th scope=col>gwas_study</th><th scope=col>n_methods</th><th scope=col>n_sig</th><th scope=col>best_pval</th><th scope=col>best_method_pval</th><th scope=col>best_method_selected</th><th scope=col>pass</th><th scope=col>final_pval</th><th scope=col>⋯</th><th scope=col>method_selected_original</th><th scope=col>region</th><th scope=col>study_context</th><th scope=col>source_file</th><th scope=col>logp</th><th scope=col>pos</th><th scope=col>#chr</th><th scope=col>TSS.y</th><th scope=col>TES</th><th scope=col>gene_name</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ENSG00000000419</td><td>Exc_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.004887676</td><td>0.634741658</td><td>FALSE</td><td>FALSE</td><td>0.634741658</td><td>⋯</td><td>TRUE</td><td>chr20_50319421_53436693 </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr20_50319421_53436693.twas.tsv.gz </td><td>-0.1974030</td><td> 51747434</td><td>chr20</td><td> 50958555</td><td> 50934867</td><td>DPM1    </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ENSG00000000457</td><td>Ast_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.314317590</td><td>0.666951750</td><td>FALSE</td><td>FALSE</td><td>0.666951750</td><td>⋯</td><td>TRUE</td><td>chr1_168438717_170228106</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz</td><td> 0.1759056</td><td>169540000</td><td>chr1 </td><td>169894267</td><td>169849631</td><td>SCYL3   </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>ENSG00000000460</td><td>Ast_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.550799973</td><td>0.681781965</td><td>FALSE</td><td>FALSE</td><td>0.681781965</td><td>⋯</td><td>TRUE</td><td>chr1_168438717_170228106</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz</td><td>-0.1663545</td><td>169540000</td><td>chr1 </td><td>169662007</td><td>169854080</td><td>C1orf112</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>ENSG00000000971</td><td>PCC_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.006727627</td><td>0.006727627</td><td>FALSE</td><td>FALSE</td><td>0.006727627</td><td>⋯</td><td>TRUE</td><td>chr1_195599253_199271134</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr1_195599253_199271134.twas.tsv.gz</td><td> 2.1721381</td><td>195093752</td><td>chr1 </td><td>196652043</td><td>196747504</td><td>CFH     </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ENSG00000001036</td><td>Ast_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.125012981</td><td>0.125012981</td><td>FALSE</td><td>FALSE</td><td>0.125012981</td><td>⋯</td><td>TRUE</td><td>chr6_143399926_145632539</td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr6_143399926_145632539.twas.tsv.gz</td><td>-0.9030449</td><td>143115860</td><td>chr6 </td><td>143511720</td><td>143494812</td><td>FUCA2   </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ENSG00000001084</td><td>Ast_DeJager_eQTL</td><td>unadjusted_wp_2020</td><td>4</td><td>0</td><td>0.166978326</td><td>0.851335412</td><td>FALSE</td><td>FALSE</td><td>0.851335412</td><td>⋯</td><td>TRUE</td><td>chr6_52730905_54027603  </td><td>ROSMAP_eQTL_wp</td><td>ROSMAP_eQTL_wp_unadjusted.chr6_52730905_54027603.twas.tsv.gz  </td><td>-0.0698993</td><td> 53557156</td><td>chr6 </td><td> 53616970</td><td> 53497341</td><td>GCLC    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 33\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & molecular\\_id & context & gwas\\_study & n\\_methods & n\\_sig & best\\_pval & best\\_method\\_pval & best\\_method\\_selected & pass & final\\_pval & ⋯ & method\\_selected\\_original & region & study\\_context & source\\_file & logp & pos & \\#chr & TSS.y & TES & gene\\_name\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <int> & <dbl> & <dbl> & <lgl> & <lgl> & <dbl> & ⋯ & <lgl> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & ENSG00000000419 & Exc\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.004887676 & 0.634741658 & FALSE & FALSE & 0.634741658 & ⋯ & TRUE & chr20\\_50319421\\_53436693  & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr20\\_50319421\\_53436693.twas.tsv.gz  & -0.1974030 &  51747434 & chr20 &  50958555 &  50934867 & DPM1    \\\\\n",
       "\t2 & ENSG00000000457 & Ast\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.314317590 & 0.666951750 & FALSE & FALSE & 0.666951750 & ⋯ & TRUE & chr1\\_168438717\\_170228106 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr1\\_168438717\\_170228106.twas.tsv.gz &  0.1759056 & 169540000 & chr1  & 169894267 & 169849631 & SCYL3   \\\\\n",
       "\t3 & ENSG00000000460 & Ast\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.550799973 & 0.681781965 & FALSE & FALSE & 0.681781965 & ⋯ & TRUE & chr1\\_168438717\\_170228106 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr1\\_168438717\\_170228106.twas.tsv.gz & -0.1663545 & 169540000 & chr1  & 169662007 & 169854080 & C1orf112\\\\\n",
       "\t4 & ENSG00000000971 & PCC\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.006727627 & 0.006727627 & FALSE & FALSE & 0.006727627 & ⋯ & TRUE & chr1\\_195599253\\_199271134 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr1\\_195599253\\_199271134.twas.tsv.gz &  2.1721381 & 195093752 & chr1  & 196652043 & 196747504 & CFH     \\\\\n",
       "\t5 & ENSG00000001036 & Ast\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.125012981 & 0.125012981 & FALSE & FALSE & 0.125012981 & ⋯ & TRUE & chr6\\_143399926\\_145632539 & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr6\\_143399926\\_145632539.twas.tsv.gz & -0.9030449 & 143115860 & chr6  & 143511720 & 143494812 & FUCA2   \\\\\n",
       "\t6 & ENSG00000001084 & Ast\\_DeJager\\_eQTL & unadjusted\\_wp\\_2020 & 4 & 0 & 0.166978326 & 0.851335412 & FALSE & FALSE & 0.851335412 & ⋯ & TRUE & chr6\\_52730905\\_54027603   & ROSMAP\\_eQTL\\_wp & ROSMAP\\_eQTL\\_wp\\_unadjusted.chr6\\_52730905\\_54027603.twas.tsv.gz   & -0.0698993 &  53557156 & chr6  &  53616970 &  53497341 & GCLC    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 33\n",
       "\n",
       "| <!--/--> | molecular_id &lt;chr&gt; | context &lt;chr&gt; | gwas_study &lt;chr&gt; | n_methods &lt;int&gt; | n_sig &lt;int&gt; | best_pval &lt;dbl&gt; | best_method_pval &lt;dbl&gt; | best_method_selected &lt;lgl&gt; | pass &lt;lgl&gt; | final_pval &lt;dbl&gt; | ⋯ ⋯ | method_selected_original &lt;lgl&gt; | region &lt;chr&gt; | study_context &lt;chr&gt; | source_file &lt;chr&gt; | logp &lt;dbl&gt; | pos &lt;dbl&gt; | #chr &lt;chr&gt; | TSS.y &lt;int&gt; | TES &lt;int&gt; | gene_name &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | ENSG00000000419 | Exc_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.004887676 | 0.634741658 | FALSE | FALSE | 0.634741658 | ⋯ | TRUE | chr20_50319421_53436693  | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr20_50319421_53436693.twas.tsv.gz  | -0.1974030 |  51747434 | chr20 |  50958555 |  50934867 | DPM1     |\n",
       "| 2 | ENSG00000000457 | Ast_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.314317590 | 0.666951750 | FALSE | FALSE | 0.666951750 | ⋯ | TRUE | chr1_168438717_170228106 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz |  0.1759056 | 169540000 | chr1  | 169894267 | 169849631 | SCYL3    |\n",
       "| 3 | ENSG00000000460 | Ast_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.550799973 | 0.681781965 | FALSE | FALSE | 0.681781965 | ⋯ | TRUE | chr1_168438717_170228106 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz | -0.1663545 | 169540000 | chr1  | 169662007 | 169854080 | C1orf112 |\n",
       "| 4 | ENSG00000000971 | PCC_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.006727627 | 0.006727627 | FALSE | FALSE | 0.006727627 | ⋯ | TRUE | chr1_195599253_199271134 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr1_195599253_199271134.twas.tsv.gz |  2.1721381 | 195093752 | chr1  | 196652043 | 196747504 | CFH      |\n",
       "| 5 | ENSG00000001036 | Ast_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.125012981 | 0.125012981 | FALSE | FALSE | 0.125012981 | ⋯ | TRUE | chr6_143399926_145632539 | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr6_143399926_145632539.twas.tsv.gz | -0.9030449 | 143115860 | chr6  | 143511720 | 143494812 | FUCA2    |\n",
       "| 6 | ENSG00000001084 | Ast_DeJager_eQTL | unadjusted_wp_2020 | 4 | 0 | 0.166978326 | 0.851335412 | FALSE | FALSE | 0.851335412 | ⋯ | TRUE | chr6_52730905_54027603   | ROSMAP_eQTL_wp | ROSMAP_eQTL_wp_unadjusted.chr6_52730905_54027603.twas.tsv.gz   | -0.0698993 |  53557156 | chr6  |  53616970 |  53497341 | GCLC     |\n",
       "\n"
      ],
      "text/plain": [
       "  molecular_id    context          gwas_study         n_methods n_sig\n",
       "1 ENSG00000000419 Exc_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "2 ENSG00000000457 Ast_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "3 ENSG00000000460 Ast_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "4 ENSG00000000971 PCC_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "5 ENSG00000001036 Ast_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "6 ENSG00000001084 Ast_DeJager_eQTL unadjusted_wp_2020 4         0    \n",
       "  best_pval   best_method_pval best_method_selected pass  final_pval  ⋯\n",
       "1 0.004887676 0.634741658      FALSE                FALSE 0.634741658 ⋯\n",
       "2 0.314317590 0.666951750      FALSE                FALSE 0.666951750 ⋯\n",
       "3 0.550799973 0.681781965      FALSE                FALSE 0.681781965 ⋯\n",
       "4 0.006727627 0.006727627      FALSE                FALSE 0.006727627 ⋯\n",
       "5 0.125012981 0.125012981      FALSE                FALSE 0.125012981 ⋯\n",
       "6 0.166978326 0.851335412      FALSE                FALSE 0.851335412 ⋯\n",
       "  method_selected_original region                   study_context \n",
       "1 TRUE                     chr20_50319421_53436693  ROSMAP_eQTL_wp\n",
       "2 TRUE                     chr1_168438717_170228106 ROSMAP_eQTL_wp\n",
       "3 TRUE                     chr1_168438717_170228106 ROSMAP_eQTL_wp\n",
       "4 TRUE                     chr1_195599253_199271134 ROSMAP_eQTL_wp\n",
       "5 TRUE                     chr6_143399926_145632539 ROSMAP_eQTL_wp\n",
       "6 TRUE                     chr6_52730905_54027603   ROSMAP_eQTL_wp\n",
       "  source_file                                                    logp      \n",
       "1 ROSMAP_eQTL_wp_unadjusted.chr20_50319421_53436693.twas.tsv.gz  -0.1974030\n",
       "2 ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz  0.1759056\n",
       "3 ROSMAP_eQTL_wp_unadjusted.chr1_168438717_170228106.twas.tsv.gz -0.1663545\n",
       "4 ROSMAP_eQTL_wp_unadjusted.chr1_195599253_199271134.twas.tsv.gz  2.1721381\n",
       "5 ROSMAP_eQTL_wp_unadjusted.chr6_143399926_145632539.twas.tsv.gz -0.9030449\n",
       "6 ROSMAP_eQTL_wp_unadjusted.chr6_52730905_54027603.twas.tsv.gz   -0.0698993\n",
       "  pos       #chr  TSS.y     TES       gene_name\n",
       "1  51747434 chr20  50958555  50934867 DPM1     \n",
       "2 169540000 chr1  169894267 169849631 SCYL3    \n",
       "3 169540000 chr1  169662007 169854080 C1orf112 \n",
       "4 195093752 chr1  196652043 196747504 CFH      \n",
       "5 143115860 chr6  143511720 143494812 FUCA2    \n",
       "6  53557156 chr6   53616970  53497341 GCLC     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "240a37d2-b50f-42d9-a8ba-177d0ea16f00",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "chr_info <- df %>%\n",
    "  group_by(chr) %>%\n",
    "  summarise(chr_len = max(pos)) %>%\n",
    "  mutate(chr_start = lag(cumsum(chr_len), default = 0))\n",
    "\n",
    "df <- df %>%\n",
    "  left_join(chr_info, by = \"chr\") %>%\n",
    "  mutate(cum_pos = pos + chr_start)\n",
    "  \n",
    "axis_df <- chr_info %>%\n",
    "  mutate(center = chr_start + chr_len/2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "f884ab9d-068c-4ee3-bcf9-67182c3e2b0a",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "df <- mr_sig_unadj %>% rename('molecular_id' = 'gene_name') %>% merge(df, . , all.x = T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "7a99be0a-499f-4ee3-adad-d0a564927c3e",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 0 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>molecular_id</th><th scope=col>context</th><th scope=col>gene_name</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 0 × 3\n",
       "\\begin{tabular}{lll}\n",
       " molecular\\_id & context & gene\\_name\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 0 × 3\n",
       "\n",
       "| molecular_id &lt;chr&gt; | context &lt;chr&gt; | gene_name &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     molecular_id context gene_name"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df %>% filter(!is.na(Q_pval)) %>% select(molecular_id, context, gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "d5e7f215-7297-4242-97c3-329cb21953f7",
   "metadata": {
    "kernel": "R",
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "\n",
    "# Add a column for MR significance (example: FDR < 0.05)\n",
    "df <- df %>%\n",
    "  mutate(category = case_when(\n",
    "    !is.na(Q_pval) & pass ~ \"TWAS & MR\",        # Both significant\n",
    "    is.na(Q_pval) & pass ~ \"TWAS only\",         # Only TWAS significant\n",
    "    !pass & is.na(Q_pval) ~ \"Not Significant\",  # Neither significant\n",
    "    TRUE ~ \"Others\"                              # Catch-all for remaining cases\n",
    "  ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "46571a3d-98d3-4def-953b-ea9879608e28",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# dir.create('data')\n",
    "saveRDS(list(df = df, unadj_cutoff = unadj_cutoff, axis_df= axis_df), 'data/TWAS_manhattan_plot_data.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0456b84a-5e08-423a-be45-5b64d32e6861",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.3"
  },
  "sos": {
   "kernels": [
    [
     "R",
     "ir",
     "R",
     "",
     ""
    ]
   ],
   "version": ""
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
