{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "9cf188ea",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "ded4a1cd",
   "metadata": {},
   "source": [
    "## Aim: \n",
    "The GWAS summary statistics were originally based on the hg19 reference genome, whereas our current LD reference panel is hg38-based. To ensure accurate LD positioning for finemapping, we converted the GWAS summary statistics to hg38 using LiftOver.\n",
    "## Input:\n",
    "* LiftOver tool: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver\n",
    "* hg19 → hg38 chain file: http://hgdownload.cse.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz\n",
    "* GWAS summary statistics:\n",
    "1. unadjusted walking pace GWAS: https://drive.google.com/drive/folders/1H1Xj33C-867dxVHOFIh5l_nLluWcnqzx\n",
    "\n",
    "* convert.sh: script to convert hg19 to hg38.\n",
    "* file_path.txt: the file recording the path to hg19 GWAS statistic(.bed)\n",
    "## Output:\n",
    "* hg_38 based GWAS summary statistics: `s3://statfungen/ftp_fgc_xqtl/GWAS/image_GWAS_hg38/`\n",
    "\n",
    "## Procedures:\n",
    "1. Format GWAS Summary Statistics for Conversion\n",
    "\n",
    "Convert hg19-based GWAS summary statistics into standard .bed format with the following four required columns:chrom (without the chr prefix) start end region_id (to facilitate merging after conversion)\n",
    "```\n",
    "chrom\tstart\tend\tregion_id\n",
    "<chr>\t<int>\t<int>\t<chr>\n",
    "chr5\t29439275\t29439275\trs667647\n",
    "chr5\t85928892\t85928892\trs113534962\n",
    "```\n",
    "**Note: LiftOver does not support .bed files with more than six columns. Since AD/aging image GWAS summary statistics share the same variant positions across multiple dimensions, we perform the conversion once for both datasets.**\n",
    "\n",
    "2. Perform LiftOver Conversion\n",
    "Use the `convert.sh` script to run LiftOver and map hg19 coordinates to hg38.\n",
    "\n",
    "**Note: the conversion introduced some additional contigs and scaffolds that represent alternative loci or regions that are difficult to place within the main chromosomes, e.g. 'chr14_GL000009v2_random''chr19_KI270938v1_alt'. They are hard to interpret, so I removed them and only keep chr1-22.**\n",
    "\n",
    "3. Merge Back to GWAS Summary Statistics\n",
    "Load the hg38 .bed file and merge it back with the original GWAS summary statistics, preserving all necessary information.\n",
    "\n",
    "## Simple summary for the conversion\n",
    "\n",
    "| Studies | before_conversion(original) | unmapped | after_conversion(final) | overall_dropped | proportion_dropped |\n",
    "|---------|----------------------------|----------|------------------------|-----------------|-------------------|\n",
    "| unadjusted walking pace GWAS | 11,335,563 | 3,939 | 11,331,624 | 5,999 | 0.0529% |\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "7204463b",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "0.000529219413274841"
      ],
      "text/latex": [
       "0.000529219413274841"
      ],
      "text/markdown": [
       "0.000529219413274841"
      ],
      "text/plain": [
       "[1] 0.0005292194"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "5999/11335563"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e5f1ad45-1da5-4572-82a5-250768da448e",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "library(data.table)\n",
    "library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "595201c0-6f88-4b2b-a091-5f07434f93eb",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "options(scipen = 999)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6b7ae2f4-06df-4c69-8c1d-64176a391c8d",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SNP</th><th scope=col>CHR</th><th scope=col>BP</th><th scope=col>EFFECT.ALLELE</th><th scope=col>ALT.ALLELE</th><th scope=col>EAF</th><th scope=col>INFO</th><th scope=col>BETA</th><th scope=col>SE</th><th scope=col>P</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1:612688_TCTC_T</td><td>1</td><td>612688</td><td>TCTC</td><td>T</td><td>0.979815</td><td>0.617496</td><td> 0.001010140</td><td>0.00584836</td><td>0.90</td></tr>\n",
       "\t<tr><td>rs545945172    </td><td>1</td><td>636285</td><td>T   </td><td>C</td><td>0.903972</td><td>0.721502</td><td>-0.000918819</td><td>0.00247797</td><td>0.71</td></tr>\n",
       "\t<tr><td>rs201942322    </td><td>1</td><td>649192</td><td>A   </td><td>T</td><td>0.882575</td><td>0.743729</td><td>-0.000343830</td><td>0.00223761</td><td>0.87</td></tr>\n",
       "\t<tr><td>rs371628865    </td><td>1</td><td>662414</td><td>C   </td><td>T</td><td>0.864997</td><td>0.694792</td><td>-0.000533999</td><td>0.00219459</td><td>0.81</td></tr>\n",
       "\t<tr><td>rs61769339     </td><td>1</td><td>662622</td><td>G   </td><td>A</td><td>0.889532</td><td>0.777266</td><td>-0.000191469</td><td>0.00224530</td><td>0.92</td></tr>\n",
       "\t<tr><td>rs539032812    </td><td>1</td><td>665266</td><td>T   </td><td>C</td><td>0.981031</td><td>0.670456</td><td> 0.004110490</td><td>0.00558377</td><td>0.45</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " SNP & CHR & BP & EFFECT.ALLELE & ALT.ALLELE & EAF & INFO & BETA & SE & P\\\\\n",
       " <chr> & <int> & <dbl> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 1:612688\\_TCTC\\_T & 1 & 612688 & TCTC & T & 0.979815 & 0.617496 &  0.001010140 & 0.00584836 & 0.90\\\\\n",
       "\t rs545945172     & 1 & 636285 & T    & C & 0.903972 & 0.721502 & -0.000918819 & 0.00247797 & 0.71\\\\\n",
       "\t rs201942322     & 1 & 649192 & A    & T & 0.882575 & 0.743729 & -0.000343830 & 0.00223761 & 0.87\\\\\n",
       "\t rs371628865     & 1 & 662414 & C    & T & 0.864997 & 0.694792 & -0.000533999 & 0.00219459 & 0.81\\\\\n",
       "\t rs61769339      & 1 & 662622 & G    & A & 0.889532 & 0.777266 & -0.000191469 & 0.00224530 & 0.92\\\\\n",
       "\t rs539032812     & 1 & 665266 & T    & C & 0.981031 & 0.670456 &  0.004110490 & 0.00558377 & 0.45\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 10\n",
       "\n",
       "| SNP &lt;chr&gt; | CHR &lt;int&gt; | BP &lt;dbl&gt; | EFFECT.ALLELE &lt;chr&gt; | ALT.ALLELE &lt;chr&gt; | EAF &lt;dbl&gt; | INFO &lt;dbl&gt; | BETA &lt;dbl&gt; | SE &lt;dbl&gt; | P &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1:612688_TCTC_T | 1 | 612688 | TCTC | T | 0.979815 | 0.617496 |  0.001010140 | 0.00584836 | 0.90 |\n",
       "| rs545945172     | 1 | 636285 | T    | C | 0.903972 | 0.721502 | -0.000918819 | 0.00247797 | 0.71 |\n",
       "| rs201942322     | 1 | 649192 | A    | T | 0.882575 | 0.743729 | -0.000343830 | 0.00223761 | 0.87 |\n",
       "| rs371628865     | 1 | 662414 | C    | T | 0.864997 | 0.694792 | -0.000533999 | 0.00219459 | 0.81 |\n",
       "| rs61769339      | 1 | 662622 | G    | A | 0.889532 | 0.777266 | -0.000191469 | 0.00224530 | 0.92 |\n",
       "| rs539032812     | 1 | 665266 | T    | C | 0.981031 | 0.670456 |  0.004110490 | 0.00558377 | 0.45 |\n",
       "\n"
      ],
      "text/plain": [
       "  SNP             CHR BP     EFFECT.ALLELE ALT.ALLELE EAF      INFO    \n",
       "1 1:612688_TCTC_T 1   612688 TCTC          T          0.979815 0.617496\n",
       "2 rs545945172     1   636285 T             C          0.903972 0.721502\n",
       "3 rs201942322     1   649192 A             T          0.882575 0.743729\n",
       "4 rs371628865     1   662414 C             T          0.864997 0.694792\n",
       "5 rs61769339      1   662622 G             A          0.889532 0.777266\n",
       "6 rs539032812     1   665266 T             C          0.981031 0.670456\n",
       "  BETA         SE         P   \n",
       "1  0.001010140 0.00584836 0.90\n",
       "2 -0.000918819 0.00247797 0.71\n",
       "3 -0.000343830 0.00223761 0.87\n",
       "4 -0.000533999 0.00219459 0.81\n",
       "5 -0.000191469 0.00224530 0.92\n",
       "6  0.004110490 0.00558377 0.45"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>11335563</li><li>10</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 11335563\n",
       "\\item 10\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 11335563\n",
       "2. 10\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 11335563       10"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unadjusted_GWAS = fread(\"~/rl3328/motor_qtl/GWAS/wp_ukb.txt.gz\")\n",
    "head(unadjusted_GWAS)\n",
    "dim(unadjusted_GWAS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c08da135-1569-4e52-973b-e13a0c19df72",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sum(grepl(\"[eE]\", unadjusted_GWAS$BP))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "534ee966-e8f2-4bce-a0ff-de69996d3c66",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8f7976b3-3a17-4279-9fa0-3c0c256f6218",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "unadjusted_GWAS_needed = unadjusted_GWAS |> mutate(chrom = paste0(\"chr\",CHR), start = BP, end = BP, region_id = SNP) |> select(chrom, start, end, region_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bd4918c6-7578-4b44-80ca-577c85fffe72",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>region_id</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>612688</td><td>612688</td><td>1:612688_TCTC_T</td></tr>\n",
       "\t<tr><td>chr1</td><td>636285</td><td>636285</td><td>rs545945172    </td></tr>\n",
       "\t<tr><td>chr1</td><td>649192</td><td>649192</td><td>rs201942322    </td></tr>\n",
       "\t<tr><td>chr1</td><td>662414</td><td>662414</td><td>rs371628865    </td></tr>\n",
       "\t<tr><td>chr1</td><td>662622</td><td>662622</td><td>rs61769339     </td></tr>\n",
       "\t<tr><td>chr1</td><td>665266</td><td>665266</td><td>rs539032812    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 4\n",
       "\\begin{tabular}{llll}\n",
       " chrom & start & end & region\\_id\\\\\n",
       " <chr> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t chr1 & 612688 & 612688 & 1:612688\\_TCTC\\_T\\\\\n",
       "\t chr1 & 636285 & 636285 & rs545945172    \\\\\n",
       "\t chr1 & 649192 & 649192 & rs201942322    \\\\\n",
       "\t chr1 & 662414 & 662414 & rs371628865    \\\\\n",
       "\t chr1 & 662622 & 662622 & rs61769339     \\\\\n",
       "\t chr1 & 665266 & 665266 & rs539032812    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 4\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;dbl&gt; | end &lt;dbl&gt; | region_id &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| chr1 | 612688 | 612688 | 1:612688_TCTC_T |\n",
       "| chr1 | 636285 | 636285 | rs545945172     |\n",
       "| chr1 | 649192 | 649192 | rs201942322     |\n",
       "| chr1 | 662414 | 662414 | rs371628865     |\n",
       "| chr1 | 662622 | 662622 | rs61769339      |\n",
       "| chr1 | 665266 | 665266 | rs539032812     |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start  end    region_id      \n",
       "1 chr1  612688 612688 1:612688_TCTC_T\n",
       "2 chr1  636285 636285 rs545945172    \n",
       "3 chr1  649192 649192 rs201942322    \n",
       "4 chr1  662414 662414 rs371628865    \n",
       "5 chr1  662622 662622 rs61769339     \n",
       "6 chr1  665266 665266 rs539032812    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>11335563</li><li>4</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 11335563\n",
       "\\item 4\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 11335563\n",
       "2. 4\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 11335563        4"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(unadjusted_GWAS_needed)\n",
    "dim(unadjusted_GWAS_needed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "57e85a46-f81a-44f0-8c68-39394a6889f4",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "fwrite(unadjusted_GWAS_needed,\"/mnt/lustre/home/rl3328/rl3328/motor_qtl/GWAS/unadjusted_GWAS_hg19.bed\", sep = '\\t',col.names=FALSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b7f3fc1-e287-4732-940e-1ee303319dc6",
   "metadata": {},
   "source": [
    "# Read in the hg38 .bed(three columns-chrom, pos, id) and merge it back to the original summary statistics "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f6639dd3-ea37-4396-9099-0a495e4c071b",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "unadjusted_GWAS_hg38 = fread(\"/mnt/lustre/home/rl3328/rl3328/motor_qtl/GWAS/unadjusted_GWAS_hg19.to_hg38.bed\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "84926983-0aa1-427f-867a-22c01a5b34fd",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>677308</td><td>677308</td><td>1:612688_TCTC_T</td></tr>\n",
       "\t<tr><td>chr1</td><td>700905</td><td>700905</td><td>rs545945172    </td></tr>\n",
       "\t<tr><td>chr1</td><td>713812</td><td>713812</td><td>rs201942322    </td></tr>\n",
       "\t<tr><td>chr1</td><td>727034</td><td>727034</td><td>rs371628865    </td></tr>\n",
       "\t<tr><td>chr1</td><td>727242</td><td>727242</td><td>rs61769339     </td></tr>\n",
       "\t<tr><td>chr1</td><td>729886</td><td>729886</td><td>rs539032812    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 4\n",
       "\\begin{tabular}{llll}\n",
       " V1 & V2 & V3 & V4\\\\\n",
       " <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t chr1 & 677308 & 677308 & 1:612688\\_TCTC\\_T\\\\\n",
       "\t chr1 & 700905 & 700905 & rs545945172    \\\\\n",
       "\t chr1 & 713812 & 713812 & rs201942322    \\\\\n",
       "\t chr1 & 727034 & 727034 & rs371628865    \\\\\n",
       "\t chr1 & 727242 & 727242 & rs61769339     \\\\\n",
       "\t chr1 & 729886 & 729886 & rs539032812    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 4\n",
       "\n",
       "| V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| chr1 | 677308 | 677308 | 1:612688_TCTC_T |\n",
       "| chr1 | 700905 | 700905 | rs545945172     |\n",
       "| chr1 | 713812 | 713812 | rs201942322     |\n",
       "| chr1 | 727034 | 727034 | rs371628865     |\n",
       "| chr1 | 727242 | 727242 | rs61769339      |\n",
       "| chr1 | 729886 | 729886 | rs539032812     |\n",
       "\n"
      ],
      "text/plain": [
       "  V1   V2     V3     V4             \n",
       "1 chr1 677308 677308 1:612688_TCTC_T\n",
       "2 chr1 700905 700905 rs545945172    \n",
       "3 chr1 713812 713812 rs201942322    \n",
       "4 chr1 727034 727034 rs371628865    \n",
       "5 chr1 727242 727242 rs61769339     \n",
       "6 chr1 729886 729886 rs539032812    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(unadjusted_GWAS_hg38)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e62a7bb2-eae8-43c2-a326-e3f97370139d",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>11331624</li><li>4</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 11331624\n",
       "\\item 4\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 11331624\n",
       "2. 4\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 11331624        4"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(unadjusted_GWAS_hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "e00d64fe-f2de-46fa-a145-1a8154310469",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "unadjusted_GWAS_hg38 = unadjusted_GWAS_hg38[,-3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "be13f391-1259-4019-81f2-6e58d5823bba",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "colnames(unadjusted_GWAS_hg38) <- c(\"chr\",\"pos\",\"rsid\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "8a35021d-c7aa-49af-af99-2ebf1782b941",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "unadjusted_GWAS_hg38 = unadjusted_GWAS_hg38 |> mutate(chr = gsub(\"chr\", \"\", chr))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ee4dc1f6-dbb7-4d0f-976d-d59b643eb38e",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“\u001b[1m\u001b[22mThere was 1 warning in `mutate()`.\n",
      "\u001b[1m\u001b[22m\u001b[36mℹ\u001b[39m In argument: `chr = as.integer(chr)`.\n",
      "Caused by warning:\n",
      "\u001b[33m!\u001b[39m NAs introduced by coercion”\n"
     ]
    }
   ],
   "source": [
    "unadjusted_GWAS_hg38 = unadjusted_GWAS_hg38 |> mutate(chr=as.integer(chr))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "7858bc7a-9057-4f97-8fd6-322cd14493ea",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "unadjusted_GWAS_hg38 = unadjusted_GWAS_hg38 |> filter(!is.na(chr))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "551f82af-74c9-44ec-82d0-7c086fe4bfca",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chr</th><th scope=col>pos</th><th scope=col>rsid</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>677308</td><td>1:612688_TCTC_T</td></tr>\n",
       "\t<tr><td>1</td><td>700905</td><td>rs545945172    </td></tr>\n",
       "\t<tr><td>1</td><td>713812</td><td>rs201942322    </td></tr>\n",
       "\t<tr><td>1</td><td>727034</td><td>rs371628865    </td></tr>\n",
       "\t<tr><td>1</td><td>727242</td><td>rs61769339     </td></tr>\n",
       "\t<tr><td>1</td><td>729886</td><td>rs539032812    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 3\n",
       "\\begin{tabular}{lll}\n",
       " chr & pos & rsid\\\\\n",
       " <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t 1 & 677308 & 1:612688\\_TCTC\\_T\\\\\n",
       "\t 1 & 700905 & rs545945172    \\\\\n",
       "\t 1 & 713812 & rs201942322    \\\\\n",
       "\t 1 & 727034 & rs371628865    \\\\\n",
       "\t 1 & 727242 & rs61769339     \\\\\n",
       "\t 1 & 729886 & rs539032812    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 3\n",
       "\n",
       "| chr &lt;int&gt; | pos &lt;int&gt; | rsid &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | 677308 | 1:612688_TCTC_T |\n",
       "| 1 | 700905 | rs545945172     |\n",
       "| 1 | 713812 | rs201942322     |\n",
       "| 1 | 727034 | rs371628865     |\n",
       "| 1 | 727242 | rs61769339      |\n",
       "| 1 | 729886 | rs539032812     |\n",
       "\n"
      ],
      "text/plain": [
       "  chr pos    rsid           \n",
       "1 1   677308 1:612688_TCTC_T\n",
       "2 1   700905 rs545945172    \n",
       "3 1   713812 rs201942322    \n",
       "4 1   727034 rs371628865    \n",
       "5 1   727242 rs61769339     \n",
       "6 1   729886 rs539032812    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>11329564</li><li>3</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 11329564\n",
       "\\item 3\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 11329564\n",
       "2. 3\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 11329564        3"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(unadjusted_GWAS_hg38)\n",
    "dim(unadjusted_GWAS_hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "3d6a9fa2-7eea-452f-8eff-9b901f8eb8d7",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>1</li><li>21</li><li>15</li><li>9</li><li>12</li><li>2</li><li>11</li><li>14</li><li>8</li><li>16</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>13</li><li>20</li><li>10</li><li>17</li><li>18</li><li>19</li><li>22</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1\n",
       "\\item 21\n",
       "\\item 15\n",
       "\\item 9\n",
       "\\item 12\n",
       "\\item 2\n",
       "\\item 11\n",
       "\\item 14\n",
       "\\item 8\n",
       "\\item 16\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 13\n",
       "\\item 20\n",
       "\\item 10\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 22\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1\n",
       "2. 21\n",
       "3. 15\n",
       "4. 9\n",
       "5. 12\n",
       "6. 2\n",
       "7. 11\n",
       "8. 14\n",
       "9. 8\n",
       "10. 16\n",
       "11. 3\n",
       "12. 4\n",
       "13. 5\n",
       "14. 6\n",
       "15. 7\n",
       "16. 13\n",
       "17. 20\n",
       "18. 10\n",
       "19. 17\n",
       "20. 18\n",
       "21. 19\n",
       "22. 22\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1]  1 21 15  9 12  2 11 14  8 16  3  4  5  6  7 13 20 10 17 18 19 22"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "unique(unadjusted_GWAS_hg38$chr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "eacbc51d-f4cd-4271-9ff7-299bca41fd02",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in inner_join(unadjusted_GWAS_remain, unadjusted_GWAS_hg38, by = c(SNP = \"rsid\")):\n",
      "“\u001b[1m\u001b[22mDetected an unexpected many-to-many relationship between `x` and `y`.\n",
      "\u001b[36mℹ\u001b[39m Row 2397 of `x` matches multiple rows in `y`.\n",
      "\u001b[36mℹ\u001b[39m Row 2397 of `y` matches multiple rows in `x`.\n",
      "\u001b[36mℹ\u001b[39m If a many-to-many relationship is expected, set `relationship =\n",
      "  \"many-to-many\"` to silence this warning.”\n"
     ]
    }
   ],
   "source": [
    "unadjusted_GWAS_remain = unadjusted_GWAS |> select(-CHR, -BP)\n",
    "unadjusted_GWAS_hg38_final = unadjusted_GWAS_remain |> inner_join(unadjusted_GWAS_hg38, by = c('SNP' ='rsid'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "c1d1ee42-c350-4fc0-86d9-9dfaf7ff8969",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SNP</th><th scope=col>EFFECT.ALLELE</th><th scope=col>ALT.ALLELE</th><th scope=col>EAF</th><th scope=col>INFO</th><th scope=col>BETA</th><th scope=col>SE</th><th scope=col>P</th><th scope=col>chr</th><th scope=col>pos</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1:612688_TCTC_T</td><td>TCTC</td><td>T</td><td>0.979815</td><td>0.617496</td><td> 0.001010140</td><td>0.00584836</td><td>0.90</td><td>1</td><td>677308</td></tr>\n",
       "\t<tr><td>rs545945172    </td><td>T   </td><td>C</td><td>0.903972</td><td>0.721502</td><td>-0.000918819</td><td>0.00247797</td><td>0.71</td><td>1</td><td>700905</td></tr>\n",
       "\t<tr><td>rs201942322    </td><td>A   </td><td>T</td><td>0.882575</td><td>0.743729</td><td>-0.000343830</td><td>0.00223761</td><td>0.87</td><td>1</td><td>713812</td></tr>\n",
       "\t<tr><td>rs371628865    </td><td>C   </td><td>T</td><td>0.864997</td><td>0.694792</td><td>-0.000533999</td><td>0.00219459</td><td>0.81</td><td>1</td><td>727034</td></tr>\n",
       "\t<tr><td>rs61769339     </td><td>G   </td><td>A</td><td>0.889532</td><td>0.777266</td><td>-0.000191469</td><td>0.00224530</td><td>0.92</td><td>1</td><td>727242</td></tr>\n",
       "\t<tr><td>rs539032812    </td><td>T   </td><td>C</td><td>0.981031</td><td>0.670456</td><td> 0.004110490</td><td>0.00558377</td><td>0.45</td><td>1</td><td>729886</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " SNP & EFFECT.ALLELE & ALT.ALLELE & EAF & INFO & BETA & SE & P & chr & pos\\\\\n",
       " <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t 1:612688\\_TCTC\\_T & TCTC & T & 0.979815 & 0.617496 &  0.001010140 & 0.00584836 & 0.90 & 1 & 677308\\\\\n",
       "\t rs545945172     & T    & C & 0.903972 & 0.721502 & -0.000918819 & 0.00247797 & 0.71 & 1 & 700905\\\\\n",
       "\t rs201942322     & A    & T & 0.882575 & 0.743729 & -0.000343830 & 0.00223761 & 0.87 & 1 & 713812\\\\\n",
       "\t rs371628865     & C    & T & 0.864997 & 0.694792 & -0.000533999 & 0.00219459 & 0.81 & 1 & 727034\\\\\n",
       "\t rs61769339      & G    & A & 0.889532 & 0.777266 & -0.000191469 & 0.00224530 & 0.92 & 1 & 727242\\\\\n",
       "\t rs539032812     & T    & C & 0.981031 & 0.670456 &  0.004110490 & 0.00558377 & 0.45 & 1 & 729886\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 10\n",
       "\n",
       "| SNP &lt;chr&gt; | EFFECT.ALLELE &lt;chr&gt; | ALT.ALLELE &lt;chr&gt; | EAF &lt;dbl&gt; | INFO &lt;dbl&gt; | BETA &lt;dbl&gt; | SE &lt;dbl&gt; | P &lt;dbl&gt; | chr &lt;int&gt; | pos &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1:612688_TCTC_T | TCTC | T | 0.979815 | 0.617496 |  0.001010140 | 0.00584836 | 0.90 | 1 | 677308 |\n",
       "| rs545945172     | T    | C | 0.903972 | 0.721502 | -0.000918819 | 0.00247797 | 0.71 | 1 | 700905 |\n",
       "| rs201942322     | A    | T | 0.882575 | 0.743729 | -0.000343830 | 0.00223761 | 0.87 | 1 | 713812 |\n",
       "| rs371628865     | C    | T | 0.864997 | 0.694792 | -0.000533999 | 0.00219459 | 0.81 | 1 | 727034 |\n",
       "| rs61769339      | G    | A | 0.889532 | 0.777266 | -0.000191469 | 0.00224530 | 0.92 | 1 | 727242 |\n",
       "| rs539032812     | T    | C | 0.981031 | 0.670456 |  0.004110490 | 0.00558377 | 0.45 | 1 | 729886 |\n",
       "\n"
      ],
      "text/plain": [
       "  SNP             EFFECT.ALLELE ALT.ALLELE EAF      INFO     BETA        \n",
       "1 1:612688_TCTC_T TCTC          T          0.979815 0.617496  0.001010140\n",
       "2 rs545945172     T             C          0.903972 0.721502 -0.000918819\n",
       "3 rs201942322     A             T          0.882575 0.743729 -0.000343830\n",
       "4 rs371628865     C             T          0.864997 0.694792 -0.000533999\n",
       "5 rs61769339      G             A          0.889532 0.777266 -0.000191469\n",
       "6 rs539032812     T             C          0.981031 0.670456  0.004110490\n",
       "  SE         P    chr pos   \n",
       "1 0.00584836 0.90 1   677308\n",
       "2 0.00247797 0.71 1   700905\n",
       "3 0.00223761 0.87 1   713812\n",
       "4 0.00219459 0.81 1   727034\n",
       "5 0.00224530 0.92 1   727242\n",
       "6 0.00558377 0.45 1   729886"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>11358832</li><li>10</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 11358832\n",
       "\\item 10\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 11358832\n",
       "2. 10\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 11358832       10"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(unadjusted_GWAS_hg38_final)\n",
    "dim(unadjusted_GWAS_hg38_final)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "3481ac1e-eda4-40eb-96b6-b0a461e1f296",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# unadjusted_GWAS_hg38_final = unadjusted_GWAS_hg38_final |> rename(A1 = EFFECT.ALLELE, A2 = ALT.ALLELE, chrom = chr)\n",
    "unadjusted_GWAS_hg38_final = unadjusted_GWAS_hg38_final |> mutate(n_sample = 1 / (2 * EAF * (1 - EAF) * SE^2)) |> rename(beta = BETA, se = SE, p = P, effect_allele_frequency = EAF)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "d754de8f-45f1-417a-a68c-30e26ad17886",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "unadjusted_GWAS_hg38_final = unadjusted_GWAS_hg38_final |> arrange(chrom, pos) |> select(chrom, pos, A1, A2, everything())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "06bd9ecc",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>pos</th><th scope=col>A1</th><th scope=col>A2</th><th scope=col>SNP</th><th scope=col>effect_allele_frequency</th><th scope=col>INFO</th><th scope=col>beta</th><th scope=col>se</th><th scope=col>p</th><th scope=col>n_sample</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td> 97269</td><td>T   </td><td>TAC</td><td>rs199687009    </td><td>0.904921</td><td>0.847312</td><td> 0.000316765</td><td>0.00224748</td><td>0.88</td><td>1150490.6</td></tr>\n",
       "\t<tr><td>1</td><td>677308</td><td>TCTC</td><td>T  </td><td>1:612688_TCTC_T</td><td>0.979815</td><td>0.617496</td><td> 0.001010140</td><td>0.00584836</td><td>0.90</td><td> 739143.9</td></tr>\n",
       "\t<tr><td>1</td><td>700905</td><td>T   </td><td>C  </td><td>rs545945172    </td><td>0.903972</td><td>0.721502</td><td>-0.000918819</td><td>0.00247797</td><td>0.71</td><td> 938048.0</td></tr>\n",
       "\t<tr><td>1</td><td>713812</td><td>A   </td><td>T  </td><td>rs201942322    </td><td>0.882575</td><td>0.743729</td><td>-0.000343830</td><td>0.00223761</td><td>0.87</td><td> 963582.8</td></tr>\n",
       "\t<tr><td>1</td><td>727034</td><td>C   </td><td>T  </td><td>rs371628865    </td><td>0.864997</td><td>0.694792</td><td>-0.000533999</td><td>0.00219459</td><td>0.81</td><td> 889007.0</td></tr>\n",
       "\t<tr><td>1</td><td>727242</td><td>G   </td><td>A  </td><td>rs61769339     </td><td>0.889532</td><td>0.777266</td><td>-0.000191469</td><td>0.00224530</td><td>0.92</td><td>1009306.8</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 11\n",
       "\\begin{tabular}{lllllllllll}\n",
       " chrom & pos & A1 & A2 & SNP & effect\\_allele\\_frequency & INFO & beta & se & p & n\\_sample\\\\\n",
       " <int> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 1 &  97269 & T    & TAC & rs199687009     & 0.904921 & 0.847312 &  0.000316765 & 0.00224748 & 0.88 & 1150490.6\\\\\n",
       "\t 1 & 677308 & TCTC & T   & 1:612688\\_TCTC\\_T & 0.979815 & 0.617496 &  0.001010140 & 0.00584836 & 0.90 &  739143.9\\\\\n",
       "\t 1 & 700905 & T    & C   & rs545945172     & 0.903972 & 0.721502 & -0.000918819 & 0.00247797 & 0.71 &  938048.0\\\\\n",
       "\t 1 & 713812 & A    & T   & rs201942322     & 0.882575 & 0.743729 & -0.000343830 & 0.00223761 & 0.87 &  963582.8\\\\\n",
       "\t 1 & 727034 & C    & T   & rs371628865     & 0.864997 & 0.694792 & -0.000533999 & 0.00219459 & 0.81 &  889007.0\\\\\n",
       "\t 1 & 727242 & G    & A   & rs61769339      & 0.889532 & 0.777266 & -0.000191469 & 0.00224530 & 0.92 & 1009306.8\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 11\n",
       "\n",
       "| chrom &lt;int&gt; | pos &lt;int&gt; | A1 &lt;chr&gt; | A2 &lt;chr&gt; | SNP &lt;chr&gt; | effect_allele_frequency &lt;dbl&gt; | INFO &lt;dbl&gt; | beta &lt;dbl&gt; | se &lt;dbl&gt; | p &lt;dbl&gt; | n_sample &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 |  97269 | T    | TAC | rs199687009     | 0.904921 | 0.847312 |  0.000316765 | 0.00224748 | 0.88 | 1150490.6 |\n",
       "| 1 | 677308 | TCTC | T   | 1:612688_TCTC_T | 0.979815 | 0.617496 |  0.001010140 | 0.00584836 | 0.90 |  739143.9 |\n",
       "| 1 | 700905 | T    | C   | rs545945172     | 0.903972 | 0.721502 | -0.000918819 | 0.00247797 | 0.71 |  938048.0 |\n",
       "| 1 | 713812 | A    | T   | rs201942322     | 0.882575 | 0.743729 | -0.000343830 | 0.00223761 | 0.87 |  963582.8 |\n",
       "| 1 | 727034 | C    | T   | rs371628865     | 0.864997 | 0.694792 | -0.000533999 | 0.00219459 | 0.81 |  889007.0 |\n",
       "| 1 | 727242 | G    | A   | rs61769339      | 0.889532 | 0.777266 | -0.000191469 | 0.00224530 | 0.92 | 1009306.8 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom pos    A1   A2  SNP             effect_allele_frequency INFO    \n",
       "1 1      97269 T    TAC rs199687009     0.904921                0.847312\n",
       "2 1     677308 TCTC T   1:612688_TCTC_T 0.979815                0.617496\n",
       "3 1     700905 T    C   rs545945172     0.903972                0.721502\n",
       "4 1     713812 A    T   rs201942322     0.882575                0.743729\n",
       "5 1     727034 C    T   rs371628865     0.864997                0.694792\n",
       "6 1     727242 G    A   rs61769339      0.889532                0.777266\n",
       "  beta         se         p    n_sample \n",
       "1  0.000316765 0.00224748 0.88 1150490.6\n",
       "2  0.001010140 0.00584836 0.90  739143.9\n",
       "3 -0.000918819 0.00247797 0.71  938048.0\n",
       "4 -0.000343830 0.00223761 0.87  963582.8\n",
       "5 -0.000533999 0.00219459 0.81  889007.0\n",
       "6 -0.000191469 0.00224530 0.92 1009306.8"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(unadjusted_GWAS_hg38_final)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "55ce4116-fc8d-4cc5-80af-e85064071791",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "fwrite(unadjusted_GWAS_hg38_final, \"/mnt/lustre/home/rl3328/rl3328/motor_qtl/GWAS/unadjusted_wp_ukb.sumstats_hg38.tsv.gz\", sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b64ef2b6-5be0-4079-a41e-094932f6a832",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# Method 1: Process header and data separately\n",
    "(zcat /mnt/lustre/home/rl3328/rl3328/motor_qtl/GWAS/unadjusted_wp_ukb.sumstats_hg38.tsv.gz | head -1 | sed 's/^/#/'; \\\n",
    " zcat /mnt/lustre/home/rl3328/rl3328/motor_qtl/GWAS/unadjusted_wp_ukb.sumstats_hg38.tsv.gz | tail -n +2 | \\\n",
    " awk 'BEGIN{OFS=\"\\t\"} {$2=int($2); print}' | sort -k1,1V -k2,2n) | \\\n",
    "bgzip > /mnt/lustre/home/rl3328/rl3328/motor_qtl/GWAS/unadjusted_wp_ukb.sumstats_hg38_sorted.tsv.gz\n",
    "\n",
    "tabix -s 1 -b 2 -e 2 /mnt/lustre/home/rl3328/rl3328/motor_qtl/GWAS/unadjusted_wp_ukb.sumstats_hg38_sorted.tsv.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88a1139c-783b-42c9-bab1-f3a2556f35e8",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
